--- a/src/main/java/io/papermc/paper/ServerBuildInfo.java
+++ b/src/main/java/io/papermc/paper/ServerBuildInfo.java
@@ -25,6 +_,14 @@
      */
     Key BRAND_PURPUR_ID = Key.key("purpurmc", "purpur");
     // Purpur end
+
+    // Deepslate start
+    /**
+     * The brand id for DeepslateMC.
+     */
+    Key BRAND_DEEPSLATE_ID = Key.key("pascalpex", "deepslate");
+    // Deepslate end
+
     /**
      * Gets the {@code ServerBuildInfo}.
      *
