From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Sun, 30 Jan 2022 14:40:49 +0100
Subject: [PATCH] Spawn functions


diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
index aa27acb098b85bb0c7a1341277e536f7e2621183..9dc401fdb1c8a11f4fe968a8adaf2ec4a27c01bd 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
@@ -267,6 +267,13 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
         MinecraftServer.getServer().server.getCommandMap().register("wartung", "Deepslate", new WartungCommand("wartung"));
         MinecraftServer.getServer().server.getCommandMap().register("help", "Deepslate", new HelpCommand("help"));
         MinecraftServer.getServer().server.getCommandMap().register("hilfe", "Deepslate", new HelpCommand("hilfe"));
+
+        MinecraftServer.getServer().server.getCommandMap().register("setspawn", "Deepslate", new SpawnCommand("setspawn"));
+        MinecraftServer.getServer().server.getCommandMap().register("spawn", "Deepslate", new SpawnCommand("spawn"));
+        MinecraftServer.getServer().server.getCommandMap().register("lobby", "Deepslate", new SpawnCommand("lobby"));
+        MinecraftServer.getServer().server.getCommandMap().register("l", "Deepslate", new SpawnCommand("l"));
+        MinecraftServer.getServer().server.getCommandMap().register("hub", "Deepslate", new SpawnCommand("hub"));
+        MinecraftServer.getServer().server.getCommandMap().register("hubschrauber", "Deepslate", new SpawnCommand("hubschrauber"));
         //Deepslate end
 
         this.setPvpAllowed(dedicatedserverproperties.pvp);
diff --git a/src/main/java/net/minecraft/server/players/PlayerList.java b/src/main/java/net/minecraft/server/players/PlayerList.java
index 78c60d2e0b852db70aac84315d7d6610a5c1293a..ca0a2d17ad76fef045ea4bd193c81c84dba88f5c 100644
--- a/src/main/java/net/minecraft/server/players/PlayerList.java
+++ b/src/main/java/net/minecraft/server/players/PlayerList.java
@@ -91,6 +91,7 @@ import net.minecraft.world.scores.Objective;
 import net.minecraft.world.scores.PlayerTeam;
 import net.minecraft.world.scores.Scoreboard; // Paper
 import net.minecraft.world.scores.Team;
+import net.pascalpex.deepslatemc.commands.SpawnCommand;
 import net.pascalpex.deepslatemc.commands.WartungCommand;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
@@ -362,6 +363,7 @@ public abstract class PlayerList {
 
         PlayerJoinEvent playerJoinEvent = new PlayerJoinEvent(bukkitPlayer, PaperAdventure.asAdventure(chatmessage)); // Paper - Adventure
         WartungCommand.playerJoin(bukkitPlayer);
+        SpawnCommand.playerJoin(bukkitPlayer);
         this.cserver.getPluginManager().callEvent(playerJoinEvent);
 
         if (!player.connection.connection.isConnected()) {
diff --git a/src/main/java/net/pascalpex/deepslatemc/Config.java b/src/main/java/net/pascalpex/deepslatemc/Config.java
index e26831d8d26081c95e9b5452057b35bfc186f13b..70a82964ac31cab89a0d27e1944f4b5aba6b98c8 100644
--- a/src/main/java/net/pascalpex/deepslatemc/Config.java
+++ b/src/main/java/net/pascalpex/deepslatemc/Config.java
@@ -1,10 +1,11 @@
 package net.pascalpex.deepslatemc;
-    
-import org.bukkit.ChatColor;
+
+import org.bukkit.Bukkit;
+import org.bukkit.Location;
+import org.bukkit.World;
 import org.bukkit.configuration.InvalidConfigurationException;
 import org.bukkit.configuration.file.FileConfiguration;
 import org.bukkit.configuration.file.YamlConfiguration;
-import org.bukkit.inventory.ItemStack;
 
 import java.io.File;
 import java.io.IOException;
@@ -30,6 +31,7 @@ public class Config {
                 message.add("&9&l/warp &9&lInfos &r&6nachlesen. Wende dich bei");
                 message.add("&6weiteren Fragen an ein &9Teammitglied.");
                 config.set("helpMessage", message);
+                config.set("spawnOnJoin", false);
                 save();
             }
             config.load(configFile);
@@ -46,10 +48,35 @@ public class Config {
         }
     }
 
+
+    public static boolean getSpawnOnJoin() {
+        return config.getBoolean("spawnOnJoin");
+    }
+
     public static boolean getOpActive() {
         return config.getBoolean("opCommandActive");
     }
 
+    public static void setSpawn(Location loc) {
+        config.set("spawn" + ".world", loc.getWorld().getName());
+        config.set("spawn" + ".X", loc.getX());
+        config.set("spawn" + ".Y", loc.getY());
+        config.set("spawn" + ".Z", loc.getZ());
+        config.set("spawn" + ".pitch",(double) loc.getPitch());
+        config.set("spawn" + ".yaw",(double) loc.getYaw());
+        save();
+    }
+
+    public static Location getSpawn() {
+        World world = Bukkit.getWorld(config.getString("spawn" + ".world"));
+        double x = config.getDouble("spawn" + ".X");
+        double y = config.getDouble("spawn" + ".Y");
+        double z = config.getDouble("spawn" + ".Z");
+        float pitch = (float) config.getDouble("spawn" + ".pitch");
+        float yaw = (float) config.getDouble("spawn" + ".yaw");
+        return new Location(world, x, y, z, yaw, pitch);
+    }
+
     public static String getHelpMessage() {
         List<String> messages = (List<String>) config.getList("helpMessage");
         String message = "";
diff --git a/src/main/java/net/pascalpex/deepslatemc/commands/SpawnCommand.java b/src/main/java/net/pascalpex/deepslatemc/commands/SpawnCommand.java
new file mode 100644
index 0000000000000000000000000000000000000000..b808c70ce97bb59aba2182fc3f3117f77f76792e
--- /dev/null
+++ b/src/main/java/net/pascalpex/deepslatemc/commands/SpawnCommand.java
@@ -0,0 +1,54 @@
+package net.pascalpex.deepslatemc.commands;
+
+import net.pascalpex.deepslatemc.Config;
+import org.bukkit.ChatColor;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+import org.jetbrains.annotations.NotNull;
+
+public class SpawnCommand extends Command {
+
+    public SpawnCommand(@NotNull String name) {
+        super(name);
+        this.description = "Teleportiert dich zum Spawn";
+        this.usageMessage = "/spawn";
+        this.setPermission("deepslate.spawn");
+    }
+
+    public static void playerJoin(Player player) {
+        if(Config.getSpawnOnJoin()) {
+            player.teleport(Config.getSpawn());
+        }
+    }
+
+    @Override
+    public boolean execute(@NotNull CommandSender sender, @NotNull String label, @NotNull String[] args) {
+        String prefix = Config.getPrefix() + " " + ChatColor.DARK_AQUA;
+        if(sender instanceof Player) {
+            
+            Player player = (Player) sender;
+            if(label.equalsIgnoreCase("setspawn")) {
+                if(player.hasPermission("deepslate.setspawn")) {
+                    Config.setSpawn(player.getLocation());
+                    player.sendMessage(prefix + ChatColor.GREEN + "Spawn erfolgreich gesetzt");
+                } else {
+                    player.sendMessage(prefix + ChatColor.RED + "Dafür hast du keine Rechte");
+                }
+            }
+            if(label.equalsIgnoreCase("l") || label.equalsIgnoreCase("hub") || label.equalsIgnoreCase("lobby") || label.equalsIgnoreCase("spawn") || label.equalsIgnoreCase("hubschrauber")) {
+                if(player.hasPermission("deepslate.spawn")) {
+                    player.teleport(Config.getSpawn());
+                    sender.sendMessage(prefix + ChatColor.GOLD + "Du wurdest zum Spawn teleportiert");
+                } else {
+                    player.sendMessage(prefix + ChatColor.RED + "Dafür hast du keine Rechte");
+                }
+            }
+            
+        } else {
+            sender.sendMessage(prefix + "Dieser Befehl ist nur für Spieler geeignet");
+        }
+        return true;
+    }
+
+}
