--- a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
+++ b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
@@ -512,6 +_,17 @@
         if (cancelledItem) {
             event.setUseItemInHand(Result.DENY);
         }
+        // Deepslate start - Configurable knowledgebook click
+        if (de.pascalpex.deepslatemc.files.Config.getPreventKnowledgebookClick()) {
+            if (event.hasItem()) {
+                if (event.getAction() == Action.RIGHT_CLICK_AIR || event.getAction() == Action.RIGHT_CLICK_BLOCK) {
+                    if (event.getItem().getType() == Material.KNOWLEDGE_BOOK) {
+                        event.setCancelled(true);
+                    }
+                }
+            }
+        }
+        // Deepslate end
         craftServer.getPluginManager().callEvent(event);
 
         // Purpur start - Ridables
