--- a/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/net/minecraft/server/dedicated/DedicatedServer.java
@@ -239,6 +_,7 @@
         io.papermc.paper.command.PaperCommands.registerCommands(this); // Paper - setup /paper command
         org.galemc.gale.command.GaleCommands.registerCommands(this); // Gale - Gale commands - register commands
         org.dreeam.leaf.command.LeafCommands.registerCommands(this); // Leaf - Leaf commands
+        de.pascalpex.deepslatemc.commands.DeepslateCmdRegisterer.registerCommands(this); // Deepslate - register commands
         this.server.spark.registerCommandBeforePlugins(this.server); // Paper - spark
         com.destroystokyo.paper.Metrics.PaperMetrics.startMetrics(); // Paper - start metrics
         /*// Purpur start - Purpur config files // Purpur - Configurable void damage height and damage
