--- a/src/main/java/io/papermc/paper/ServerBuildInfoImpl.java
+++ b/src/main/java/io/papermc/paper/ServerBuildInfoImpl.java
@@ -37,6 +_,7 @@
     private static final String BRAND_PURPUR_NAME = "Purpur";
     private static final String BRAND_LEAF_NAME = "Leaf";
     // Leaf end - Rebrand & Fix Pufferfish and Purpur patches
+    private static final String BRAND_DEEPSLATE_NAME = "DeepslateMC"; // Deepslate
 
     private static final String BUILD_DEV = "DEV";
 
@@ -48,9 +_,9 @@
         this(
             getManifestAttribute(manifest, ATTRIBUTE_BRAND_ID)
                 .map(Key::key)
-                .orElse(BRAND_LEAF_ID), // Gale - branding changes // Leaf - Rebrand
+                .orElse(BRAND_DEEPSLATE_ID), // Gale - branding changes // Leaf - Rebrand // Deepslate
             getManifestAttribute(manifest, ATTRIBUTE_BRAND_NAME)
-                .orElse(BRAND_LEAF_NAME), // Gale - branding changes // Leaf - Rebrand
+                .orElse(BRAND_DEEPSLATE_NAME), // Gale - branding changes // Leaf - Rebrand // Deepslate
             SharedConstants.getCurrentVersion().id(),
             SharedConstants.getCurrentVersion().name(),
             getManifestAttribute(manifest, ATTRIBUTE_BUILD_NUMBER)
@@ -72,7 +_,8 @@
             || brandId.equals(BRAND_PAPER_ID)
             || brandId.equals(BRAND_GALE_ID) // Gale - branding changes
             || brandId.equals(BRAND_PUFFERFISH_ID)
-            || brandId.equals(BRAND_PURPUR_ID);
+            || brandId.equals(BRAND_PURPUR_ID)
+            || brandId.equals(BRAND_LEAF_ID); // Deepslate
             // Leaf end - Rebrand & Fix Pufferfish and Purpur patches
     }
 
