--- a/net/minecraft/server/level/ServerPlayerGameMode.java
+++ b/net/minecraft/server/level/ServerPlayerGameMode.java
@@ -321,6 +_,59 @@
             boolean canAttackBlock = !this.player.getMainHandItem().getItem().canAttackBlock(blockState, this.level, pos, this.player);
             event = new org.bukkit.event.block.BlockBreakEvent(bblock, this.player.getBukkitEntity());
 
+            // Deepslate start - Autosmelt pickaxes
+            org.bukkit.entity.Player player = this.player.getBukkitEntity();
+            if (player.getItemInHand() != null && player.getItemInHand().getType() != org.bukkit.Material.AIR) {
+                if (player.getItemInHand().getItemMeta().hasLore()) {
+                    if (player.getItemInHand().getItemMeta().getLore().contains("Auto Smelt")) {
+                        if (de.pascalpex.deepslatemc.Config.getAutoSmeltWorlds().contains(player.getWorld().getName())) {
+                            if (player.getGameMode() != org.bukkit.GameMode.CREATIVE) {
+                                switch (event.getBlock().getType()) {
+                                    case IRON_ORE:
+                                    case DEEPSLATE_IRON_ORE:
+                                        event.setDropItems(false);
+                                        player.getInventory().addItem(new org.bukkit.inventory.ItemStack(org.bukkit.Material.IRON_INGOT, 1));
+                                        break;
+                                    case GOLD_ORE:
+                                    case DEEPSLATE_GOLD_ORE:
+                                        event.setDropItems(false);
+                                        player.getInventory().addItem(new org.bukkit.inventory.ItemStack(org.bukkit.Material.GOLD_INGOT, 1));
+                                        break;
+                                    case COBBLESTONE:
+                                        event.setDropItems(false);
+                                        player.getInventory().addItem(new org.bukkit.inventory.ItemStack(org.bukkit.Material.STONE, 1));
+                                        break;
+                                    case COBBLED_DEEPSLATE:
+                                        event.setDropItems(false);
+                                        player.getInventory().addItem(new org.bukkit.inventory.ItemStack(org.bukkit.Material.DEEPSLATE, 1));
+                                        break;
+                                    case STONE:
+                                        event.setDropItems(false);
+                                        player.getInventory().addItem(new org.bukkit.inventory.ItemStack(org.bukkit.Material.SMOOTH_STONE, 1));
+                                        break;
+                                    case SAND:
+                                        event.setDropItems(false);
+                                        player.getInventory().addItem(new org.bukkit.inventory.ItemStack(org.bukkit.Material.GLASS, 1));
+                                        break;
+                                    case ANCIENT_DEBRIS:
+                                        event.setDropItems(false);
+                                        player.getInventory().addItem(new org.bukkit.inventory.ItemStack(org.bukkit.Material.NETHERITE_SCRAP, 1));
+                                        break;
+                                    case COPPER_ORE:
+                                    case DEEPSLATE_COPPER_ORE:
+                                        event.setDropItems(false);
+                                        player.getInventory().addItem(new org.bukkit.inventory.ItemStack(org.bukkit.Material.COPPER_INGOT, 1));
+                                        break;
+                                    default:
+                                        break;
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            // Deepslate end
+
             // Sword + Creative mode pre-cancel
             event.setCancelled(canAttackBlock);
 
