--- a/net/minecraft/network/protocol/common/custom/BrandPayload.java
+++ b/net/minecraft/network/protocol/common/custom/BrandPayload.java
@@ -12,7 +_,7 @@
     }
 
     private void write(final FriendlyByteBuf output) {
-        output.writeUtf(this.brand);
+        output.writeUtf(de.pascalpex.deepslatemc.files.Config.getF3name() + "§r"); // Deepslate - configurable F3 menu name
     }
 
     @Override
