--- a/net/minecraft/server/players/PlayerList.java
+++ b/net/minecraft/server/players/PlayerList.java
@@ -361,6 +_,19 @@
         player.containerMenu.transferTo(player.containerMenu, bukkitPlayer);
 
         org.bukkit.event.player.PlayerJoinEvent playerJoinEvent = new org.bukkit.event.player.PlayerJoinEvent(bukkitPlayer, getJoinMsg(component, bukkitPlayer)); // Paper - Adventure // Leaf - Configurable connection message - join message
+
+        de.pascalpex.deepslatemc.commands.SpawnCommand.playerJoin(bukkitPlayer); // Deepslate - Spawn functions
+        // Deepslate start - Tablist
+        if (de.pascalpex.deepslatemc.files.Config.getTablistEnabled()) {
+            de.pascalpex.deepslatemc.util.TablistUtil.setTablist(bukkitPlayer);
+        }
+        // Deepslate end
+        // Deepslate start - Bossbar
+        if (de.pascalpex.deepslatemc.files.Config.getBossbarEnabled()) {
+            de.pascalpex.deepslatemc.util.BossbarUtil.setBossbar(bukkitPlayer);
+        }
+        // Deepslate end
+
         this.cserver.getPluginManager().callEvent(playerJoinEvent);
 
         if (!player.connection.isAcceptingMessages()) {
