--- a/net/minecraft/server/players/PlayerList.java
+++ b/net/minecraft/server/players/PlayerList.java
@@ -444,6 +_,19 @@
         player.containerMenu.transferTo(player.containerMenu, bukkitPlayer);
 
         org.bukkit.event.player.PlayerJoinEvent playerJoinEvent = new org.bukkit.event.player.PlayerJoinEvent(bukkitPlayer, getJoinMsg(mutableComponent, bukkitPlayer)); // Paper - Adventure // Leaf - Configurable connection message - join message
+
+        de.pascalpex.deepslatemc.commands.SpawnCommand.playerJoin(bukkitPlayer); // Deepslate - Spawn functions
+        // Deepslate start - Tablist
+        if (de.pascalpex.deepslatemc.files.Config.getTablistEnabled()) {
+            de.pascalpex.deepslatemc.util.TablistUtil.setTablist(bukkitPlayer);
+        }
+        // Deepslate end
+        // Deepslate start - Bossbar
+        if (de.pascalpex.deepslatemc.files.Config.getBossbarEnabled()) {
+            de.pascalpex.deepslatemc.util.BossbarUtil.setBossbar(bukkitPlayer);
+        }
+        // Deepslate end
+
         this.cserver.getPluginManager().callEvent(playerJoinEvent);
 
         if (!player.connection.isAcceptingMessages()) {
@@ -849,6 +_,15 @@
             }
         }
         this.cserver.getPluginManager().callEvent(event);
+        // Deepslate start - Maintenance mode
+        if(de.pascalpex.deepslatemc.files.Config.getMaintenanceMode()) {
+            if(!player.hasPermission("deepslate.maintenancebypass")) {
+                net.kyori.adventure.text.Component prefix = de.pascalpex.deepslatemc.files.MessagesFile.getMessage(de.pascalpex.deepslatemc.files.MessagesEntry.PREFIX);
+                net.kyori.adventure.text.Component kickMessage = de.pascalpex.deepslatemc.files.MessagesFile.getMessage(de.pascalpex.deepslatemc.files.MessagesEntry.MAINTENANCE_KICK);
+                event.disallow(org.bukkit.event.player.PlayerLoginEvent.Result.KICK_OTHER, prefix.appendSpace().append(kickMessage));
+            }
+        }
+        // Deepslate end
         if (event.getResult() != org.bukkit.event.player.PlayerLoginEvent.Result.ALLOWED) {
             loginlistener.disconnect(io.papermc.paper.adventure.PaperAdventure.asVanilla(event.kickMessage())); // Paper - Adventure
             return null;
@@ -896,7 +_,8 @@
         // Leaf start - SparklyPaper - parallel world ticking mod (make configurable)
         if (org.dreeam.leaf.config.modules.async.SparklyPaperParallelWorldTicking.enabled) {
             // SparklyPaper - parallel world ticking (additional concurrency issues logs)
-            System.out.println("respawning player - current player container is " + player.containerMenu + " but their inventory is " + player.inventoryMenu);
+            if (org.dreeam.leaf.config.modules.async.SparklyPaperParallelWorldTicking.logContainerCreationStacktraces) // Deepslate - Only log player respawns if container logs are on
+                System.out.println("respawning player - current player container is " + player.containerMenu + " but their inventory is " + player.inventoryMenu);
             if (location != null) // Leaf - THIS CAN BE NULL; see PlayerList::respawn(ServerPlayer, boolean, Entity.RemovalReason, PlayerRespawnEvent.RespawnReason)
                 ca.spottedleaf.moonrise.common.util.TickThread.ensureOnlyTickThread("Cannot respawn player off-main, from world " + player.serverLevel().getWorld().getName() + " to world " + location.getWorld().getName());
             else
