--- a/net/minecraft/commands/Commands.java
+++ b/net/minecraft/commands/Commands.java
@@ -153,6 +_,10 @@
     }
     public Commands(Commands.CommandSelection selection, CommandBuildContext context, final boolean modern) {
     // Paper end - Brigadier API - modern minecraft overloads that do not use redirects but are copies instead
+        // Deepslate start
+        de.pascalpex.deepslatemc.files.Config.load();
+        de.pascalpex.deepslatemc.files.MessagesFile.load();
+        // Deepslate end
         AdvancementCommands.register(this.dispatcher);
         AttributeCommand.register(this.dispatcher, context);
         ExecuteCommand.register(this.dispatcher, context);
@@ -175,7 +_,7 @@
         GameModeCommand.register(this.dispatcher);
         GameRuleCommand.register(this.dispatcher, context);
         GiveCommand.register(this.dispatcher, context);
-        HelpCommand.register(this.dispatcher);
+        //HelpCommand.register(this.dispatcher); Deepslate - configurable help command
         ItemCommands.register(this.dispatcher, context);
         KickCommand.register(this.dispatcher);
         KillCommand.register(this.dispatcher);
@@ -236,7 +_,9 @@
             BanListCommands.register(this.dispatcher);
             BanPlayerCommands.register(this.dispatcher);
             DeOpCommands.register(this.dispatcher);
-            OpCommand.register(this.dispatcher);
+            if (de.pascalpex.deepslatemc.files.Config.getOpActive()) { // Deepslate - Toggleable op command
+                OpCommand.register(this.dispatcher);
+            }
             PardonCommand.register(this.dispatcher);
             PardonIpCommand.register(this.dispatcher);
             PerfCommand.register(this.dispatcher);
