--- a/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -1516,7 +_,7 @@
                                             io.papermc.paper.event.player.PlayerFailMoveEvent event = fireFailMove(io.papermc.paper.event.player.PlayerFailMoveEvent.FailReason.MOVED_TOO_QUICKLY,
                                                     toX, toY, toZ, toYaw, toPitch, true);
                                             if (!event.isAllowed()) {
-                                                if (event.getLogWarning()) {
+                                                if (de.pascalpex.deepslatemc.files.Config.getMovedQuickly() && event.getLogWarning()) { // Deepslate
                                                 LOGGER.warn("{} moved too quickly! {},{},{}", this.player.getName().getString(), d3, d4, d5);
                                                 }
                                                 this.teleport(
@@ -1594,7 +_,7 @@
                                             toX, toY, toZ, toYaw, toPitch, true);
                                     if (!event.isAllowed()) {
                                         movedWrongly = true;
-                                        if (event.getLogWarning())
+                                        if (de.pascalpex.deepslatemc.files.Config.getMovedWrongly() && event.getLogWarning()) // Deepslate
                                      // Paper end
                                     LOGGER.warn("{} moved wrongly!, ({})", this.player.getName().getString(), verticalDelta); // Purpur - AFK API
                                     } // Paper
