--- a/net/minecraft/server/network/ServerConfigurationPacketListenerImpl.java
+++ b/net/minecraft/server/network/ServerConfigurationPacketListenerImpl.java
@@ -106,7 +_,11 @@
         // Paper start
         org.bukkit.craftbukkit.CraftServerLinks links = new org.bukkit.craftbukkit.CraftServerLinks(serverLinks);
         new org.bukkit.event.player.PlayerLinksSendEvent(this.paperConnection, links).callEvent();
-        if (!links.getServerLinks().isEmpty()) {
+        if (!links.getServerLinks().isEmpty() || !de.pascalpex.deepslatemc.util.ServerLinkUtil.getLinks().isEmpty()) { // Deepslate - server links
+            // Deepslate start - server links
+            java.util.Map<net.kyori.adventure.text.Component, java.net.URI> deepslateLinks = de.pascalpex.deepslatemc.util.ServerLinkUtil.getLinks();
+            deepslateLinks.keySet().forEach(key -> links.addLink(key, deepslateLinks.get(key)));
+            // Deepslate end - server links
             this.send(new ClientboundServerLinksPacket(links.getServerLinks().untrust()));
         // Paper end
         }
