--- a/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -1629,7 +_,7 @@
                                             io.papermc.paper.event.player.PlayerFailMoveEvent event = fireFailMove(io.papermc.paper.event.player.PlayerFailMoveEvent.FailReason.MOVED_TOO_QUICKLY,
                                                     targetX, targetY, targetZ, targetYRot, targetXRot, true);
                                             if (!event.isAllowed()) {
-                                                if (event.getLogWarning()) {
+                                                if (de.pascalpex.deepslatemc.files.Config.getMovedQuickly() && event.getLogWarning()) { // Deepslate
                                             LOGGER.warn("{} moved too quickly! {},{},{}", this.player.getPlainTextName(), xDist, yDist, zDist);
                                                 }
                                                 this.teleport(
@@ -1695,7 +_,7 @@
                                             targetX, targetY, targetZ, targetYRot, targetXRot, true);
                                     if (!event.isAllowed()) {
                                         movedWrongly = true;
-                                        if (event.getLogWarning())
+                                        if (de.pascalpex.deepslatemc.files.Config.getMovedWrongly() && event.getLogWarning()) // Deepslate
                                      // Paper end
                                     LOGGER.warn("{} moved wrongly!, ({})", this.player.getPlainTextName(), oyDist); // Purpur - AFK API
                                     } // Paper
