From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Mon, 21 Mar 2022 15:07:19 +0100
Subject: [PATCH] Discord command


diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
index b7dcc8d838eaf5dcbeae2e53868e6113913ab93e..9606f09c82fcc40a287289ca3d592812f6196209 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
@@ -59,6 +59,7 @@ import net.minecraft.world.level.GameType;
 import net.minecraft.world.level.block.entity.SkullBlockEntity;
 import net.minecraft.world.level.storage.LevelStorageSource;
 import net.pascalpex.deepslatemc.commands.DeepslateCommand;
+import net.pascalpex.deepslatemc.commands.DiscordCommand;
 import org.slf4j.Logger;
 
 // CraftBukkit start
@@ -260,6 +261,8 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
 
         // Deepslate start
         MinecraftServer.getServer().server.getCommandMap().register("deepslate", "Deepslate", new DeepslateCommand("deepslate"));
+        MinecraftServer.getServer().server.getCommandMap().register("discord", "Deepslate", new DiscordCommand("discord"));
+        MinecraftServer.getServer().server.getCommandMap().register("dc", "Deepslate", new DiscordCommand("dc"));
         // Deepslate end
 
         this.setPvpAllowed(dedicatedserverproperties.pvp);
diff --git a/src/main/java/net/pascalpex/deepslatemc/Config.java b/src/main/java/net/pascalpex/deepslatemc/Config.java
index 994b992d2acb48576af874137df074bb7c659c88..6240a269ab54ebbb9729d737e7f0cedfe2e2139a 100644
--- a/src/main/java/net/pascalpex/deepslatemc/Config.java
+++ b/src/main/java/net/pascalpex/deepslatemc/Config.java
@@ -26,6 +26,10 @@ public class Config {
                     save();
                 }
                 config.load(configFile);
+                if(!config.contains("discordLink")) {
+                    config.set("discordLink", "https://google.de/");
+                }
+                save();
             } catch (IOException | InvalidConfigurationException e) {
                 e.printStackTrace();
             }
@@ -39,4 +43,8 @@ public class Config {
             }
         }
 
+        public static String getDiscordLink() {
+            return config.getString("discordLink");
+        }
+
 }
diff --git a/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java b/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
index 0016a0e047060c670dc2d1efe20afc8f5ea4288e..1090737d07457f9884162ce15c324d4816441111 100644
--- a/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
+++ b/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
@@ -35,6 +35,12 @@ public class MessagesFile {
                 if(!config.contains("configReloaded")) {
                     config.set("configReloaded", "&aConfig got reloaded successfully");
                 }
+                if(!config.contains("onlyForPlayers")) {
+                    config.set("onlyForPlayers", "&cThis command can only be used by players");
+                }
+                if(!config.contains("discordMessage")) {
+                    config.set("discordMessage", "&6The link to our Discord server: &b%link%");
+                }
                 save();
             } catch (IOException | InvalidConfigurationException e) {
                 e.printStackTrace();
@@ -64,4 +70,14 @@ public class MessagesFile {
             return prefix.replace("&", "§");
         }
 
+        public static String getOnlyForPlayers() {
+            String prefix = config.getString("onlyForPlayers");
+            return prefix.replace("&", "§");
+        }
+
+        public static String getDiscordMessage() {
+            String prefix = config.getString("discordMessage");
+            return prefix.replace("&", "§");
+        }
+
 }
diff --git a/src/main/java/net/pascalpex/deepslatemc/commands/DiscordCommand.java b/src/main/java/net/pascalpex/deepslatemc/commands/DiscordCommand.java
new file mode 100644
index 0000000000000000000000000000000000000000..24717b7f0ead489c38635c7db36e22b88a2bd387
--- /dev/null
+++ b/src/main/java/net/pascalpex/deepslatemc/commands/DiscordCommand.java
@@ -0,0 +1,32 @@
+package net.pascalpex.deepslatemc.commands;
+    
+import net.pascalpex.deepslatemc.Config;
+import net.pascalpex.deepslatemc.MessagesFile;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+import org.jetbrains.annotations.NotNull;
+
+public class DiscordCommand extends Command {
+
+    public DiscordCommand(String name) {
+        super(name);
+        this.description = "Discord link command";
+        this.usageMessage = "/discord";
+    }
+
+    @Override
+    public boolean execute(@NotNull CommandSender sender, @NotNull String commandLabel, @NotNull String[] args) {
+        String prefix = MessagesFile.getPrefix() + " ";
+        if (commandLabel.equalsIgnoreCase("dc") || commandLabel.equalsIgnoreCase("discord")) {
+            if (sender instanceof Player) {
+                Player player = (Player) sender;
+                String link = Config.getDiscordLink();
+                player.sendMessage(prefix + MessagesFile.getDiscordMessage().replace("%link%", link));
+            } else {
+                sender.sendMessage(prefix + MessagesFile.getOnlyForPlayers());
+            }
+        }
+        return true;
+    }
+}
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 6516f6da5857a3903db2dc4d6c701c94827c3f78..6ebda0fede3bf5e8a7686d5d06e34a051ea1c5ec 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -112,6 +112,7 @@ import net.minecraft.world.level.storage.PrimaryLevelData;
 import net.minecraft.world.level.storage.loot.LootTables;
 import net.minecraft.world.phys.Vec3;
 import net.pascalpex.deepslatemc.commands.DeepslateCommand;
+import net.pascalpex.deepslatemc.commands.DiscordCommand;
 import org.apache.commons.lang.Validate;
 import org.bukkit.BanList;
 import org.bukkit.Bukkit;
@@ -996,6 +997,8 @@ public final class CraftServer implements Server {
 
         // Deepslate start
         MinecraftServer.getServer().server.getCommandMap().register("deepslate", "Deepslate", new DeepslateCommand("deepslate"));
+        MinecraftServer.getServer().server.getCommandMap().register("discord", "Deepslate", new DiscordCommand("discord"));
+        MinecraftServer.getServer().server.getCommandMap().register("dc", "Deepslate", new DiscordCommand("dc"));
         // Deepslate end
 
         this.overrideAllCommandBlockCommands = this.commandsConfiguration.getStringList("command-block-overrides").contains("*");
