From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Mon, 21 Mar 2022 15:16:16 +0100
Subject: [PATCH] Buildworld command


diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
index 549f32a403c6f983ff1aa60c92db58c994ed5f52..aa8470ce91d3435df605d038db2521389655107b 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
@@ -58,6 +58,7 @@ import net.minecraft.world.level.GameRules;
 import net.minecraft.world.level.GameType;
 import net.minecraft.world.level.block.entity.SkullBlockEntity;
 import net.minecraft.world.level.storage.LevelStorageSource;
+import net.pascalpex.deepslatemc.commands.BuildworldCommand;
 import net.pascalpex.deepslatemc.commands.ClearchatCommand;
 import net.pascalpex.deepslatemc.commands.DeepslateCommand;
 import net.pascalpex.deepslatemc.commands.DiscordCommand;
@@ -266,6 +267,10 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
         MinecraftServer.getServer().server.getCommandMap().register("dc", "Deepslate", new DiscordCommand("dc"));
         MinecraftServer.getServer().server.getCommandMap().register("cc", "Deepslate", new ClearchatCommand("cc"));
         MinecraftServer.getServer().server.getCommandMap().register("clearchat", "Deepslate", new ClearchatCommand("clearchat"));
+        MinecraftServer.getServer().server.getCommandMap().register("setbauwelt", "Deepslate", new BuildworldCommand("setbauwelt"));
+        MinecraftServer.getServer().server.getCommandMap().register("bauwelt", "Deepslate", new BuildworldCommand("bauwelt"));
+        MinecraftServer.getServer().server.getCommandMap().register("setbuildworld", "Deepslate", new BuildworldCommand("setbuildworld"));
+        MinecraftServer.getServer().server.getCommandMap().register("buildworld", "Deepslate", new BuildworldCommand("buildworld"));
         // Deepslate end
 
         this.setPvpAllowed(dedicatedserverproperties.pvp);
diff --git a/src/main/java/net/pascalpex/deepslatemc/Config.java b/src/main/java/net/pascalpex/deepslatemc/Config.java
index 6240a269ab54ebbb9729d737e7f0cedfe2e2139a..5d65ca0019450b16102acb94756d14df523620e0 100644
--- a/src/main/java/net/pascalpex/deepslatemc/Config.java
+++ b/src/main/java/net/pascalpex/deepslatemc/Config.java
@@ -47,4 +47,13 @@ public class Config {
             return config.getString("discordLink");
         }
 
+        public static String getBuildworld() {
+            return config.getString("buildworld");
+        }
+
+        public static void setBuildworld(String worldName) {
+            config.set("buildworld", worldName);
+            save();
+        }
+
 }
diff --git a/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java b/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
index e2ba05e75440f78c60089b81a156f7c03c2bb0a5..036689bc447a1a143f0b336feab659b5ca03eea7 100644
--- a/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
+++ b/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
@@ -47,6 +47,15 @@ public class MessagesFile {
                 if(!config.contains("clearedChat")) {
                     config.set("clearedChat", "&6The chat got cleared by &b%clearer%");
                 }
+                if(!config.contains("buildworldSet")) {
+                    config.set("buildworldSet", "&aThe buildworld got set successfully");
+                }
+                if(!config.contains("buildworldWelcome")) {
+                    config.set("buildworldWelcome", "&aWelcome to the buildworld");
+                }
+                if(!config.contains("buildworldNotSet")) {
+                    config.set("buildworldNotSet", "&cThe buildworld is not set");
+                }
                 save();
             } catch (IOException | InvalidConfigurationException e) {
                 e.printStackTrace();
@@ -96,4 +105,17 @@ public class MessagesFile {
             return prefix.replace("&", "§");
         }
 
+        public static String getBuildworldSet() {
+            String prefix = config.getString("buildworldSet");
+            return prefix.replace("&", "§");
+        }
+        public static String getBuildworldWelcome() {
+            String prefix = config.getString("buildworldWelcome");
+            return prefix.replace("&", "§");
+        }
+        public static String getBuildworldNotSet() {
+            String prefix = config.getString("buildworldNotSet");
+            return prefix.replace("&", "§");
+        }
+
 }
diff --git a/src/main/java/net/pascalpex/deepslatemc/commands/BuildworldCommand.java b/src/main/java/net/pascalpex/deepslatemc/commands/BuildworldCommand.java
new file mode 100644
index 0000000000000000000000000000000000000000..f406beb184e03b67658eb2fe4ab3a3e4f1f29ffd
--- /dev/null
+++ b/src/main/java/net/pascalpex/deepslatemc/commands/BuildworldCommand.java
@@ -0,0 +1,52 @@
+package net.pascalpex.deepslatemc.commands;
+    
+import net.pascalpex.deepslatemc.Config;
+import net.pascalpex.deepslatemc.MessagesFile;
+import org.bukkit.Bukkit;
+import org.bukkit.World;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+import org.jetbrains.annotations.NotNull;
+
+public class BuildworldCommand extends Command {
+
+    public BuildworldCommand(@NotNull String name) {
+        super(name);
+        this.description = "Teleports builders to the buildworld";
+        this.usageMessage = "/buildworld";
+        this.setPermission("deepslate.buildworld");
+    }
+
+    @Override
+    public boolean execute(@NotNull CommandSender sender, @NotNull String label, @NotNull String[] args) {
+        String prefix = MessagesFile.getPrefix() + " ";
+        if (sender instanceof Player) {
+            Player player = (Player) sender;
+            if(label.equalsIgnoreCase("setbuildworld") || label.equalsIgnoreCase("setbauwelt")) {
+                if(player.hasPermission("deepslate.setbuildworld")) {
+                    World world = player.getLocation().getWorld();
+                    Config.setBuildworld(world.getName());
+                    player.sendMessage(prefix + MessagesFile.getBuildworldSet());
+                } else {
+                    player.sendMessage(prefix + MessagesFile.getNoPermissions());
+                }
+            }
+            if(label.equalsIgnoreCase("bauwelt") || label.equalsIgnoreCase("buildworld")) {
+                if(player.hasPermission("deepslate.buildworld")) {
+                    try {
+                        player.teleport(Bukkit.getWorld(Config.getBuildworld()).getSpawnLocation());
+                        player.sendMessage(prefix + MessagesFile.getBuildworldWelcome());
+                    } catch (Exception e) {
+                        player.sendMessage(prefix + MessagesFile.getBuildworldNotSet());
+                    }
+                } else {
+                    player.sendMessage(prefix + MessagesFile.getNoPermissions());
+                }
+            }
+        } else {
+            sender.sendMessage(prefix + MessagesFile.getOnlyForPlayers());
+        }
+        return true;
+    }
+}
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 248d0b92e2dffb12f3563305e2ec2fd2f6014690..57cbe970773a7e78526b429ae58f9c84af3ac7e0 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -111,6 +111,7 @@ import net.minecraft.world.level.storage.PlayerDataStorage;
 import net.minecraft.world.level.storage.PrimaryLevelData;
 import net.minecraft.world.level.storage.loot.LootTables;
 import net.minecraft.world.phys.Vec3;
+import net.pascalpex.deepslatemc.commands.BuildworldCommand;
 import net.pascalpex.deepslatemc.commands.ClearchatCommand;
 import net.pascalpex.deepslatemc.commands.DeepslateCommand;
 import net.pascalpex.deepslatemc.commands.DiscordCommand;
@@ -1002,6 +1003,10 @@ public final class CraftServer implements Server {
         MinecraftServer.getServer().server.getCommandMap().register("dc", "Deepslate", new DiscordCommand("dc"));
         MinecraftServer.getServer().server.getCommandMap().register("cc", "Deepslate", new ClearchatCommand("cc"));
         MinecraftServer.getServer().server.getCommandMap().register("clearchat", "Deepslate", new ClearchatCommand("clearchat"));
+        MinecraftServer.getServer().server.getCommandMap().register("setbauwelt", "Deepslate", new BuildworldCommand("setbauwelt"));
+        MinecraftServer.getServer().server.getCommandMap().register("bauwelt", "Deepslate", new BuildworldCommand("bauwelt"));
+        MinecraftServer.getServer().server.getCommandMap().register("setbuildworld", "Deepslate", new BuildworldCommand("setbuildworld"));
+        MinecraftServer.getServer().server.getCommandMap().register("buildworld", "Deepslate", new BuildworldCommand("buildworld"));
         // Deepslate end
 
         this.overrideAllCommandBlockCommands = this.commandsConfiguration.getStringList("command-block-overrides").contains("*");
