--- a/net/minecraft/server/MinecraftServer.java
+++ b/net/minecraft/server/MinecraftServer.java
@@ -1353,6 +_,7 @@
             org.spigotmc.WatchdogThread.hasStarted = true; // Paper
             org.dreeam.leaf.config.LeafConfig.regSparkExtraConfig(); // Leaf - Leaf config
             net.minecraft.world.level.block.Blocks.initPathType(); // Leaf - Cache block state tags
+            de.pascalpex.deepslatemc.files.Config.addSparkEntry(); // Deepslate - add deepslate.yml to spark
             // Paper start - Add onboarding message for initial server start
             if (io.papermc.paper.configuration.GlobalConfiguration.isFirstStart) {
                 LOGGER.info("*************************************************************************************");
@@ -1836,6 +_,19 @@
     private ServerStatus.Players buildPlayerStatus() {
         List<ServerPlayer> players = this.playerList.realPlayers; // Leaves - only real player
         int maxPlayers = this.getMaxPlayers();
+
+        // Deepslate start - Custom playerlist hover text
+        if(de.pascalpex.deepslatemc.files.Config.getPlayerlistHoverEnabled()) {
+            List<String> textList = de.pascalpex.deepslatemc.files.Config.getPlayerlistHoverLines();
+            ObjectArrayList<NameAndId> objectarraylist = new ObjectArrayList<>(textList.size());
+            for(String line : textList) {
+                GameProfile profile = new GameProfile(UUID.randomUUID(), line.replace("&", "§"));
+                objectarraylist.add(new NameAndId(profile));
+            }
+            return new ServerStatus.Players(maxPlayers, players.size(), objectarraylist);
+        }
+        // Deepslate end
+
         if (this.hidesOnlinePlayers()) {
             return new ServerStatus.Players(maxPlayers, players.size(), List.of());
         } else {
@@ -2265,12 +_,29 @@
 
     @Override
     public String getMotd() {
-        return net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer.legacySection().serialize(this.motd); // Paper - Adventure
+        // Deepslate start - MiniMessage MOTD
+        if(de.pascalpex.deepslatemc.files.Config.getMinimessageMotd()) {
+            return net.kyori.adventure.text.minimessage.MiniMessage.miniMessage().serialize(this.motd);
+        } else {
+            return net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer.legacySection().serialize(this.motd); // Paper - Adventure
+        }
+        // Deepslate end - MiniMessage MOTD
     }
 
     public void setMotd(String motd) {
         // Paper start - Adventure
-        this.motd = net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer.legacySection().deserializeOr(motd, net.kyori.adventure.text.Component.empty());
+        // Deepslate start - MiniMessage MOTD
+        if(de.pascalpex.deepslatemc.files.Config.getMinimessageMotd()) {
+            try {
+                this.motd = net.kyori.adventure.text.minimessage.MiniMessage.miniMessage().deserializeOr(motd, net.kyori.adventure.text.Component.empty());
+            } catch (net.kyori.adventure.text.minimessage.internal.parser.ParsingExceptionImpl e) {
+                LOGGER.error("You have enabled DeepslateMC's miniMessageMotdSupport but your motd contains legacy color codes. Will fallback to legacy colors.");
+                this.motd = net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer.legacySection().deserializeOr(motd, net.kyori.adventure.text.Component.empty());
+            }
+        } else {
+            this.motd = net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer.legacySection().deserializeOr(motd, net.kyori.adventure.text.Component.empty());
+        }
+        // Deepslate end - MiniMessage MOTD
     }
 
     public net.kyori.adventure.text.Component motd() {
