--- a/src/main/java/org/spigotmc/WatchdogThread.java
+++ b/src/main/java/org/spigotmc/WatchdogThread.java
@@ -82,15 +_,15 @@
                     We do not want people to report thread issues to Paper,
                     but we do want people to report thread issues to Gale.
                     */
-                    logger.log(Level.SEVERE, "The server has stopped responding! This is (probably) not a Paper bug. This could be a Leaf bug."); // Paper // Leaf
+                    logger.log(Level.SEVERE, "The server has stopped responding! This is (probably) not a Paper bug. This could be a DeepslateMC bug."); // Paper // Leaf // Deepslate
                     // Gale end - branding changes
                     logger.log(Level.SEVERE, "If you see a plugin in the Server thread dump below, then please report it to that author");
                     logger.log(Level.SEVERE, "\t *Especially* if it looks like HTTP or MySQL operations are occurring");
                     logger.log(Level.SEVERE, "If you see a world save or edit, then it means you did far more than your server can handle at once");
                     logger.log(Level.SEVERE, "\t If this is the case, consider increasing timeout-time in spigot.yml but note that this will replace the crash with LARGE lag spikes");
-                    logger.log(Level.SEVERE, "If you are unsure or think this is a Leaf bug, please report this to https://github.com/Winds-Studio/Leaf/issues - and if you think this is a Paper bug, please report this to https://github.com/PaperMC/Paper/issues"); // Gale - branding changes // Leaf
+                    logger.log(Level.SEVERE, "If you are unsure or think this is a DeepslateMC bug, please report this to https://github.com/Pascalpex/DeepslateMC/issues - and if you think this is a Paper bug, please report this to https://github.com/PaperMC/Paper/issues"); // Gale - branding changes // Leaf // Deepslate
                     logger.log(Level.SEVERE, "Be sure to include ALL relevant console errors and Minecraft crash reports");
-                    logger.log(Level.SEVERE, "Leaf version: " + Bukkit.getServer().getVersion()); // Gale - branding changes // Leaf
+                    logger.log(Level.SEVERE, "DeepslateMC version: " + Bukkit.getServer().getVersion()); // Gale - branding changes // Leaf // Deepslate
 
                     if (net.minecraft.world.level.Level.lastPhysicsProblem != null) {
                         logger.log(Level.SEVERE, "------------------------------");
@@ -115,13 +_,13 @@
                     We do not want people to report thread issues to Paper,
                     but we do want people to report thread issues to Gale.
                     */
-                    logger.log(Level.SEVERE, "--- DO NOT REPORT THIS TO PAPER - If you think this is a Leaf bug, please report it at https://github.com/Winds-Studio/Leaf/issues - THIS IS NOT A PAPER BUG OR CRASH - " + Bukkit.getServer().getVersion() + " ---"); // Leaf
+                    logger.log(Level.SEVERE, "--- DO NOT REPORT THIS TO PAPER - If you think this is a DeepslateMC bug, please report it at https://github.com/Pascalpex/DeepslateMC/issues - THIS IS NOT A PAPER BUG OR CRASH - " + Bukkit.getServer().getVersion() + " ---"); // Leaf // Deepslate
                     // Gale end - branding changes
                     logger.log(Level.SEVERE, "The server has not responded for " + (currentTime - lastTick) / 1000 + " seconds! Creating thread dump");
                 }
                 // Paper end - Different message for short timeout
                 logger.log(Level.SEVERE, "------------------------------");
-                logger.log(Level.SEVERE, "Server thread dump (Look for plugins here before reporting to Leaf!):"); // Paper // Gale - branding changes // Leaf
+                logger.log(Level.SEVERE, "Server thread dump (Look for plugins here before reporting to DeepslateMC!):"); // Paper // Gale - branding changes // Leaf // Deepslate
                 FeatureHooks.dumpAllChunkLoadInfo(MinecraftServer.getServer(), isLongTimeout); // Paper - log detailed tick information
                 WatchdogThread.dumpThread(ManagementFactory.getThreadMXBean().getThreadInfo(MinecraftServer.getServer().serverThread.getId(), Integer.MAX_VALUE), logger);
                 logger.log(Level.SEVERE, "------------------------------");
@@ -139,7 +_,7 @@
                     We do not want people to report thread issues to Paper,
                     but we do want people to report thread issues to Gale.
                     */
-                    logger.log(Level.SEVERE, "--- DO NOT REPORT THIS TO PAPER - If you think this is a Leaf bug, please report it at https://github.com/Winds-Studio/Leaf/issues - THIS IS NOT A PAPER BUG OR CRASH ---"); // Leaf
+                    logger.log(Level.SEVERE, "--- DO NOT REPORT THIS TO PAPER - If you think this is a DeepslateMC bug, please report it at https://github.com/Pascalpex/DeepslateMC/issues - THIS IS NOT A PAPER BUG OR CRASH ---"); // Leaf // Deepslate
                     // Gale end - branding changes
                 }
 
