From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Mon, 20 May 2024 16:23:30 +0200
Subject: [PATCH] Rebranding


diff --git a/src/main/java/io/papermc/paper/ServerBuildInfo.java b/src/main/java/io/papermc/paper/ServerBuildInfo.java
index 5873b42c198d7972e4147d019c790b612ae1b735..e9d37e9247f5bd62e5404043286277a25d82db4e 100644
--- a/src/main/java/io/papermc/paper/ServerBuildInfo.java
+++ b/src/main/java/io/papermc/paper/ServerBuildInfo.java
@@ -44,6 +44,14 @@ public interface ServerBuildInfo {
     Key BRAND_LEAF_ID = Key.key("winds-studio", "leaf");
     // Leaf end
 
+    // Deepslate start
+    /**
+     * The brand id for Gale.
+     */
+    Key BRAND_DEEPSLATE_ID = Key.key("pascalpex", "deepslate");
+    // Deepslate end
+
+
     /**
      * Gets the {@code ServerBuildInfo}.
      *
