From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Thu, 14 Jul 2022 13:59:24 +0200
Subject: [PATCH] Enable reload command


diff --git a/src/main/java/org/bukkit/command/defaults/ReloadCommand.java b/src/main/java/org/bukkit/command/defaults/ReloadCommand.java
index 3ec32b46264cfff857b50129b5e0fa5584943ec6..94bc77b727e10d45804484fc109cc63e76b6e20e 100644
--- a/src/main/java/org/bukkit/command/defaults/ReloadCommand.java
+++ b/src/main/java/org/bukkit/command/defaults/ReloadCommand.java
@@ -22,6 +22,11 @@ public class ReloadCommand extends BukkitCommand {
     public boolean execute(@NotNull CommandSender sender, @NotNull String currentAlias, @NotNull String[] args) { // Paper
         if (!testPermission(sender)) return true;
 
+        // Deepslate start - Re-enable reload command with extra warning
+        sender.sendMessage(ChatColor.DARK_RED + "Warning!" + ChatColor.RED + " The reload command can cause issues with certain plugins");
+        sender.sendMessage(ChatColor.DARK_RED + "Warning!" + ChatColor.RED + " A restart is recommended instead");
+        // Deepslate end
+
         // Paper start - Reload permissions.yml & require confirm
         boolean confirmed = System.getProperty("LetMeReload") != null;
         if (args.length == 1) {
