From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Mon, 20 May 2024 16:23:30 +0200
Subject: [PATCH] Rebranding


diff --git a/src/main/java/io/papermc/paper/ServerBuildInfo.java b/src/main/java/io/papermc/paper/ServerBuildInfo.java
index 0cafbbcb8fe294bcd84b93da8fba96368971dffd..da6c553e7e810c7e024b69c0207552c550da8332 100644
--- a/src/main/java/io/papermc/paper/ServerBuildInfo.java
+++ b/src/main/java/io/papermc/paper/ServerBuildInfo.java
@@ -43,6 +43,14 @@ public interface ServerBuildInfo {
     Key BRAND_LEAF_ID = Key.key("winds-studio", "leaf");
     // Leaf end
 
+    // Deepslate start
+    /**
+     * The brand id for Gale.
+     */
+    Key BRAND_DEEPSLATE_ID = Key.key("pascalpex", "deepslate");
+    // Deepslate end
+
+
     /**
      * Gets the {@code ServerBuildInfo}.
      *
