From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Mon, 20 May 2024 16:23:30 +0200
Subject: [PATCH] Rebranding


diff --git a/src/main/java/io/papermc/paper/ServerBuildInfo.java b/src/main/java/io/papermc/paper/ServerBuildInfo.java
index 9fa8d7fac5bf415b40c3a9a38108fa5476ed7619..bd37463a0c153efed8bcd2859eb41e783f634238 100644
--- a/src/main/java/io/papermc/paper/ServerBuildInfo.java
+++ b/src/main/java/io/papermc/paper/ServerBuildInfo.java
@@ -39,6 +39,14 @@ public interface ServerBuildInfo {
     Key BRAND_LEAF_ID = Key.key("winds-studio", "leaf");
     // Leaf end
 
+    // Deepslate start
+    /**
+     * The brand id for Gale.
+     */
+    Key BRAND_DEEPSLATE_ID = Key.key("pascalpex", "deepslate");
+    // Deepslate end
+
+
     /**
      * Gets the {@code ServerBuildInfo}.
      *
