From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Sat, 5 Feb 2022 19:42:56 +0100
Subject: [PATCH] Enable reload command


diff --git a/src/main/java/org/bukkit/command/defaults/ReloadCommand.java b/src/main/java/org/bukkit/command/defaults/ReloadCommand.java
index 535800197c43f43c30854d2b193107808dc3b6c7..2bff0ecafb4c8a33c8da91535a9f5a43548f0352 100644
--- a/src/main/java/org/bukkit/command/defaults/ReloadCommand.java
+++ b/src/main/java/org/bukkit/command/defaults/ReloadCommand.java
@@ -21,17 +21,8 @@ public class ReloadCommand extends BukkitCommand {
     @Override
     public boolean execute(@NotNull CommandSender sender, @NotNull String currentAlias, @NotNull String[] args) { // Paper
         if (!testPermission(sender)) return true;
-        // Yatopia start - disable reload command
-        if (Boolean.parseBoolean(System.getProperty("Mirai.DisableReloadCommand", "true"))) {
-            sender.sendMessage(ChatColor.RED + "Operation denied.");
-            sender.sendMessage(ChatColor.YELLOW + "Reload command SHOULD NEVER EVER EVER be used in whatever circumstances.");
-            sender.sendMessage(ChatColor.YELLOW + "Mirai has intentionally disabled it in order to stop you using it, instead of restarting your server.");
-            sender.sendMessage(ChatColor.GREEN + "---------------------------------------------");
-            sender.sendMessage(ChatColor.YELLOW + "RESTART YOUR SERVER AND NEVER USE /reload");
-            sender.sendMessage(ChatColor.YELLOW + "For plugin developers: learn what a HOTSWAP AGENT is and stop using /reload");
-            return true;
-        }
-        // Yatopia end
+        sender.sendMessage(ChatColor.DARK_RED + "Warning!" + ChatColor.RED + " The reload command can cause issues with certain plugins");
+        sender.sendMessage(ChatColor.DARK_RED + "Warning!" + ChatColor.RED + " A restart is recommended instead");
 
         // Paper start - Reload permissions.yml & require confirm
         boolean confirmed = System.getProperty("LetMeReload") != null;
