From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Sun, 30 Jan 2022 14:03:03 +0100
Subject: [PATCH] Deepslate-Command


diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
index c434373557639463b862615552183a61665f01d0..b70c9a2446489d5217c93c977da5dfb7ed49520b 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
@@ -58,6 +58,7 @@ import net.minecraft.world.level.GameType;
 import net.minecraft.world.level.block.entity.SkullBlockEntity;
 import net.minecraft.world.level.storage.LevelStorageSource;
 import net.minecraft.world.level.storage.WorldData;
+import net.pascalpex.deepslatemc.commands.DeepslateCommand;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.apache.logging.log4j.Level;
@@ -254,6 +255,8 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
         wtf.etil.mirai.MiraiConfig.registerCommands();
         // Mirai end
 
+        MinecraftServer.getServer().server.getCommandMap().register("deepslate", "Deepslate", new DeepslateCommand("deepslate"));
+
         this.setPvpAllowed(dedicatedserverproperties.pvp);
         this.setFlightAllowed(dedicatedserverproperties.allowFlight);
         this.setResourcePack(dedicatedserverproperties.resourcePack, this.getPackHash());
diff --git a/src/main/java/net/pascalpex/deepslatemc/commands/DeepslateCommand.java b/src/main/java/net/pascalpex/deepslatemc/commands/DeepslateCommand.java
new file mode 100644
index 0000000000000000000000000000000000000000..9d6a418ef1a3c467150f13636c69c585b994c852
--- /dev/null
+++ b/src/main/java/net/pascalpex/deepslatemc/commands/DeepslateCommand.java
@@ -0,0 +1,64 @@
+package net.pascalpex.deepslatemc.commands;
+    
+import net.pascalpex.deepslatemc.Config;
+import org.bukkit.ChatColor;
+import org.bukkit.Location;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+    
+public class DeepslateCommand extends Command {
+
+    public DeepslateCommand(String name) {
+        super(name);
+        this.description = "Deepslate Befehl";
+        this.usageMessage = "/deepslate [reload | version]";
+        this.setPermission("deepslate.command");
+    }
+
+    @Override
+    public List<String> tabComplete(CommandSender sender, String alias, String[] args, Location location) throws IllegalArgumentException {
+        if (args.length == 1) {
+            return Stream.of("reload", "version")
+                .filter(arg -> arg.startsWith(args[0].toLowerCase()))
+                .collect(Collectors.toList());
+        }
+        return Collections.emptyList();
+    }
+
+    @Override
+    public boolean execute(CommandSender sender, String commandLabel, String[] args) {
+        String prefix = Config.getPrefix() + " " + ChatColor.DARK_AQUA;
+        Player player = (Player) sender;
+        if(player.hasPermission("deepslate.command")) {
+            if(args.length == 0) {
+                player.sendMessage(prefix + "Nutze /deepslate [reload | version]");
+            }
+            if(args.length == 1) {
+                if(args[0].equalsIgnoreCase("reload")) {
+                    Config.load();
+                    player.sendMessage(prefix + "Config wurde neugeladen");
+                } else {
+                    if(args[0].equalsIgnoreCase("version")) {
+                        player.sendMessage(prefix + "Dieser Server läuft mit DeepslateMC Version 1.0");
+                    } else {
+                        player.sendMessage(prefix + "Nutze /deepslate [reload | version]");
+                    }
+                }
+            }
+            if(args.length >= 2) {
+                player.sendMessage(prefix + "Nutze /deepslate [reload | version]");
+            }
+        } else {
+            player.sendMessage(prefix + "Dafür hast du keine Rechte");
+        }
+        return true;
+    }
+
+}
