From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Mon, 21 Mar 2022 14:34:12 +0100
Subject: [PATCH] Config and messages file


diff --git a/src/main/java/net/minecraft/commands/Commands.java b/src/main/java/net/minecraft/commands/Commands.java
index 7c30d8d61a8e82d8a1f61451090e9bbbca933d2a..e254b9a190056b3fb81ff3679cdf733b37a52f85 100644
--- a/src/main/java/net/minecraft/commands/Commands.java
+++ b/src/main/java/net/minecraft/commands/Commands.java
@@ -106,6 +106,8 @@ import net.minecraft.server.commands.WorldBorderCommand;
 import net.minecraft.server.commands.data.DataCommands;
 import net.minecraft.server.level.ServerPlayer;
 import net.minecraft.util.profiling.jfr.JvmProfiler;
+import net.pascalpex.deepslatemc.Config;
+import net.pascalpex.deepslatemc.MessagesFile;
 import org.slf4j.Logger;
 
 // CraftBukkit start
@@ -128,6 +130,10 @@ public class Commands {
 
     public Commands(Commands.CommandSelection environment) {
         this(); // CraftBukkit
+        // Deepslate start
+        Config.load();
+        MessagesFile.load();
+        // Deepslate end
         AdvancementCommands.register(this.dispatcher);
         AttributeCommand.register(this.dispatcher);
         ExecuteCommand.register(this.dispatcher);
diff --git a/src/main/java/net/pascalpex/deepslatemc/Config.java b/src/main/java/net/pascalpex/deepslatemc/Config.java
new file mode 100644
index 0000000000000000000000000000000000000000..994b992d2acb48576af874137df074bb7c659c88
--- /dev/null
+++ b/src/main/java/net/pascalpex/deepslatemc/Config.java
@@ -0,0 +1,42 @@
+package net.pascalpex.deepslatemc;
+    
+import org.bukkit.configuration.InvalidConfigurationException;
+import org.bukkit.configuration.file.FileConfiguration;
+import org.bukkit.configuration.file.YamlConfiguration;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Config {
+
+    public static File configFile = new File("deepslate", "deepslate.yml");
+    public static FileConfiguration config = YamlConfiguration.loadConfiguration(configFile);
+
+        public static void load() {
+            try {
+                configFile.getParentFile().mkdirs();
+                if (!configFile.exists()) {
+                    configFile.createNewFile();
+                    List<String> header = new ArrayList<>();
+                    header.add("This is the main configuration file of DeepslateMC");
+                    header.add("You can find an explanation of all options here: https://github.com/Pascalpex/DeepslateMC/wiki");
+                    config.options().setHeader(header);
+                    save();
+                }
+                config.load(configFile);
+            } catch (IOException | InvalidConfigurationException e) {
+                e.printStackTrace();
+            }
+        }
+
+        public static void save() {
+            try {
+                config.save(configFile);
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+
+}
diff --git a/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java b/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
new file mode 100644
index 0000000000000000000000000000000000000000..acf071fc155ba214c00d125b8a52d101c92e6a45
--- /dev/null
+++ b/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
@@ -0,0 +1,51 @@
+package net.pascalpex.deepslatemc;
+    
+import org.bukkit.configuration.InvalidConfigurationException;
+import org.bukkit.configuration.file.FileConfiguration;
+import org.bukkit.configuration.file.YamlConfiguration;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MessagesFile {
+
+    public static File configFile = new File("deepslate", "messages.yml");
+    public static FileConfiguration config = YamlConfiguration.loadConfiguration(configFile);
+
+        public static void load() {
+            try {
+                configFile.getParentFile().mkdirs();
+                if (!configFile.exists()) {
+                    configFile.createNewFile();
+                    List<String> header = new ArrayList<>();
+                    header.add("This is the messages file of DeepslateMC");
+                    header.add("You can find some pre-translated files here: https://pascalpex.ddns.net/files/deepslate/translations/");
+                    config.options().setHeader(header);
+                    save();
+                }
+                config.load(configFile);
+                if(!config.contains("prefix")) {
+                    config.set("prefix", "&0[&8DeepslateMC&0]");
+                }
+                save();
+            } catch (IOException | InvalidConfigurationException e) {
+                e.printStackTrace();
+            }
+        }
+
+        public static void save() {
+            try {
+                config.save(configFile);
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+
+        public static String getPrefix() {
+            String prefix = config.getString("prefix");
+            return prefix.replace("&", "§");
+        }
+
+}
