From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Thu, 14 Jul 2022 14:17:46 +0200
Subject: [PATCH] Config and messages file


diff --git a/src/main/java/net/minecraft/commands/Commands.java b/src/main/java/net/minecraft/commands/Commands.java
index 139a57b38eac74887c950041e890e1613d8d3073..af029550650080a4f1c2787759781780ee8eb2dc 100644
--- a/src/main/java/net/minecraft/commands/Commands.java
+++ b/src/main/java/net/minecraft/commands/Commands.java
@@ -114,6 +114,8 @@ import net.minecraft.server.commands.data.DataCommands;
 import net.minecraft.server.level.ServerPlayer;
 import net.minecraft.tags.TagKey;
 import net.minecraft.util.profiling.jfr.JvmProfiler;
+import net.pascalpex.deepslatemc.Config;
+import net.pascalpex.deepslatemc.MessagesFile;
 import org.slf4j.Logger;
 
 // CraftBukkit start
@@ -137,6 +139,10 @@ public class Commands {
 
     public Commands(Commands.CommandSelection environment, CommandBuildContext commandRegistryAccess) {
         this(); // CraftBukkit
+        // Deepslate start
+        Config.load();
+        MessagesFile.load();
+        // Deepslate end
         AdvancementCommands.register(this.dispatcher);
         AttributeCommand.register(this.dispatcher, commandRegistryAccess);
         ExecuteCommand.register(this.dispatcher, commandRegistryAccess);
diff --git a/src/main/java/net/pascalpex/deepslatemc/Config.java b/src/main/java/net/pascalpex/deepslatemc/Config.java
new file mode 100644
index 0000000000000000000000000000000000000000..9a03cd8f8fc82450fdf7b46a405ea6b6482524d2
--- /dev/null
+++ b/src/main/java/net/pascalpex/deepslatemc/Config.java
@@ -0,0 +1,43 @@
+package net.pascalpex.deepslatemc;
+    
+import org.bukkit.configuration.InvalidConfigurationException;
+import org.bukkit.configuration.file.FileConfiguration;
+import org.bukkit.configuration.file.YamlConfiguration;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Config {
+
+    public static File configFile = new File("deepslate", "deepslate.yml");
+    public static FileConfiguration config = YamlConfiguration.loadConfiguration(configFile);
+
+    public static void load() {
+        try {
+            configFile.getParentFile().mkdirs();
+            if (!configFile.exists()) {
+                configFile.createNewFile();
+                List<String> header = new ArrayList<>();
+                header.add("This is the main configuration file of DeepslateMC");
+                header.add("You can find an explanation of all options here: https://github.com/Pascalpex/DeepslateMC/wiki");
+                header.add("You can get support on our Discord server: https://discord.gg/BGrhNnVczp");
+                config.options().setHeader(header);
+                save();
+            }
+            config.load(configFile);
+        } catch (IOException | InvalidConfigurationException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public static void save() {
+        try {
+            config.save(configFile);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+}
diff --git a/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java b/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
new file mode 100644
index 0000000000000000000000000000000000000000..28901cafd9e58b5a111ef665019cec875318d297
--- /dev/null
+++ b/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
@@ -0,0 +1,51 @@
+package net.pascalpex.deepslatemc;
+    
+import org.bukkit.configuration.InvalidConfigurationException;
+import org.bukkit.configuration.file.FileConfiguration;
+import org.bukkit.configuration.file.YamlConfiguration;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MessagesFile {
+
+    public static File configFile = new File("deepslate", "messages.yml");
+    public static FileConfiguration config = YamlConfiguration.loadConfiguration(configFile);
+
+    public static void load() {
+        try {
+            configFile.getParentFile().mkdirs();
+            if (!configFile.exists()) {
+                configFile.createNewFile();
+                List<String> header = new ArrayList<>();
+                header.add("This is the messages file of DeepslateMC");
+                header.add("You can find some pre-translated files here: https://pascalpex.ddns.net/files/deepslate/translations/");
+                config.options().setHeader(header);
+                save();
+            }
+            config.load(configFile);
+            if(!config.contains("prefix")) {
+                config.set("prefix", "&0[&8DeepslateMC&0]");
+            }
+            save();
+        } catch (IOException | InvalidConfigurationException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public static void save() {
+        try {
+            config.save(configFile);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public static String getPrefix() {
+        String prefix = config.getString("prefix");
+        return prefix.replace("&", "§");
+    }
+
+}
