From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Sat, 3 Dec 2022 13:06:36 +0100
Subject: [PATCH] Deepslate command


diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
index b9cc843ddd245d2260b440e1c9143a9fd7b161cc..4aca871f3557a2ad08a8fb3bb55aadbfecb9d696 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
@@ -50,6 +50,7 @@ import net.minecraft.world.level.GameRules;
 import net.minecraft.world.level.GameType;
 import net.minecraft.world.level.block.entity.SkullBlockEntity;
 import net.minecraft.world.level.storage.LevelStorageSource;
+import net.pascalpex.deepslatemc.commands.*;
 import org.slf4j.Logger;
 
 // CraftBukkit start
@@ -237,6 +238,11 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
 
         dev.etil.mirai.MiraiConfig.load(); // Mirai
         dev.etil.mirai.MiraiCommand.init(); // Mirai
+
+        // Deepslate start
+        MinecraftServer.getServer().server.getCommandMap().register("deepslate", "Deepslate", new DeepslateCommand("deepslate"));
+        // Deepslate end
+
         this.setPvpAllowed(dedicatedserverproperties.pvp);
         this.setFlightAllowed(dedicatedserverproperties.allowFlight);
         this.setMotd(dedicatedserverproperties.motd);
diff --git a/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java b/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
index 28901cafd9e58b5a111ef665019cec875318d297..96833e6e5dc97d13e0ea778530b0606937f14ae4 100644
--- a/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
+++ b/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
@@ -29,6 +29,12 @@ public class MessagesFile {
             if(!config.contains("prefix")) {
                 config.set("prefix", "&0[&8DeepslateMC&0]");
             }
+            if(!config.contains("wrongSyntax")) {
+                config.set("wrongSyntax", "&cWrong usage, use: &b%usage%");
+            }
+            if(!config.contains("configReloaded")) {
+                config.set("configReloaded", "&aConfig got reloaded successfully");
+            }
             save();
         } catch (IOException | InvalidConfigurationException e) {
             e.printStackTrace();
@@ -47,5 +53,13 @@ public class MessagesFile {
         String prefix = config.getString("prefix");
         return prefix.replace("&", "§");
     }
+    public static String getWrongSyntax() {
+        String prefix = config.getString("wrongSyntax");
+        return prefix.replace("&", "§");
+    }
+    public static String getConfigReloaded() {
+        String prefix = config.getString("configReloaded");
+        return prefix.replace("&", "§");
+    }
 
 }
diff --git a/src/main/java/net/pascalpex/deepslatemc/commands/DeepslateCommand.java b/src/main/java/net/pascalpex/deepslatemc/commands/DeepslateCommand.java
new file mode 100644
index 0000000000000000000000000000000000000000..c01afb333b667c92275028ecf9e2c77a9e2f96b1
--- /dev/null
+++ b/src/main/java/net/pascalpex/deepslatemc/commands/DeepslateCommand.java
@@ -0,0 +1,90 @@
+package net.pascalpex.deepslatemc.commands;
+    
+import net.pascalpex.deepslatemc.Config;
+import net.pascalpex.deepslatemc.MessagesFile;
+import org.bukkit.ChatColor;
+import org.bukkit.Location;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+    
+public class DeepslateCommand extends Command {
+
+    public DeepslateCommand(String name) {
+        super(name);
+        this.description = "Deepslate command";
+        this.usageMessage = "/deepslate [reload | version]";
+    }
+
+    @Override
+    public List<String> tabComplete(CommandSender sender, String alias, String[] args, Location location) throws IllegalArgumentException {
+        if (args.length == 1) {
+            return Stream.of("reload", "version")
+                .filter(arg -> arg.startsWith(args[0].toLowerCase()))
+                .collect(Collectors.toList());
+        }
+        return Collections.emptyList();
+    }
+
+    @Override
+    public boolean execute(CommandSender sender, String commandLabel, String[] args) {
+        if (commandLabel.equalsIgnoreCase("deepslate")) {
+            String prefix = MessagesFile.getPrefix() + " ";
+            if (sender instanceof Player) {
+                Player player = (Player) sender;
+                if(player.hasPermission("deepslate.command")) {
+                    if(args.length == 0) {
+                        player.sendMessage(prefix + MessagesFile.getWrongSyntax().replace("%usage%", "/deepslate [reload | version]"));
+                    }
+                    if(args.length == 1) {
+                        if(args[0].equalsIgnoreCase("reload")) {
+                            MessagesFile.load();
+                            Config.load();
+                            player.sendMessage(prefix + MessagesFile.getConfigReloaded());
+                        } else {
+                            if(args[0].equalsIgnoreCase("version")) {
+                                String deepslateVersion = this.getClass().getPackage().getImplementationVersion().replace('"', ' ').replace(" ", "");
+                                player.sendMessage(prefix + ChatColor.GOLD + "This server is running DeepslateMC from Pascalpex: " + deepslateVersion);
+                            } else {
+                                player.sendMessage(prefix + MessagesFile.getWrongSyntax().replace("%usage%", "/deepslate [reload | version]"));
+                            }
+                        }
+                    }
+                    if(args.length >= 2) {
+                        player.sendMessage(prefix + MessagesFile.getWrongSyntax().replace("%usage%", "/deepslate [reload | version]"));
+                    }
+                } else {
+                    player.sendMessage(prefix + ChatColor.GOLD + "This server is running DeepslateMC from Pascalpex");
+                }
+            } else {
+                if(args.length == 0) {
+                    sender.sendMessage(prefix + MessagesFile.getWrongSyntax().replace("%usage%", "/deepslate [reload | version]"));
+                }
+                if(args.length == 1) {
+                    if(args[0].equalsIgnoreCase("reload")) {
+                        MessagesFile.load();
+                        Config.load();
+                        sender.sendMessage(prefix + MessagesFile.getConfigReloaded());
+                    } else {
+                        if(args[0].equalsIgnoreCase("version")) {
+                            String deepslateVersion = this.getClass().getPackage().getImplementationVersion().replace('"', ' ').replace(" ", "");
+                            sender.sendMessage(prefix + ChatColor.GOLD + "This server is running DeepslateMC from Pascalpex: " + deepslateVersion);
+                        } else {
+                            sender.sendMessage(prefix + MessagesFile.getWrongSyntax().replace("%usage%", "/deepslate [reload | version]"));
+                        }
+                    }
+                }
+                if(args.length >= 2) {
+                    sender.sendMessage(prefix + MessagesFile.getWrongSyntax().replace("%usage%", "/deepslate [reload | version]"));
+                }
+            }
+        }
+        return true;
+    }
+
+}
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index a6b3b08e2f6f2b51a66d8c870522fbc9463fa9b5..2ca6f04a1a4d85ab85e4cdb8365460b8348db542 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -119,6 +119,7 @@ import net.minecraft.world.level.storage.WorldData;
 import net.minecraft.world.level.storage.loot.LootDataManager;
 import net.minecraft.world.level.validation.ContentValidationException;
 import net.minecraft.world.phys.Vec3;
+import net.pascalpex.deepslatemc.commands.*;
 import org.bukkit.BanList;
 import org.bukkit.Bukkit;
 import org.bukkit.ChatColor;
@@ -1089,6 +1090,11 @@ public final class CraftServer implements Server {
         org.spigotmc.SpigotConfig.registerCommands(); // Spigot
         io.papermc.paper.command.PaperCommands.registerCommands(this.console); // Paper
         org.purpurmc.purpur.PurpurConfig.registerCommands(); // Purpur
+
+        // Deepslate start
+        MinecraftServer.getServer().server.getCommandMap().register("deepslate", "Deepslate", new DeepslateCommand("deepslate"));
+        // Deepslate end
+
         this.overrideAllCommandBlockCommands = this.commandsConfiguration.getStringList("command-block-overrides").contains("*");
         this.ignoreVanillaPermissions = this.commandsConfiguration.getBoolean("ignore-vanilla-permissions");
 
