From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Tue, 15 Feb 2022 13:20:52 +0100
Subject: [PATCH] Deepslate command


diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
index 8bb7e1d2747eb132dfe051af831523c887dd2d5b..b51aaef4c7d4639ba3d5d9ed9d8957f5349e4113 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
@@ -60,6 +60,7 @@ import net.minecraft.world.level.storage.LevelStorageSource;
 import net.minecraft.world.level.storage.WorldData;
 import net.pascalpex.deepslatemc.Config;
 import net.pascalpex.deepslatemc.MessagesFile;
+import net.pascalpex.deepslatemc.commands.DeepslateCommand;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.apache.logging.log4j.Level;
@@ -259,6 +260,8 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
         // Deepslate start
         MessagesFile.load();
         Config.load();
+
+        MinecraftServer.getServer().server.getCommandMap().register("deepslate", "Deepslate", new DeepslateCommand("deepslate"));
         // Deepslate end
 
         this.setPvpAllowed(dedicatedserverproperties.pvp);
diff --git a/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java b/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
index b6f529b7e5adf0b497d5c8d783e70fec072e456c..f59866b9e16774cb5efd995ddb4276c66a940542 100644
--- a/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
+++ b/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
@@ -23,6 +23,12 @@ public class MessagesFile {
             if(!config.contains("prefix")) {
                 config.set("prefix", "&0[&8DeepslateMC&0]");
             }
+            if(!config.contains("wrongSyntax")) {
+                config.set("wrongSyntax", "&cWrong usage, use: &b%usage%");
+            }
+            if(!config.contains("configReloaded")) {
+                config.set("configReloaded", "&aConfig got reloaded successfully");
+            }
             save();
         } catch (IOException | InvalidConfigurationException e) {
             e.printStackTrace();
@@ -42,4 +48,14 @@ public class MessagesFile {
         return prefix.replace("&", "§");
     }
 
+    public static String getWrongSyntax() {
+        String prefix = config.getString("wrongSyntax");
+        return prefix.replace("&", "§");
+    }
+
+    public static String getConfigReloaded() {
+        String prefix = config.getString("configReloaded");
+        return prefix.replace("&", "§");
+    }
+
 }
diff --git a/src/main/java/net/pascalpex/deepslatemc/commands/DeepslateCommand.java b/src/main/java/net/pascalpex/deepslatemc/commands/DeepslateCommand.java
new file mode 100644
index 0000000000000000000000000000000000000000..3a25ba35539baeac4b96c2fad6319e524055bd8d
--- /dev/null
+++ b/src/main/java/net/pascalpex/deepslatemc/commands/DeepslateCommand.java
@@ -0,0 +1,93 @@
+package net.pascalpex.deepslatemc.commands;
+    
+import net.pascalpex.deepslatemc.Config;
+import net.pascalpex.deepslatemc.MessagesFile;
+import org.bukkit.ChatColor;
+import org.bukkit.Location;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+    
+public class DeepslateCommand extends Command {
+
+    public DeepslateCommand(String name) {
+        super(name);
+        this.description = "Deepslate Command";
+        this.usageMessage = "/deepslate [reload | version]";
+        this.setPermission("deepslate.command");
+    }
+
+    @Override
+    public List<String> tabComplete(CommandSender sender, String alias, String[] args, Location location) throws IllegalArgumentException {
+        if (args.length == 1) {
+            return Stream.of("reload", "version")
+                .filter(arg -> arg.startsWith(args[0].toLowerCase()))
+                .collect(Collectors.toList());
+        }
+        return Collections.emptyList();
+    }
+
+    @Override
+    public boolean execute(CommandSender sender, String commandLabel, String[] args) {
+        if (commandLabel.equalsIgnoreCase("deepslate")) {
+            String prefix = MessagesFile.getPrefix() + " ";
+            if (sender instanceof Player) {
+                Player player = (Player) sender;
+                if(player.hasPermission("deepslate.command")) {
+                    if(args.length == 0) {
+                        player.sendMessage(prefix + MessagesFile.getWrongSyntax().replace("%usage%", "/deepslate [reload | version]"));
+                    }
+                    if(args.length == 1) {
+                        if(args[0].equalsIgnoreCase("reload")) {
+                            MessagesFile.load();
+                            Config.load();
+                            player.sendMessage(prefix + MessagesFile.getConfigReloaded());
+                        } else {
+                            if(args[0].equalsIgnoreCase("version")) {
+                                String deepslateVersion = this.getClass().getPackage().getImplementationVersion().replace('"', ' ').replace(" ", "");
+                                player.sendMessage(prefix + ChatColor.GOLD + "This server is running DeepslateMC from Pascalpex: " + deepslateVersion);
+                            } else {
+                                player.sendMessage(prefix + MessagesFile.getWrongSyntax().replace("%usage%", "/deepslate [reload | version]"));
+                            }
+                        }
+                    }
+                    if(args.length >= 2) {
+                        player.sendMessage(prefix + MessagesFile.getWrongSyntax().replace("%usage%", "/deepslate [reload | version]"));
+                    }
+                } else {
+                    player.sendMessage(prefix + ChatColor.GOLD + "This server is running DeepslateMC from Pascalpex");
+                }
+            } else {
+                if(args.length == 0) {
+                    sender.sendMessage(prefix + MessagesFile.getWrongSyntax().replace("%usage%", "/deepslate [reload | version]"));
+                }
+                if(args.length == 1) {
+                    if(args[0].equalsIgnoreCase("reload")) {
+                        MessagesFile.load();
+                        Config.load();
+                        sender.sendMessage(prefix + MessagesFile.getConfigReloaded());
+                    } else {
+                        if(args[0].equalsIgnoreCase("version")) {
+                            String deepslateVersion = this.getClass().getPackage().getImplementationVersion().replace('"', ' ').replace(" ", "");
+                            sender.sendMessage(prefix + ChatColor.GOLD + "This server is running DeepslateMC from Pascalpex: " + deepslateVersion);
+                        } else {
+                            sender.sendMessage(prefix + MessagesFile.getWrongSyntax().replace("%usage%", "/deepslate [reload | version]"));
+                        }
+                    }
+                }
+                if(args.length >= 2) {
+                    sender.sendMessage(prefix + MessagesFile.getWrongSyntax().replace("%usage%", "/deepslate [reload | version]"));
+                }
+            }
+                        
+        }
+        return true;
+    }
+
+}
