From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Sun, 23 Jan 2022 18:46:32 +0100
Subject: [PATCH] Deepslate Config


diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
index a8d32dbcfab10622c77a7030feae96fdb985e679..c8aa8d8b16c8571baa82303a68041d7b9d1e2083 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
@@ -58,6 +58,7 @@ import net.minecraft.world.level.GameType;
 import net.minecraft.world.level.block.entity.SkullBlockEntity;
 import net.minecraft.world.level.storage.LevelStorageSource;
 import net.minecraft.world.level.storage.WorldData;
+import net.pascalpex.deepslatemc.Config;
 import net.pascalpex.deepslatemc.commands.DeepslateCommand;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
@@ -245,6 +246,7 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
         gg.pufferfish.pufferfish.PufferfishConfig.load(); // Pufferfish
         gg.pufferfish.pufferfish.PufferfishCommand.init(); // Pufferfish
 
+        Config.load();
         MinecraftServer.getServer().server.getCommandMap().register("deepslate", "Deepslate", new DeepslateCommand("deepslate"));
 
         // Yatopia start
diff --git a/src/main/java/net/pascalpex/deepslatemc/Config.java b/src/main/java/net/pascalpex/deepslatemc/Config.java
new file mode 100644
index 0000000000000000000000000000000000000000..99122d5f56c40ee12d3d01b90d052de98eff2668
--- /dev/null
+++ b/src/main/java/net/pascalpex/deepslatemc/Config.java
@@ -0,0 +1,44 @@
+package net.pascalpex.deepslatemc;
+    
+import org.bukkit.ChatColor;
+import org.bukkit.configuration.InvalidConfigurationException;
+import org.bukkit.configuration.file.FileConfiguration;
+import org.bukkit.configuration.file.YamlConfiguration;
+import org.bukkit.inventory.ItemStack;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+
+public class Config {
+
+    public static File configFile = new File("deepslate", "deepslate.yml");
+    public static FileConfiguration config = YamlConfiguration.loadConfiguration(configFile);
+
+    public static void load() {
+        try {
+            configFile.getParentFile().mkdirs();
+            if (!configFile.exists()) {
+                configFile.createNewFile();
+                config.set("prefix", ChatColor.BLACK + "[" + ChatColor.DARK_GRAY + "DeepslateMC" + ChatColor.BLACK + "]");
+                save();
+            }
+            config.load(configFile);
+        } catch (IOException | InvalidConfigurationException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public static void save() {
+        try {
+            config.save(configFile);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public static String getPrefix() {
+        return config.getString("prefix");
+    }
+
+}
