From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Sun, 23 Jan 2022 19:35:46 +0100
Subject: [PATCH] Discord Command


diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
index c8aa8d8b16c8571baa82303a68041d7b9d1e2083..8ec849fe9b4736ead7ca925c868eaa3d9bf15c4b 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
@@ -60,6 +60,7 @@ import net.minecraft.world.level.storage.LevelStorageSource;
 import net.minecraft.world.level.storage.WorldData;
 import net.pascalpex.deepslatemc.Config;
 import net.pascalpex.deepslatemc.commands.DeepslateCommand;
+import net.pascalpex.deepslatemc.commands.DiscordCommand;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.apache.logging.log4j.Level;
@@ -248,6 +249,8 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
 
         Config.load();
         MinecraftServer.getServer().server.getCommandMap().register("deepslate", "Deepslate", new DeepslateCommand("deepslate"));
+        MinecraftServer.getServer().server.getCommandMap().register("discord", "Deepslate", new DiscordCommand("discord"));
+        MinecraftServer.getServer().server.getCommandMap().register("dc", "Deepslate", new DiscordCommand("dc"));
 
         // Yatopia start
         try {
diff --git a/src/main/java/net/pascalpex/deepslatemc/Config.java b/src/main/java/net/pascalpex/deepslatemc/Config.java
index f557502a700159620fd8ba53d0498f2d9cfce948..e0136508752affde08f1c785192a3cb3e45ebe9d 100644
--- a/src/main/java/net/pascalpex/deepslatemc/Config.java
+++ b/src/main/java/net/pascalpex/deepslatemc/Config.java
@@ -21,6 +21,7 @@ public class Config {
             if (!configFile.exists()) {
                 configFile.createNewFile();
                 config.set("prefix", ChatColor.BLACK + "[" + ChatColor.DARK_GRAY + "DeepslateMC" + ChatColor.BLACK + "]");
+                config.set("discordLink", "https://google.de/");
                 save();
             }
             config.load(configFile);
@@ -41,4 +42,8 @@ public class Config {
         return config.getString("prefix");
     }
 
+    public static String getDiscordLink() {
+        return config.getString("discordLink");
+    }
+
 }
diff --git a/src/main/java/net/pascalpex/deepslatemc/commands/DeepslateCommand.java b/src/main/java/net/pascalpex/deepslatemc/commands/DeepslateCommand.java
index 7985449b11ae5c246c56a8795480568496b93024..5d383176cc333c3715d9b925384deece3a9c0758 100644
--- a/src/main/java/net/pascalpex/deepslatemc/commands/DeepslateCommand.java
+++ b/src/main/java/net/pascalpex/deepslatemc/commands/DeepslateCommand.java
@@ -34,29 +34,53 @@ public class DeepslateCommand extends Command {
 
     @Override
     public boolean execute(CommandSender sender, String commandLabel, String[] args) {
-        String prefix = Config.getPrefix() + " " + ChatColor.DARK_AQUA;
-        Player player = (Player) sender;
-        if(player.hasPermission("deepslate.command")) {
-            if(args.length == 0) {
-                player.sendMessage(prefix + "Nutze /deepslate [reload | version]");
-            }
-            if(args.length == 1) {
-                if(args[0].equalsIgnoreCase("reload")) {
-                    Config.load();
-                    player.sendMessage(prefix + "Config wurde neugeladen");
+        if (commandLabel.equalsIgnoreCase("deepslate")) {
+            String prefix = Config.getPrefix() + " " + ChatColor.DARK_AQUA;
+            if (sender instanceof Player) {
+                Player player = (Player) sender;
+                if(player.hasPermission("deepslate.command")) {
+                    if(args.length == 0) {
+                        player.sendMessage(prefix + "Nutze /deepslate [reload | version]");
+                    }
+                    if(args.length == 1) {
+                        if(args[0].equalsIgnoreCase("reload")) {
+                            Config.load();
+                            player.sendMessage(prefix + "Config wurde neugeladen");
+                        } else {
+                            if(args[0].equalsIgnoreCase("version")) {
+                                player.sendMessage(prefix + "Dieser Server läuft mit DeepslateMC von Pascalpex");
+                            } else {
+                                player.sendMessage(prefix + "Nutze /deepslate [reload | version]");
+                            }
+                        }
+                    }
+                    if(args.length >= 2) {
+                        player.sendMessage(prefix + "Nutze /deepslate [reload | version]");
+                    }
                 } else {
-                    if(args[0].equalsIgnoreCase("version")) {
-                        player.sendMessage(prefix + "Dieser Server läuft mit DeepslateMC Version 1.0");
+                    player.sendMessage(prefix + "Dieser Server läuft mit DeepslateMC Version 1.0");
+                }
+            } else {
+                if(args.length == 0) {
+                    sender.sendMessage(prefix + "Nutze /deepslate [reload | version]");
+                }
+                if(args.length == 1) {
+                    if(args[0].equalsIgnoreCase("reload")) {
+                        Config.load();
+                        sender.sendMessage(prefix + "Config wurde neugeladen");
                     } else {
-                        player.sendMessage(prefix + "Nutze /deepslate [reload | version]");
+                        if(args[0].equalsIgnoreCase("version")) {
+                            sender.sendMessage(prefix + "Dieser Server läuft mit DeepslateMC von Pascalpex");
+                        } else {
+                            sender.sendMessage(prefix + "Nutze /deepslate [reload | version]");
+                        }
                     }
                 }
+                if(args.length >= 2) {
+                    sender.sendMessage(prefix + "Nutze /deepslate [reload | version]");
+                }
+
             }
-            if(args.length >= 2) {
-                player.sendMessage(prefix + "Nutze /deepslate [reload | version]");
-            }
-        } else {
-            player.sendMessage(prefix + "Dafür hast du keine Rechte");
         }
         return true;
     }
diff --git a/src/main/java/net/pascalpex/deepslatemc/commands/DiscordCommand.java b/src/main/java/net/pascalpex/deepslatemc/commands/DiscordCommand.java
new file mode 100644
index 0000000000000000000000000000000000000000..e8b90d4b5e05ddb467189839899fe36b407433bc
--- /dev/null
+++ b/src/main/java/net/pascalpex/deepslatemc/commands/DiscordCommand.java
@@ -0,0 +1,33 @@
+package net.pascalpex.deepslatemc.commands;
+    
+import net.pascalpex.deepslatemc.Config;
+import org.bukkit.ChatColor;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+import org.jetbrains.annotations.NotNull;
+
+public class DiscordCommand extends Command {
+
+    public DiscordCommand(String name) {
+        super(name);
+        this.description = "Discord Link Befehl";
+        this.usageMessage = "/discord";
+        this.setPermission("deepslate.discord");
+    }
+
+    @Override
+    public boolean execute(@NotNull CommandSender sender, @NotNull String commandLabel, @NotNull String[] args) {
+        if (commandLabel.equalsIgnoreCase("dc") || commandLabel.equalsIgnoreCase("discord")) {
+            if (sender instanceof Player) {
+                Player player = (Player) sender;
+                String link = Config.getDiscordLink();
+                player.sendMessage(ChatColor.DARK_GRAY + "[" + ChatColor.GREEN + "Pascalpex" + ChatColor.DARK_GRAY + "]" + ChatColor.GOLD + " Der Link zu unserem Discord:");
+                player.sendMessage(ChatColor.GOLD + link);
+            } else {
+                sender.sendMessage("Dieser Befehl ist nur für Spieler geeignet");
+            }
+        }
+        return true;
+    }
+}
