From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Sat, 3 Dec 2022 13:13:46 +0100
Subject: [PATCH] Discord command


diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
index 8a423e94473433a28789413b62266a1e5e70e5d1..22eddf655ed27ce87f2e0e5346f3686cf5db33e6 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
@@ -240,6 +240,8 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
 
         // Deepslate start
         MinecraftServer.getServer().server.getCommandMap().register("deepslate", "Deepslate", new DeepslateCommand("deepslate"));
+        MinecraftServer.getServer().server.getCommandMap().register("discord", "Deepslate", new DiscordCommand("discord"));
+        MinecraftServer.getServer().server.getCommandMap().register("dc", "Deepslate", new DiscordCommand("dc"));
         // Deepslate end
 
         this.setPvpAllowed(dedicatedserverproperties.pvp);
diff --git a/src/main/java/net/pascalpex/deepslatemc/Config.java b/src/main/java/net/pascalpex/deepslatemc/Config.java
index 9a03cd8f8fc82450fdf7b46a405ea6b6482524d2..b0ed05f69d66a863042767768fc8709221abe232 100644
--- a/src/main/java/net/pascalpex/deepslatemc/Config.java
+++ b/src/main/java/net/pascalpex/deepslatemc/Config.java
@@ -27,6 +27,10 @@ public class Config {
                 save();
             }
             config.load(configFile);
+            if(!config.contains("discordLink")) {
+                config.set("discordLink", "https://discord.gg/BGrhNnVczp");
+            }
+            save();
         } catch (IOException | InvalidConfigurationException e) {
             e.printStackTrace();
         }
@@ -40,4 +44,8 @@ public class Config {
         }
     }
 
+    public static String getDiscordLink() {
+        return config.getString("discordLink");
+    }
+
 }
diff --git a/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java b/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
index 96833e6e5dc97d13e0ea778530b0606937f14ae4..073ee76cdefd6f0e40f7f1f80312f25e2406118b 100644
--- a/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
+++ b/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
@@ -35,6 +35,12 @@ public class MessagesFile {
             if(!config.contains("configReloaded")) {
                 config.set("configReloaded", "&aConfig got reloaded successfully");
             }
+            if(!config.contains("onlyForPlayers")) {
+                config.set("onlyForPlayers", "&cThis command can only be used by players");
+            }
+            if(!config.contains("discordMessage")) {
+                config.set("discordMessage", "&6The link to our Discord server: &b%link%");
+            }
             save();
         } catch (IOException | InvalidConfigurationException e) {
             e.printStackTrace();
@@ -61,5 +67,13 @@ public class MessagesFile {
         String prefix = config.getString("configReloaded");
         return prefix.replace("&", "§");
     }
+    public static String getOnlyForPlayers() {
+        String prefix = config.getString("onlyForPlayers");
+        return prefix.replace("&", "§");
+    }
+    public static String getDiscordMessage() {
+        String prefix = config.getString("discordMessage");
+        return prefix.replace("&", "§");
+    }
 
 }
diff --git a/src/main/java/net/pascalpex/deepslatemc/commands/DiscordCommand.java b/src/main/java/net/pascalpex/deepslatemc/commands/DiscordCommand.java
new file mode 100644
index 0000000000000000000000000000000000000000..a9beebbafcaf947107e70ba11fa196053fbbdf0f
--- /dev/null
+++ b/src/main/java/net/pascalpex/deepslatemc/commands/DiscordCommand.java
@@ -0,0 +1,26 @@
+package net.pascalpex.deepslatemc.commands;
+    
+import net.pascalpex.deepslatemc.Config;
+import net.pascalpex.deepslatemc.MessagesFile;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+
+public class DiscordCommand extends Command {
+
+    public DiscordCommand(String name) {
+        super(name);
+        this.description = "Discord link command";
+        this.usageMessage = "/discord";
+    }
+
+    @Override
+    public boolean execute(CommandSender sender, String commandLabel, String[] args) {
+        String prefix = MessagesFile.getPrefix() + " ";
+        if (commandLabel.equalsIgnoreCase("dc") || commandLabel.equalsIgnoreCase("discord")) {
+            String link = Config.getDiscordLink();
+            sender.sendMessage(prefix + MessagesFile.getDiscordMessage().replace("%link%", link));
+        }
+        return true;
+    }
+}
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 07be1fd4d4aef033ac0388f47a60ff860741ecb4..6d526f9e3a5b3599623395bebe41dcf3b643bab9 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -1111,6 +1111,8 @@ public final class CraftServer implements Server {
 
         // Deepslate start
         MinecraftServer.getServer().server.getCommandMap().register("deepslate", "Deepslate", new DeepslateCommand("deepslate"));
+        MinecraftServer.getServer().server.getCommandMap().register("discord", "Deepslate", new DiscordCommand("discord"));
+        MinecraftServer.getServer().server.getCommandMap().register("dc", "Deepslate", new DiscordCommand("dc"));
         // Deepslate end
 
         this.overrideAllCommandBlockCommands = this.commandsConfiguration.getStringList("command-block-overrides").contains("*");
