From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Mon, 21 Mar 2022 15:07:19 +0100
Subject: [PATCH] Discord command


diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
index 21c2ae6baf3c74f2e1134630bc8081856dd47016..1215b6b4bb18c7a040dbd27dd92dfc44af6f4163 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
@@ -56,6 +56,7 @@ import net.minecraft.world.level.GameType;
 import net.minecraft.world.level.block.entity.SkullBlockEntity;
 import net.minecraft.world.level.storage.LevelStorageSource;
 import net.pascalpex.deepslatemc.commands.DeepslateCommand;
+import net.pascalpex.deepslatemc.commands.DiscordCommand;
 import org.slf4j.Logger;
 
 // CraftBukkit start
@@ -241,6 +242,8 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
 
         // Deepslate start
         MinecraftServer.getServer().server.getCommandMap().register("deepslate", "Deepslate", new DeepslateCommand("deepslate"));
+        MinecraftServer.getServer().server.getCommandMap().register("discord", "Deepslate", new DiscordCommand("discord"));
+        MinecraftServer.getServer().server.getCommandMap().register("dc", "Deepslate", new DiscordCommand("dc"));
         // Deepslate end
 
         this.setPvpAllowed(dedicatedserverproperties.pvp);
diff --git a/src/main/java/net/pascalpex/deepslatemc/Config.java b/src/main/java/net/pascalpex/deepslatemc/Config.java
index 66389dedb5b0ad82839f5c324d6676591c108096..1a843527d24e8a48fb6e0ad9d8fafa679512c868 100644
--- a/src/main/java/net/pascalpex/deepslatemc/Config.java
+++ b/src/main/java/net/pascalpex/deepslatemc/Config.java
@@ -22,10 +22,15 @@ public class Config {
                 List<String> header = new ArrayList<>();
                 header.add("This is the main configuration file of DeepslateMC");
                 header.add("You can find an explanation of all options here: https://github.com/Pascalpex/DeepslateMC/wiki");
+                header.add("You can get support on our Discord server: https://discord.gg/BGrhNnVczp");
                 config.options().setHeader(header);
                 save();
             }
             config.load(configFile);
+                if(!config.contains("discordLink")) {
+                    config.set("discordLink", "https://discord.gg/BGrhNnVczp");
+                }
+                save();
         } catch (IOException | InvalidConfigurationException e) {
             e.printStackTrace();
         }
@@ -39,4 +44,8 @@ public class Config {
         }
     }
 
+        public static String getDiscordLink() {
+            return config.getString("discordLink");
+        }
+
 }
diff --git a/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java b/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
index cf13be02d96049534d46ab981c1dd78f8eccd0da..bb13ce36183389dab6ba595a63891bfab99d9401 100644
--- a/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
+++ b/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
@@ -35,6 +35,12 @@ public class MessagesFile {
             if(!config.contains("configReloaded")) {
                 config.set("configReloaded", "&aConfig got reloaded successfully");
             }
+                if(!config.contains("onlyForPlayers")) {
+                    config.set("onlyForPlayers", "&cThis command can only be used by players");
+                }
+                if(!config.contains("discordMessage")) {
+                    config.set("discordMessage", "&6The link to our Discord server: &b%link%");
+                }
             save();
         } catch (IOException | InvalidConfigurationException e) {
             e.printStackTrace();
@@ -63,4 +69,14 @@ public class MessagesFile {
         return prefix.replace("&", "§");
     }
 
+        public static String getOnlyForPlayers() {
+            String prefix = config.getString("onlyForPlayers");
+            return prefix.replace("&", "§");
+        }
+
+        public static String getDiscordMessage() {
+            String prefix = config.getString("discordMessage");
+            return prefix.replace("&", "§");
+        }
+
 }
diff --git a/src/main/java/net/pascalpex/deepslatemc/commands/DiscordCommand.java b/src/main/java/net/pascalpex/deepslatemc/commands/DiscordCommand.java
new file mode 100644
index 0000000000000000000000000000000000000000..24717b7f0ead489c38635c7db36e22b88a2bd387
--- /dev/null
+++ b/src/main/java/net/pascalpex/deepslatemc/commands/DiscordCommand.java
@@ -0,0 +1,32 @@
+package net.pascalpex.deepslatemc.commands;
+    
+import net.pascalpex.deepslatemc.Config;
+import net.pascalpex.deepslatemc.MessagesFile;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+import org.jetbrains.annotations.NotNull;
+
+public class DiscordCommand extends Command {
+
+    public DiscordCommand(String name) {
+        super(name);
+        this.description = "Discord link command";
+        this.usageMessage = "/discord";
+    }
+
+    @Override
+    public boolean execute(@NotNull CommandSender sender, @NotNull String commandLabel, @NotNull String[] args) {
+        String prefix = MessagesFile.getPrefix() + " ";
+        if (commandLabel.equalsIgnoreCase("dc") || commandLabel.equalsIgnoreCase("discord")) {
+            if (sender instanceof Player) {
+                Player player = (Player) sender;
+                String link = Config.getDiscordLink();
+                player.sendMessage(prefix + MessagesFile.getDiscordMessage().replace("%link%", link));
+            } else {
+                sender.sendMessage(prefix + MessagesFile.getOnlyForPlayers());
+            }
+        }
+        return true;
+    }
+}
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 9c02913ee18db8994bf2abce1775557efe35dcd8..f47da2b9c03aa37ef1c5a37ddfd4a78cc40f81e1 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -106,6 +106,7 @@ import net.minecraft.world.level.storage.PrimaryLevelData;
 import net.minecraft.world.level.storage.loot.LootTables;
 import net.minecraft.world.phys.Vec3;
 import net.pascalpex.deepslatemc.commands.DeepslateCommand;
+import net.pascalpex.deepslatemc.commands.DiscordCommand;
 import org.apache.commons.lang.Validate;
 import org.bukkit.BanList;
 import org.bukkit.Bukkit;
@@ -1029,6 +1030,8 @@ public final class CraftServer implements Server {
 
         // Deepslate start
         MinecraftServer.getServer().server.getCommandMap().register("deepslate", "Deepslate", new DeepslateCommand("deepslate"));
+        MinecraftServer.getServer().server.getCommandMap().register("discord", "Deepslate", new DiscordCommand("discord"));
+        MinecraftServer.getServer().server.getCommandMap().register("dc", "Deepslate", new DiscordCommand("dc"));
         // Deepslate end
 
         this.overrideAllCommandBlockCommands = this.commandsConfiguration.getStringList("command-block-overrides").contains("*");
