From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: BuildTools <unconfigured@null.spigotmc.org>
Date: Sat, 11 Sep 2021 14:15:41 +0200
Subject: [PATCH] Clearchat-Command


diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
index b30c880ddfc277fbbcbe0ef359abde546ffcb7a1..66fd09b02721075e34d3e85486c3d42db9381d7b 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
@@ -63,6 +63,7 @@ import net.minecraft.world.level.block.entity.SkullBlockEntity;
 import net.minecraft.world.level.storage.LevelStorageSource;
 import net.minecraft.world.level.storage.WorldData;
 import net.pascalpex.deepslatemc.Config;
+import net.pascalpex.deepslatemc.commands.ClearchatCommand;
 import net.pascalpex.deepslatemc.commands.DeepslateCommand;
 import net.pascalpex.deepslatemc.commands.DiscordCommand;
 import org.apache.logging.log4j.LogManager;
@@ -239,10 +240,14 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
         gg.airplane.AirplaneConfig.load(); // Airplane - config
         gg.airplane.commands.AirplaneCommands.init(); // Airplane - command
 
+        // Deepslate start
         Config.load();
         MinecraftServer.getServer().server.getCommandMap().register("deepslate", "Deepslate", new DeepslateCommand("deepslate"));
         MinecraftServer.getServer().server.getCommandMap().register("discord", "Deepslate", new DiscordCommand("discord"));
         MinecraftServer.getServer().server.getCommandMap().register("dc", "Deepslate", new DiscordCommand("dc"));
+        MinecraftServer.getServer().server.getCommandMap().register("cc", "Deepslate", new ClearchatCommand("cc"));
+        MinecraftServer.getServer().server.getCommandMap().register("clearchat", "Deepslate", new ClearchatCommand("clearchat"));
+        // Deepslate end
 
         // Sugarcane start - config
         try {
diff --git a/src/main/java/net/pascalpex/deepslatemc/commands/ClearchatCommand.java b/src/main/java/net/pascalpex/deepslatemc/commands/ClearchatCommand.java
new file mode 100644
index 0000000000000000000000000000000000000000..5eae1e19bcdffa70da7f328140b4dca39b507992
--- /dev/null
+++ b/src/main/java/net/pascalpex/deepslatemc/commands/ClearchatCommand.java
@@ -0,0 +1,33 @@
+package net.pascalpex.deepslatemc.commands;
+
+import org.bukkit.Bukkit;
+import org.bukkit.ChatColor;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+import org.jetbrains.annotations.NotNull;
+
+public class ClearchatCommand extends Command {
+    public ClearchatCommand(@NotNull String name) {
+        super(name);
+        this.description = "Leert den Chat";
+        this.usageMessage = "/clearchat";
+        this.setPermission("deepslate.clearchat");
+    }
+
+    @Override
+    public boolean execute(@NotNull CommandSender sender, @NotNull String commandLabel, @NotNull String[] args) {
+        Player player = (Player) sender;
+        if (commandLabel.equalsIgnoreCase("cc") || commandLabel.equalsIgnoreCase("clearchat")) {
+            if(player.hasPermission("deepslate.clearchat")) {
+                for (int x = 0; x < 150; x++){
+                    Bukkit.broadcastMessage("");
+                }
+                Bukkit.broadcastMessage(ChatColor.YELLOW + "Der Chat wurde von " + ChatColor.RED + player.getName() + ChatColor.YELLOW + " geleert");
+            }
+        } else {
+            player.sendMessage(ChatColor.RED + "Dafür hast du keine Rechte");
+        }
+        return true;
+    }
+}
diff --git a/src/main/java/net/pascalpex/deepslatemc/commands/DeepslateCommand.java b/src/main/java/net/pascalpex/deepslatemc/commands/DeepslateCommand.java
index 9db7691b43f177a554b5768d5f446a0bb33390fa..65006575ec9180086266ab5eaf3d78de63e866c2 100644
--- a/src/main/java/net/pascalpex/deepslatemc/commands/DeepslateCommand.java
+++ b/src/main/java/net/pascalpex/deepslatemc/commands/DeepslateCommand.java
@@ -56,7 +56,7 @@ public class DeepslateCommand extends Command {
                     player.sendMessage(prefix + "Nutze /deepslate [reload | version]");
                 }
             } else {
-                player.sendMessage(prefix + "Dafür hast du keine Rechte");
+                player.sendMessage(prefix + "Dieser Server läuft mit DeepslateMC Version 1.0");
             }
         }
         return true;
diff --git a/src/main/java/net/pascalpex/deepslatemc/commands/DiscordCommand.java b/src/main/java/net/pascalpex/deepslatemc/commands/DiscordCommand.java
index 9d2fe8cfe962d86f1bb22c76e64d6cddb4809425..9ab77020d5a8f2403713e5622cb1204cda27d8f9 100644
--- a/src/main/java/net/pascalpex/deepslatemc/commands/DiscordCommand.java
+++ b/src/main/java/net/pascalpex/deepslatemc/commands/DiscordCommand.java
@@ -14,7 +14,6 @@ public class DiscordCommand extends Command {
         super(name);
         this.description = "Discord Link Befehl";
         this.usageMessage = "/discord";
-        this.setPermission("deepslate.discord");
     }
 
     @Override
