From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Sun, 23 Jan 2022 19:45:39 +0100
Subject: [PATCH] Clearchat Command


diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
index 8ec849fe9b4736ead7ca925c868eaa3d9bf15c4b..2b762a99c51b6eb6c68e4dbbc7194a3a300bdbf7 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
@@ -61,6 +61,7 @@ import net.minecraft.world.level.storage.WorldData;
 import net.pascalpex.deepslatemc.Config;
 import net.pascalpex.deepslatemc.commands.DeepslateCommand;
 import net.pascalpex.deepslatemc.commands.DiscordCommand;
+import net.pascalpex.deepslatemc.commands.ClearchatCommand;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.apache.logging.log4j.Level;
@@ -247,10 +248,14 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
         gg.pufferfish.pufferfish.PufferfishConfig.load(); // Pufferfish
         gg.pufferfish.pufferfish.PufferfishCommand.init(); // Pufferfish
 
+        //Deepslate start
         Config.load();
         MinecraftServer.getServer().server.getCommandMap().register("deepslate", "Deepslate", new DeepslateCommand("deepslate"));
         MinecraftServer.getServer().server.getCommandMap().register("discord", "Deepslate", new DiscordCommand("discord"));
         MinecraftServer.getServer().server.getCommandMap().register("dc", "Deepslate", new DiscordCommand("dc"));
+        MinecraftServer.getServer().server.getCommandMap().register("cc", "Deepslate", new ClearchatCommand("cc"));
+        MinecraftServer.getServer().server.getCommandMap().register("clearchat", "Deepslate", new ClearchatCommand("clearchat"));
+        //Deepslate end
 
         // Yatopia start
         try {
diff --git a/src/main/java/net/pascalpex/deepslatemc/commands/ClearchatCommand.java b/src/main/java/net/pascalpex/deepslatemc/commands/ClearchatCommand.java
new file mode 100644
index 0000000000000000000000000000000000000000..c3bd198b4f653599289261f2ab1535b0e8e9b611
--- /dev/null
+++ b/src/main/java/net/pascalpex/deepslatemc/commands/ClearchatCommand.java
@@ -0,0 +1,37 @@
+package net.pascalpex.deepslatemc.commands;
+    
+import org.bukkit.Bukkit;
+import org.bukkit.ChatColor;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+import org.jetbrains.annotations.NotNull;
+
+public class ClearchatCommand extends Command {
+    public ClearchatCommand(@NotNull String name) {
+        super(name);
+        this.description = "Leert den Chat";
+        this.usageMessage = "/clearchat";
+        this.setPermission("deepslate.clearchat");
+    }
+
+    @Override
+    public boolean execute(@NotNull CommandSender sender, @NotNull String commandLabel, @NotNull String[] args) {
+        if (sender instanceof Player) {
+            Player player = (Player) sender;
+            if (commandLabel.equalsIgnoreCase("cc") || commandLabel.equalsIgnoreCase("clearchat")) {
+                if(player.hasPermission("deepslate.clearchat")) {
+                    for (int x = 0; x < 150; x++){
+                        Bukkit.broadcastMessage("");
+                    }
+                    Bukkit.broadcastMessage(ChatColor.YELLOW + "Der Chat wurde von " + ChatColor.RED + player.getName() + ChatColor.YELLOW + " geleert");
+                }
+            } else {
+                player.sendMessage(ChatColor.RED + "Dafür hast du keine Rechte");
+            }
+        } else {
+            sender.sendMessage("Dieser Befehl ist nur für Spieler geeignet");
+        }
+        return true;
+    }
+}
diff --git a/src/main/java/net/pascalpex/deepslatemc/commands/DiscordCommand.java b/src/main/java/net/pascalpex/deepslatemc/commands/DiscordCommand.java
index e8b90d4b5e05ddb467189839899fe36b407433bc..9276201f7ed8757c534f47aa5f703c7f9ac6dd75 100644
--- a/src/main/java/net/pascalpex/deepslatemc/commands/DiscordCommand.java
+++ b/src/main/java/net/pascalpex/deepslatemc/commands/DiscordCommand.java
@@ -13,7 +13,6 @@ public class DiscordCommand extends Command {
         super(name);
         this.description = "Discord Link Befehl";
         this.usageMessage = "/discord";
-        this.setPermission("deepslate.discord");
     }
 
     @Override
