From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Sat, 3 Dec 2022 13:24:57 +0100
Subject: [PATCH] Clearchat command


diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
index 655d4a1586955c291944fcb903b5934e43e9c049..2230a0d2718a49ab4ffa3a61c21f7d2af0dac650 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
@@ -270,6 +270,8 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
         MinecraftServer.getServer().server.getCommandMap().register("deepslate", "Deepslate", new DeepslateCommand("deepslate"));
         MinecraftServer.getServer().server.getCommandMap().register("discord", "Deepslate", new DiscordCommand("discord"));
         MinecraftServer.getServer().server.getCommandMap().register("dc", "Deepslate", new DiscordCommand("dc"));
+        MinecraftServer.getServer().server.getCommandMap().register("cc", "Deepslate", new ClearchatCommand("cc"));
+        MinecraftServer.getServer().server.getCommandMap().register("clearchat", "Deepslate", new ClearchatCommand("clearchat"));
         // Deepslate end
 
         this.setPvpAllowed(dedicatedserverproperties.pvp);
diff --git a/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java b/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
index 092f25e176d9dc43ce7fafc7befe75df4b7d6aa1..67b1efd8e81b0c756b21bf04c19dbd3030fefe69 100644
--- a/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
+++ b/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
@@ -41,6 +41,12 @@ public class MessagesFile {
             if(!config.contains("discordMessage")) {
                 config.set("discordMessage", "&6The link to our Discord server: &b%link%");
             }
+            if(!config.contains("noPermissions")) {
+                config.set("noPermissions", "&cYou have no permission to do this");
+            }
+            if(!config.contains("clearedChat")) {
+                config.set("clearedChat", "&6The chat got cleared by &b%clearer%");
+            }
             save();
         } catch (IOException | InvalidConfigurationException e) {
             e.printStackTrace();
@@ -75,5 +81,13 @@ public class MessagesFile {
         String prefix = config.getString("discordMessage");
         return prefix.replace("&", "§");
     }
+    public static String getNoPermissions() {
+        String prefix = config.getString("noPermissions");
+        return prefix.replace("&", "§");
+    }
+    public static String getClearedChat() {
+        String prefix = config.getString("clearedChat");
+        return prefix.replace("&", "§");
+    }
 
 }
diff --git a/src/main/java/net/pascalpex/deepslatemc/commands/ClearchatCommand.java b/src/main/java/net/pascalpex/deepslatemc/commands/ClearchatCommand.java
new file mode 100644
index 0000000000000000000000000000000000000000..8fbb23119c33b11fe341701abfa9c34511c2dec7
--- /dev/null
+++ b/src/main/java/net/pascalpex/deepslatemc/commands/ClearchatCommand.java
@@ -0,0 +1,32 @@
+package net.pascalpex.deepslatemc.commands;
+    
+import net.pascalpex.deepslatemc.MessagesFile;
+import org.bukkit.Bukkit;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+
+public class ClearchatCommand extends Command {
+    public ClearchatCommand(String name) {
+        super(name);
+        this.description = "Clears the chat";
+        this.usageMessage = "/clearchat";
+        this.setPermission("deepslate.clearchat");
+    }
+
+    @Override
+    public boolean execute(CommandSender sender, String commandLabel, String[] args) {
+        String prefix = MessagesFile.getPrefix() + " ";
+        if (commandLabel.equalsIgnoreCase("cc") || commandLabel.equalsIgnoreCase("clearchat")) {
+            if(sender.hasPermission("deepslate.clearchat")) {
+                for (int x = 0; x < 150; x++){
+                    Bukkit.broadcastMessage("");
+                }
+                Bukkit.broadcastMessage(prefix + MessagesFile.getClearedChat().replace("%clearer%", sender.getName()));
+            } else {
+                sender.sendMessage(prefix + MessagesFile.getNoPermissions());
+            }
+        }
+        return true;
+    }
+}
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index a53b912e96aa39020b3f706607721da9817d800e..21f27e248f9008736b1746d7d07c36572e26976a 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -1132,6 +1132,8 @@ public final class CraftServer implements Server {
         MinecraftServer.getServer().server.getCommandMap().register("deepslate", "Deepslate", new DeepslateCommand("deepslate"));
         MinecraftServer.getServer().server.getCommandMap().register("discord", "Deepslate", new DiscordCommand("discord"));
         MinecraftServer.getServer().server.getCommandMap().register("dc", "Deepslate", new DiscordCommand("dc"));
+        MinecraftServer.getServer().server.getCommandMap().register("cc", "Deepslate", new ClearchatCommand("cc"));
+        MinecraftServer.getServer().server.getCommandMap().register("clearchat", "Deepslate", new ClearchatCommand("clearchat"));
         // Deepslate end
 
         this.overrideAllCommandBlockCommands = this.commandsConfiguration.getStringList("command-block-overrides").contains("*");
