From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Sat, 3 Dec 2022 13:24:57 +0100
Subject: [PATCH] Clearchat command


diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
index d0bc28c820b9f834d059ca924673418693d5c078..e711234946078c25dd10d00079eaafa2af5c7e80 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
@@ -237,6 +237,8 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
         MinecraftServer.getServer().server.getCommandMap().register("deepslate", "Deepslate", new DeepslateCommand("deepslate"));
         MinecraftServer.getServer().server.getCommandMap().register("discord", "Deepslate", new DiscordCommand("discord"));
         MinecraftServer.getServer().server.getCommandMap().register("dc", "Deepslate", new DiscordCommand("dc"));
+        MinecraftServer.getServer().server.getCommandMap().register("cc", "Deepslate", new ClearchatCommand("cc"));
+        MinecraftServer.getServer().server.getCommandMap().register("clearchat", "Deepslate", new ClearchatCommand("clearchat"));
         // Deepslate end
 
         // KeYi start - init config
diff --git a/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java b/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
index 073ee76cdefd6f0e40f7f1f80312f25e2406118b..0ae0c2a045739f98ecd10ecb3858bd7ceb7c8dfe 100644
--- a/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
+++ b/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
@@ -41,6 +41,12 @@ public class MessagesFile {
             if(!config.contains("discordMessage")) {
                 config.set("discordMessage", "&6The link to our Discord server: &b%link%");
             }
+            if(!config.contains("noPermissions")) {
+                config.set("noPermissions", "&cYou have no permission to do this");
+            }
+            if(!config.contains("clearedChat")) {
+                config.set("clearedChat", "&6The chat got cleared by &b%clearer%");
+            }
             save();
         } catch (IOException | InvalidConfigurationException e) {
             e.printStackTrace();
@@ -75,5 +81,13 @@ public class MessagesFile {
         String prefix = config.getString("discordMessage");
         return prefix.replace("&", "§");
     }
+    public static String getNoPermissions() {
+        String prefix = config.getString("noPermissions");
+        return prefix.replace("&", "§");
+    }
+    public static String getClearedChat() {
+        String prefix = config.getString("clearedChat");
+        return prefix.replace("&", "§");
+    }
 
 }
diff --git a/src/main/java/net/pascalpex/deepslatemc/commands/ClearchatCommand.java b/src/main/java/net/pascalpex/deepslatemc/commands/ClearchatCommand.java
new file mode 100644
index 0000000000000000000000000000000000000000..c596ee8baf01e908cac6c2d80c1c11a649b483c5
--- /dev/null
+++ b/src/main/java/net/pascalpex/deepslatemc/commands/ClearchatCommand.java
@@ -0,0 +1,37 @@
+package net.pascalpex.deepslatemc.commands;
+    
+import net.pascalpex.deepslatemc.MessagesFile;
+import org.bukkit.Bukkit;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+
+public class ClearchatCommand extends Command {
+    public ClearchatCommand(String name) {
+        super(name);
+        this.description = "Clears the chat";
+        this.usageMessage = "/clearchat";
+        this.setPermission("deepslate.clearchat");
+    }
+
+    @Override
+    public boolean execute(CommandSender sender, String commandLabel, String[] args) {
+        String prefix = MessagesFile.getPrefix() + " ";
+        if (commandLabel.equalsIgnoreCase("cc") || commandLabel.equalsIgnoreCase("clearchat")) {
+            if (sender instanceof Player) {
+                Player player = (Player) sender;
+                if(player.hasPermission("deepslate.clearchat")) {
+                    for (int x = 0; x < 150; x++){
+                        Bukkit.broadcastMessage("");
+                    }
+                    Bukkit.broadcastMessage(prefix + MessagesFile.getClearedChat().replace("%clearer%", player.getName()));
+                } else {
+                    player.sendMessage(prefix + MessagesFile.getNoPermissions());
+                }
+            } else {
+                sender.sendMessage(prefix + MessagesFile.getOnlyForPlayers());
+            }
+        }
+        return true;
+    }
+}
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 4384a7315b5b217de5f51b76657432e66b8d7444..c749c439142d2da1676be0632245e06d160ce494 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -1033,6 +1033,8 @@ public final class CraftServer implements Server {
         MinecraftServer.getServer().server.getCommandMap().register("deepslate", "Deepslate", new DeepslateCommand("deepslate"));
         MinecraftServer.getServer().server.getCommandMap().register("discord", "Deepslate", new DiscordCommand("discord"));
         MinecraftServer.getServer().server.getCommandMap().register("dc", "Deepslate", new DiscordCommand("dc"));
+        MinecraftServer.getServer().server.getCommandMap().register("cc", "Deepslate", new ClearchatCommand("cc"));
+        MinecraftServer.getServer().server.getCommandMap().register("clearchat", "Deepslate", new ClearchatCommand("clearchat"));
         // Deepslate end
 
         this.overrideAllCommandBlockCommands = this.commandsConfiguration.getStringList("command-block-overrides").contains("*");
