From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Sun, 23 Jan 2022 19:49:26 +0100
Subject: [PATCH] Bauwelt Command


diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
index 2b762a99c51b6eb6c68e4dbbc7194a3a300bdbf7..c118f596fd51cd09589f70100f9a77c726dce8d6 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
@@ -62,6 +62,7 @@ import net.pascalpex.deepslatemc.Config;
 import net.pascalpex.deepslatemc.commands.DeepslateCommand;
 import net.pascalpex.deepslatemc.commands.DiscordCommand;
 import net.pascalpex.deepslatemc.commands.ClearchatCommand;
+import net.pascalpex.deepslatemc.commands.BauweltCommand;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.apache.logging.log4j.Level;
@@ -255,6 +256,8 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
         MinecraftServer.getServer().server.getCommandMap().register("dc", "Deepslate", new DiscordCommand("dc"));
         MinecraftServer.getServer().server.getCommandMap().register("cc", "Deepslate", new ClearchatCommand("cc"));
         MinecraftServer.getServer().server.getCommandMap().register("clearchat", "Deepslate", new ClearchatCommand("clearchat"));
+        MinecraftServer.getServer().server.getCommandMap().register("setbauwelt", "Deepslate", new BauweltCommand("setbauwelt"));
+        MinecraftServer.getServer().server.getCommandMap().register("bauwelt", "Deepslate", new BauweltCommand("bauwelt"));
         //Deepslate end
 
         // Yatopia start
diff --git a/src/main/java/net/pascalpex/deepslatemc/Config.java b/src/main/java/net/pascalpex/deepslatemc/Config.java
index e0136508752affde08f1c785192a3cb3e45ebe9d..4cfc701c47db9a2a513b4e4af9a9230459f887bc 100644
--- a/src/main/java/net/pascalpex/deepslatemc/Config.java
+++ b/src/main/java/net/pascalpex/deepslatemc/Config.java
@@ -46,4 +46,13 @@ public class Config {
         return config.getString("discordLink");
     }
 
+    public static String getBauwelt() {
+        return config.getString("bauwelt");
+    }
+
+    public static void setBauwelt(String worldName) {
+        config.set("bauwelt", worldName);
+        save();
+    }
+
 }
diff --git a/src/main/java/net/pascalpex/deepslatemc/commands/BauweltCommand.java b/src/main/java/net/pascalpex/deepslatemc/commands/BauweltCommand.java
new file mode 100644
index 0000000000000000000000000000000000000000..5ceab092ef009ecbf65d035a57ac4933385e8526
--- /dev/null
+++ b/src/main/java/net/pascalpex/deepslatemc/commands/BauweltCommand.java
@@ -0,0 +1,43 @@
+package net.pascalpex.deepslatemc.commands;
+    
+import net.pascalpex.deepslatemc.Config;
+import org.bukkit.Bukkit;
+import org.bukkit.ChatColor;
+import org.bukkit.World;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+import org.jetbrains.annotations.NotNull;
+
+public class BauweltCommand extends Command {
+
+    public BauweltCommand(@NotNull String name) {
+        super(name);
+        this.description = "Teleportiert Builder in die Bauwelt";
+        this.usageMessage = "/bauwelt";
+        this.setPermission("deepslate.bauwelt");
+    }
+
+    @Override
+    public boolean execute(@NotNull CommandSender sender, @NotNull String label, @NotNull String[] args) {
+        Player player = (Player) sender;
+        if(label.equalsIgnoreCase("setbauwelt")) {
+            if(player.hasPermission("deepslate.setbauwelt")) {
+                World world = player.getLocation().getWorld();
+                Config.setBauwelt(world.getName());
+                player.sendMessage(ChatColor.GREEN + "Bauwelt gesetzt");
+            }
+        }
+        if(label.equalsIgnoreCase("bauwelt")) {
+            if(player.hasPermission("deepslate.bauwelt")) {
+                try {
+                    player.teleport(Bukkit.getWorld(Config.getBauwelt()).getSpawnLocation());
+                    player.sendMessage(ChatColor.GREEN + "Wilkommen in der Bauwelt");
+                } catch (Exception e) {
+                    player.sendMessage(ChatColor.RED + "Die Bauwelt wurde nicht gesetzt");
+                }
+            }
+        }
+        return true;
+    }
+}
