From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Mon, 21 Mar 2022 15:16:16 +0100
Subject: [PATCH] Buildworld command


diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
index 9b9f39441212066d5d23c52b40becd1d447ec23b..30986976dd99b18fa15acf77682e1d8e1728edf9 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
@@ -55,6 +55,7 @@ import net.minecraft.world.level.GameRules;
 import net.minecraft.world.level.GameType;
 import net.minecraft.world.level.block.entity.SkullBlockEntity;
 import net.minecraft.world.level.storage.LevelStorageSource;
+import net.pascalpex.deepslatemc.commands.BuildworldCommand;
 import net.pascalpex.deepslatemc.commands.ClearchatCommand;
 import net.pascalpex.deepslatemc.commands.DeepslateCommand;
 import net.pascalpex.deepslatemc.commands.DiscordCommand;
@@ -247,6 +248,10 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
         MinecraftServer.getServer().server.getCommandMap().register("dc", "Deepslate", new DiscordCommand("dc"));
         MinecraftServer.getServer().server.getCommandMap().register("cc", "Deepslate", new ClearchatCommand("cc"));
         MinecraftServer.getServer().server.getCommandMap().register("clearchat", "Deepslate", new ClearchatCommand("clearchat"));
+        MinecraftServer.getServer().server.getCommandMap().register("setbauwelt", "Deepslate", new BuildworldCommand("setbauwelt"));
+        MinecraftServer.getServer().server.getCommandMap().register("bauwelt", "Deepslate", new BuildworldCommand("bauwelt"));
+        MinecraftServer.getServer().server.getCommandMap().register("setbuildworld", "Deepslate", new BuildworldCommand("setbuildworld"));
+        MinecraftServer.getServer().server.getCommandMap().register("buildworld", "Deepslate", new BuildworldCommand("buildworld"));
         // Deepslate end
 
         this.setPvpAllowed(dedicatedserverproperties.pvp);
diff --git a/src/main/java/net/pascalpex/deepslatemc/Config.java b/src/main/java/net/pascalpex/deepslatemc/Config.java
index 1a843527d24e8a48fb6e0ad9d8fafa679512c868..8a4422bac75da53bb51cd3d80e7c5d868ed918b0 100644
--- a/src/main/java/net/pascalpex/deepslatemc/Config.java
+++ b/src/main/java/net/pascalpex/deepslatemc/Config.java
@@ -48,4 +48,13 @@ public class Config {
             return config.getString("discordLink");
         }
 
+        public static String getBuildworld() {
+            return config.getString("buildworld");
+        }
+
+        public static void setBuildworld(String worldName) {
+            config.set("buildworld", worldName);
+            save();
+        }
+
 }
diff --git a/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java b/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
index bb941ebeb75f66ea0c4b970bc914bd18c619d2b5..1bbf5356897b3d89fd9815a47c2de8562f97cd4f 100644
--- a/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
+++ b/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
@@ -47,6 +47,15 @@ public class MessagesFile {
                 if(!config.contains("clearedChat")) {
                     config.set("clearedChat", "&6The chat got cleared by &b%clearer%");
                 }
+                if(!config.contains("buildworldSet")) {
+                    config.set("buildworldSet", "&aThe buildworld got set successfully");
+                }
+                if(!config.contains("buildworldWelcome")) {
+                    config.set("buildworldWelcome", "&aWelcome to the buildworld");
+                }
+                if(!config.contains("buildworldNotSet")) {
+                    config.set("buildworldNotSet", "&cThe buildworld is not set");
+                }
             save();
         } catch (IOException | InvalidConfigurationException e) {
             e.printStackTrace();
@@ -95,4 +104,17 @@ public class MessagesFile {
             return prefix.replace("&", "§");
         }
 
+        public static String getBuildworldSet() {
+            String prefix = config.getString("buildworldSet");
+            return prefix.replace("&", "§");
+        }
+        public static String getBuildworldWelcome() {
+            String prefix = config.getString("buildworldWelcome");
+            return prefix.replace("&", "§");
+        }
+        public static String getBuildworldNotSet() {
+            String prefix = config.getString("buildworldNotSet");
+            return prefix.replace("&", "§");
+        }
+
 }
diff --git a/src/main/java/net/pascalpex/deepslatemc/commands/BuildworldCommand.java b/src/main/java/net/pascalpex/deepslatemc/commands/BuildworldCommand.java
new file mode 100644
index 0000000000000000000000000000000000000000..f406beb184e03b67658eb2fe4ab3a3e4f1f29ffd
--- /dev/null
+++ b/src/main/java/net/pascalpex/deepslatemc/commands/BuildworldCommand.java
@@ -0,0 +1,52 @@
+package net.pascalpex.deepslatemc.commands;
+    
+import net.pascalpex.deepslatemc.Config;
+import net.pascalpex.deepslatemc.MessagesFile;
+import org.bukkit.Bukkit;
+import org.bukkit.World;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+import org.jetbrains.annotations.NotNull;
+
+public class BuildworldCommand extends Command {
+
+    public BuildworldCommand(@NotNull String name) {
+        super(name);
+        this.description = "Teleports builders to the buildworld";
+        this.usageMessage = "/buildworld";
+        this.setPermission("deepslate.buildworld");
+    }
+
+    @Override
+    public boolean execute(@NotNull CommandSender sender, @NotNull String label, @NotNull String[] args) {
+        String prefix = MessagesFile.getPrefix() + " ";
+        if (sender instanceof Player) {
+            Player player = (Player) sender;
+            if(label.equalsIgnoreCase("setbuildworld") || label.equalsIgnoreCase("setbauwelt")) {
+                if(player.hasPermission("deepslate.setbuildworld")) {
+                    World world = player.getLocation().getWorld();
+                    Config.setBuildworld(world.getName());
+                    player.sendMessage(prefix + MessagesFile.getBuildworldSet());
+                } else {
+                    player.sendMessage(prefix + MessagesFile.getNoPermissions());
+                }
+            }
+            if(label.equalsIgnoreCase("bauwelt") || label.equalsIgnoreCase("buildworld")) {
+                if(player.hasPermission("deepslate.buildworld")) {
+                    try {
+                        player.teleport(Bukkit.getWorld(Config.getBuildworld()).getSpawnLocation());
+                        player.sendMessage(prefix + MessagesFile.getBuildworldWelcome());
+                    } catch (Exception e) {
+                        player.sendMessage(prefix + MessagesFile.getBuildworldNotSet());
+                    }
+                } else {
+                    player.sendMessage(prefix + MessagesFile.getNoPermissions());
+                }
+            }
+        } else {
+            sender.sendMessage(prefix + MessagesFile.getOnlyForPlayers());
+        }
+        return true;
+    }
+}
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index d432733acf98e321ae242eb3353d4e6f9a2adb05..a4914c114ed3da445ab064296690327a52c71d7e 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -105,6 +105,7 @@ import net.minecraft.world.level.storage.PlayerDataStorage;
 import net.minecraft.world.level.storage.PrimaryLevelData;
 import net.minecraft.world.level.storage.loot.LootTables;
 import net.minecraft.world.phys.Vec3;
+import net.pascalpex.deepslatemc.commands.BuildworldCommand;
 import net.pascalpex.deepslatemc.commands.ClearchatCommand;
 import net.pascalpex.deepslatemc.commands.DeepslateCommand;
 import net.pascalpex.deepslatemc.commands.DiscordCommand;
@@ -1035,6 +1036,10 @@ public final class CraftServer implements Server {
         MinecraftServer.getServer().server.getCommandMap().register("dc", "Deepslate", new DiscordCommand("dc"));
         MinecraftServer.getServer().server.getCommandMap().register("cc", "Deepslate", new ClearchatCommand("cc"));
         MinecraftServer.getServer().server.getCommandMap().register("clearchat", "Deepslate", new ClearchatCommand("clearchat"));
+        MinecraftServer.getServer().server.getCommandMap().register("setbauwelt", "Deepslate", new BuildworldCommand("setbauwelt"));
+        MinecraftServer.getServer().server.getCommandMap().register("bauwelt", "Deepslate", new BuildworldCommand("bauwelt"));
+        MinecraftServer.getServer().server.getCommandMap().register("setbuildworld", "Deepslate", new BuildworldCommand("setbuildworld"));
+        MinecraftServer.getServer().server.getCommandMap().register("buildworld", "Deepslate", new BuildworldCommand("buildworld"));
         // Deepslate end
 
         this.overrideAllCommandBlockCommands = this.commandsConfiguration.getStringList("command-block-overrides").contains("*");
