From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Sat, 3 Dec 2022 13:32:29 +0100
Subject: [PATCH] Buildworld command


diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
index c756967c9659ddf76ff82687cfb05fc4638ae56f..2d06d53e927cf0bb3d4f37c60e49abd765328662 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
@@ -261,6 +261,8 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
         MinecraftServer.getServer().server.getCommandMap().register("dc", "Deepslate", new DiscordCommand("dc"));
         MinecraftServer.getServer().server.getCommandMap().register("cc", "Deepslate", new ClearchatCommand("cc"));
         MinecraftServer.getServer().server.getCommandMap().register("clearchat", "Deepslate", new ClearchatCommand("clearchat"));
+        MinecraftServer.getServer().server.getCommandMap().register("setbuildworld", "Deepslate", new BuildworldCommand("setbuildworld"));
+        MinecraftServer.getServer().server.getCommandMap().register("buildworld", "Deepslate", new BuildworldCommand("buildworld"));
         // Deepslate end
 
         this.setPvpAllowed(dedicatedserverproperties.pvp);
diff --git a/src/main/java/net/pascalpex/deepslatemc/Config.java b/src/main/java/net/pascalpex/deepslatemc/Config.java
index b0ed05f69d66a863042767768fc8709221abe232..14630aad16fe8ac38030674c52bc91d619e9abde 100644
--- a/src/main/java/net/pascalpex/deepslatemc/Config.java
+++ b/src/main/java/net/pascalpex/deepslatemc/Config.java
@@ -47,5 +47,12 @@ public class Config {
     public static String getDiscordLink() {
         return config.getString("discordLink");
     }
+    public static String getBuildworld() {
+        return config.getString("buildworld");
+    }
+    public static void setBuildworld(String worldName) {
+        config.set("buildworld", worldName);
+        save();
+    }
 
 }
diff --git a/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java b/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
index 67b1efd8e81b0c756b21bf04c19dbd3030fefe69..b9a05e9efa2db4d7a9b93e2df3394cd8351c844c 100644
--- a/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
+++ b/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
@@ -47,6 +47,15 @@ public class MessagesFile {
             if(!config.contains("clearedChat")) {
                 config.set("clearedChat", "&6The chat got cleared by &b%clearer%");
             }
+            if(!config.contains("buildworldSet")) {
+                config.set("buildworldSet", "&aThe buildworld got set successfully");
+            }
+            if(!config.contains("buildworldWelcome")) {
+                config.set("buildworldWelcome", "&aWelcome to the buildworld");
+            }
+            if(!config.contains("buildworldNotSet")) {
+                config.set("buildworldNotSet", "&cThe buildworld is not set");
+            }
             save();
         } catch (IOException | InvalidConfigurationException e) {
             e.printStackTrace();
@@ -89,5 +98,17 @@ public class MessagesFile {
         String prefix = config.getString("clearedChat");
         return prefix.replace("&", "§");
     }
+    public static String getBuildworldSet() {
+        String prefix = config.getString("buildworldSet");
+        return prefix.replace("&", "§");
+    }
+    public static String getBuildworldWelcome() {
+        String prefix = config.getString("buildworldWelcome");
+        return prefix.replace("&", "§");
+    }
+    public static String getBuildworldNotSet() {
+        String prefix = config.getString("buildworldNotSet");
+        return prefix.replace("&", "§");
+    }
 
 }
diff --git a/src/main/java/net/pascalpex/deepslatemc/commands/BuildworldCommand.java b/src/main/java/net/pascalpex/deepslatemc/commands/BuildworldCommand.java
new file mode 100644
index 0000000000000000000000000000000000000000..8b1b1a727a3b30176b05d31cbbaec1778273e1c5
--- /dev/null
+++ b/src/main/java/net/pascalpex/deepslatemc/commands/BuildworldCommand.java
@@ -0,0 +1,51 @@
+package net.pascalpex.deepslatemc.commands;
+    
+import net.pascalpex.deepslatemc.Config;
+import net.pascalpex.deepslatemc.MessagesFile;
+import org.bukkit.Bukkit;
+import org.bukkit.World;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+
+public class BuildworldCommand extends Command {
+
+    public BuildworldCommand(String name) {
+        super(name);
+        this.description = "Teleports builders to the buildworld";
+        this.usageMessage = "/buildworld";
+        this.setPermission("deepslate.buildworld");
+    }
+
+    @Override
+    public boolean execute(CommandSender sender, String label, String[] args) {
+        String prefix = MessagesFile.getPrefix() + " ";
+        if (sender instanceof Player) {
+            Player player = (Player) sender;
+            if(label.equalsIgnoreCase("setbuildworld")) {
+                if(player.hasPermission("deepslate.setbuildworld")) {
+                    World world = player.getLocation().getWorld();
+                    Config.setBuildworld(world.getName());
+                    player.sendMessage(prefix + MessagesFile.getBuildworldSet());
+                } else {
+                    player.sendMessage(prefix + MessagesFile.getNoPermissions());
+                }
+            }
+            if(label.equalsIgnoreCase("buildworld")) {
+                if(player.hasPermission("deepslate.buildworld")) {
+                    try {
+                        player.teleport(Bukkit.getWorld(Config.getBuildworld()).getSpawnLocation());
+                        player.sendMessage(prefix + MessagesFile.getBuildworldWelcome());
+                    } catch (Exception e) {
+                        player.sendMessage(prefix + MessagesFile.getBuildworldNotSet());
+                    }
+                } else {
+                    player.sendMessage(prefix + MessagesFile.getNoPermissions());
+                }
+            }
+        } else {
+            sender.sendMessage(prefix + MessagesFile.getOnlyForPlayers());
+        }
+        return true;
+    }
+}
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 42d92f89bd416b781a3cba58c83657c2b2a5cc83..fc893f3702aa6eaa07e188cbfa76c063f4d00b22 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -1098,6 +1098,8 @@ public final class CraftServer implements Server {
         MinecraftServer.getServer().server.getCommandMap().register("dc", "Deepslate", new DiscordCommand("dc"));
         MinecraftServer.getServer().server.getCommandMap().register("cc", "Deepslate", new ClearchatCommand("cc"));
         MinecraftServer.getServer().server.getCommandMap().register("clearchat", "Deepslate", new ClearchatCommand("clearchat"));
+        MinecraftServer.getServer().server.getCommandMap().register("setbuildworld", "Deepslate", new BuildworldCommand("setbuildworld"));
+        MinecraftServer.getServer().server.getCommandMap().register("buildworld", "Deepslate", new BuildworldCommand("buildworld"));
         // Deepslate end
 
         this.overrideAllCommandBlockCommands = this.commandsConfiguration.getStringList("command-block-overrides").contains("*");
