From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Sun, 30 Jan 2022 14:24:52 +0100
Subject: [PATCH] Wartung Mode


diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
index 12c8970943008d35a17b661f55e72ca71c47d401..3aba32673a5499a9e0eb391f9da342ea4fcde08b 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
@@ -59,10 +59,7 @@ import net.minecraft.world.level.block.entity.SkullBlockEntity;
 import net.minecraft.world.level.storage.LevelStorageSource;
 import net.minecraft.world.level.storage.WorldData;
 import net.pascalpex.deepslatemc.Config;
-import net.pascalpex.deepslatemc.commands.BauweltCommand;
-import net.pascalpex.deepslatemc.commands.ClearchatCommand;
-import net.pascalpex.deepslatemc.commands.DeepslateCommand;
-import net.pascalpex.deepslatemc.commands.DiscordCommand;
+import net.pascalpex.deepslatemc.commands.*;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.apache.logging.log4j.Level;
@@ -268,6 +265,7 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
         MinecraftServer.getServer().server.getCommandMap().register("clearchat", "Deepslate", new ClearchatCommand("clearchat"));
         MinecraftServer.getServer().server.getCommandMap().register("setbauwelt", "Deepslate", new BauweltCommand("setbauwelt"));
         MinecraftServer.getServer().server.getCommandMap().register("bauwelt", "Deepslate", new BauweltCommand("bauwelt"));
+        MinecraftServer.getServer().server.getCommandMap().register("wartung", "Deepslate", new WartungCommand("wartung"));
         //Deepslate end
 
         this.setPvpAllowed(dedicatedserverproperties.pvp);
diff --git a/src/main/java/net/minecraft/server/players/PlayerList.java b/src/main/java/net/minecraft/server/players/PlayerList.java
index 5b993156daa75ecdd0cdcd014dadcb3b6b01df1b..78c60d2e0b852db70aac84315d7d6610a5c1293a 100644
--- a/src/main/java/net/minecraft/server/players/PlayerList.java
+++ b/src/main/java/net/minecraft/server/players/PlayerList.java
@@ -91,6 +91,7 @@ import net.minecraft.world.scores.Objective;
 import net.minecraft.world.scores.PlayerTeam;
 import net.minecraft.world.scores.Scoreboard; // Paper
 import net.minecraft.world.scores.Team;
+import net.pascalpex.deepslatemc.commands.WartungCommand;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
@@ -360,6 +361,7 @@ public abstract class PlayerList {
         player.containerMenu.transferTo(player.containerMenu, bukkitPlayer);
 
         PlayerJoinEvent playerJoinEvent = new PlayerJoinEvent(bukkitPlayer, PaperAdventure.asAdventure(chatmessage)); // Paper - Adventure
+        WartungCommand.playerJoin(bukkitPlayer);
         this.cserver.getPluginManager().callEvent(playerJoinEvent);
 
         if (!player.connection.connection.isConnected()) {
diff --git a/src/main/java/net/pascalpex/deepslatemc/commands/WartungCommand.java b/src/main/java/net/pascalpex/deepslatemc/commands/WartungCommand.java
new file mode 100644
index 0000000000000000000000000000000000000000..2580dc98414b6b6113d00f44dadcaf7ef87e50f5
--- /dev/null
+++ b/src/main/java/net/pascalpex/deepslatemc/commands/WartungCommand.java
@@ -0,0 +1,58 @@
+package net.pascalpex.deepslatemc.commands;
+
+import net.pascalpex.deepslatemc.Config;
+import org.bukkit.ChatColor;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+import org.jetbrains.annotations.NotNull;
+
+public class WartungCommand extends Command {
+
+    private static boolean wartungMode = false;
+
+    public WartungCommand(@NotNull String name) {
+        super(name);
+        this.description = "Schaltet den Wartungsmodus an/aus";
+        this.usageMessage = "/wartung";
+        this.setPermission("deepslate.wartung");
+    }
+
+    public static void playerJoin(Player player) {
+        if(!player.hasPermission("deepslate.wartungbypass")) {
+            if(wartungMode) {
+                player.kickPlayer(ChatColor.DARK_GRAY + "[" + ChatColor.GREEN + "Pascalpex" + ChatColor.DARK_GRAY + "]" + ChatColor.GOLD + " Der Server ist aktuell im Wartungsmodus");
+            }
+        }
+    }
+
+    @Override
+    public boolean execute(@NotNull CommandSender sender, @NotNull String label, @NotNull String[] args) {
+        String prefix = Config.getPrefix() + " " + ChatColor.DARK_AQUA;
+        if (sender instanceof Player) {
+            Player player = (Player) sender;
+            if(label.equalsIgnoreCase("wartung")) {
+                if(player.hasPermission("deepslate.wartung")) {
+                    if(wartungMode) {
+                        wartungMode = false;
+                        player.sendMessage(prefix + "Der Wartungsmodus ist nun deaktiviert");
+                    } else {
+                        wartungMode = true;
+                        player.sendMessage(prefix + "Der Wartungsmodus ist nun aktiviert");
+                    }
+                }
+            }
+        } else {
+            if(label.equalsIgnoreCase("wartung")) {
+                if(wartungMode) {
+                    wartungMode = false;
+                    sender.sendMessage(prefix + "Der Wartungsmodus ist nun deaktiviert");
+                } else {
+                    wartungMode = true;
+                    sender.sendMessage(prefix + "Der Wartungsmodus ist nun aktiviert");
+                }
+            }
+        }
+        return true;
+    }
+}
