From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Sun, 23 Jan 2022 20:01:50 +0100
Subject: [PATCH] Wartung Mode


diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
index c118f596fd51cd09589f70100f9a77c726dce8d6..22fbad3f47d500937da383f49b4d4b1ad0755e37 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
@@ -59,10 +59,7 @@ import net.minecraft.world.level.block.entity.SkullBlockEntity;
 import net.minecraft.world.level.storage.LevelStorageSource;
 import net.minecraft.world.level.storage.WorldData;
 import net.pascalpex.deepslatemc.Config;
-import net.pascalpex.deepslatemc.commands.DeepslateCommand;
-import net.pascalpex.deepslatemc.commands.DiscordCommand;
-import net.pascalpex.deepslatemc.commands.ClearchatCommand;
-import net.pascalpex.deepslatemc.commands.BauweltCommand;
+import net.pascalpex.deepslatemc.commands.*;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.apache.logging.log4j.Level;
@@ -258,6 +255,7 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
         MinecraftServer.getServer().server.getCommandMap().register("clearchat", "Deepslate", new ClearchatCommand("clearchat"));
         MinecraftServer.getServer().server.getCommandMap().register("setbauwelt", "Deepslate", new BauweltCommand("setbauwelt"));
         MinecraftServer.getServer().server.getCommandMap().register("bauwelt", "Deepslate", new BauweltCommand("bauwelt"));
+        MinecraftServer.getServer().server.getCommandMap().register("wartung", "Deepslate", new WartungCommand("wartung"));
         //Deepslate end
 
         // Yatopia start
diff --git a/src/main/java/net/minecraft/server/players/PlayerList.java b/src/main/java/net/minecraft/server/players/PlayerList.java
index 15afcf4e8048fc15f1a2d407972cca90df8f66c0..571fd37ed09a2bc8b5838b7a3d159fa0d99753c8 100644
--- a/src/main/java/net/minecraft/server/players/PlayerList.java
+++ b/src/main/java/net/minecraft/server/players/PlayerList.java
@@ -91,6 +91,7 @@ import net.minecraft.world.scores.Objective;
 import net.minecraft.world.scores.PlayerTeam;
 import net.minecraft.world.scores.Scoreboard; // Paper
 import net.minecraft.world.scores.Team;
+import net.pascalpex.deepslatemc.commands.WartungCommand;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
@@ -356,6 +357,7 @@ public abstract class PlayerList {
         player.containerMenu.transferTo(player.containerMenu, bukkitPlayer);
 
         PlayerJoinEvent playerJoinEvent = new PlayerJoinEvent(bukkitPlayer, PaperAdventure.asAdventure(chatmessage)); // Paper - Adventure
+        WartungCommand.playerJoin(bukkitPlayer);
         this.cserver.getPluginManager().callEvent(playerJoinEvent);
 
         if (!player.connection.connection.isConnected()) {
diff --git a/src/main/java/net/pascalpex/deepslatemc/commands/WartungCommand.java b/src/main/java/net/pascalpex/deepslatemc/commands/WartungCommand.java
new file mode 100644
index 0000000000000000000000000000000000000000..8edd4e20d05acac568a364509604716989aa0941
--- /dev/null
+++ b/src/main/java/net/pascalpex/deepslatemc/commands/WartungCommand.java
@@ -0,0 +1,58 @@
+package net.pascalpex.deepslatemc.commands;
+    
+import net.pascalpex.deepslatemc.Config;
+import org.bukkit.ChatColor;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+import org.jetbrains.annotations.NotNull;
+
+public class WartungCommand extends Command {
+
+    private static boolean wartungMode = false;
+
+    public WartungCommand(@NotNull String name) {
+        super(name);
+        this.description = "Schaltet den Wartungsmodus an/aus";
+        this.usageMessage = "/wartung";
+        this.setPermission("deepslate.wartung");
+    }
+
+    public static void playerJoin(Player player) {
+        if(!player.hasPermission("deepslate.wartungbypass")) {
+            if(wartungMode) {
+                player.kickPlayer(ChatColor.DARK_GRAY + "[" + ChatColor.GREEN + "Pascalpex" + ChatColor.DARK_GRAY + "]" + ChatColor.GOLD + " Der Server ist aktuell im Wartungsmodus");
+            }
+        }
+    }
+
+    @Override
+    public boolean execute(@NotNull CommandSender sender, @NotNull String label, @NotNull String[] args) {
+        String prefix = Config.getPrefix() + " " + ChatColor.DARK_AQUA;
+        if (sender instanceof Player) {
+            Player player = (Player) sender;
+            if(label.equalsIgnoreCase("wartung")) {
+                if(player.hasPermission("deepslate.wartung")) {
+                    if(wartungMode) {
+                        wartungMode = false;
+                        player.sendMessage(prefix + "Der Wartungsmodus ist nun deaktiviert");
+                    } else {
+                        wartungMode = true;
+                        player.sendMessage(prefix + "Der Wartungsmodus ist nun aktiviert");
+                    }
+                }
+            }
+        } else {
+            if(label.equalsIgnoreCase("wartung")) {
+                if(wartungMode) {
+                    wartungMode = false;
+                    sender.sendMessage(prefix + "Der Wartungsmodus ist nun deaktiviert");
+                } else {
+                    wartungMode = true;
+                    sender.sendMessage(prefix + "Der Wartungsmodus ist nun aktiviert");
+                }
+            }
+        }
+        return true;
+    }
+}
