From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Mon, 25 Dec 2023 15:20:22 +0100
Subject: [PATCH] Maintenance mode


diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
index 2ac92d733c63032aefe815e2eda380cdc1d535fa..f5374cfc948192334af9a95a3fa4d55fdebc95ea 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
@@ -252,6 +252,7 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
         MinecraftServer.getServer().server.getCommandMap().register("clearchat", "Deepslate", new ClearchatCommand("clearchat"));
         MinecraftServer.getServer().server.getCommandMap().register("setbuildworld", "Deepslate", new BuildworldCommand("setbuildworld"));
         MinecraftServer.getServer().server.getCommandMap().register("buildworld", "Deepslate", new BuildworldCommand("buildworld"));
+        MinecraftServer.getServer().server.getCommandMap().register("maintenance", "Deepslate", new MaintenanceMode("maintenance"));
         // Deepslate end
 
         this.setPvpAllowed(dedicatedserverproperties.pvp);
diff --git a/src/main/java/net/minecraft/server/players/PlayerList.java b/src/main/java/net/minecraft/server/players/PlayerList.java
index f30825df1eb2a4a9383b1fe33fa76ebba98bc999..ae2139b892b5968dafae394b7b0e086164af108d 100644
--- a/src/main/java/net/minecraft/server/players/PlayerList.java
+++ b/src/main/java/net/minecraft/server/players/PlayerList.java
@@ -103,6 +103,8 @@ import net.minecraft.world.phys.Vec3;
 import net.minecraft.world.scores.DisplaySlot;
 import net.minecraft.world.scores.Objective;
 import net.minecraft.world.scores.PlayerTeam;
+import net.pascalpex.deepslatemc.Config;
+import net.pascalpex.deepslatemc.MessagesFile;
 import org.galemc.gale.configuration.GaleGlobalConfiguration;
 import org.slf4j.Logger;
 
@@ -752,6 +754,15 @@ public abstract class PlayerList {
         Player player = entity.getBukkitEntity();
         PlayerLoginEvent event = new PlayerLoginEvent(player, loginlistener.connection.hostname, ((java.net.InetSocketAddress) socketaddress).getAddress(), ((java.net.InetSocketAddress) loginlistener.connection.channel.remoteAddress()).getAddress());
 
+        // Deepslate start - Maintenance mode
+        if(Config.getMaintenanceMode()) {
+            if(!player.hasPermission("deepslate.maintenancebypass")) {
+                String prefix = MessagesFile.getPrefix();
+                event.disallow(PlayerLoginEvent.Result.KICK_OTHER, prefix + MessagesFile.getMaintenanceKick());
+            }
+        }
+        // Deepslate end
+
         // Paper start - Fix MC-158900
         UserBanListEntry gameprofilebanentry;
         if (getBans().isBanned(gameprofile) && (gameprofilebanentry = getBans().get(gameprofile)) != null) {
diff --git a/src/main/java/net/pascalpex/deepslatemc/Config.java b/src/main/java/net/pascalpex/deepslatemc/Config.java
index 14630aad16fe8ac38030674c52bc91d619e9abde..97119bdc69d79216241bf1cf0c0b2ae5b230134b 100644
--- a/src/main/java/net/pascalpex/deepslatemc/Config.java
+++ b/src/main/java/net/pascalpex/deepslatemc/Config.java
@@ -30,6 +30,9 @@ public class Config {
             if(!config.contains("discordLink")) {
                 config.set("discordLink", "https://discord.gg/BGrhNnVczp");
             }
+            if(!config.contains("maintenance")) {
+                config.set("maintenance", false);
+            }
             save();
         } catch (IOException | InvalidConfigurationException e) {
             e.printStackTrace();
@@ -54,5 +57,12 @@ public class Config {
         config.set("buildworld", worldName);
         save();
     }
+    public static boolean getMaintenanceMode() {
+        return config.getBoolean("maintenance");
+    }
+    public static void toggleMaintenanceMode() {
+        config.set("maintenance", !getMaintenanceMode());
+        save();
+    }
 
 }
diff --git a/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java b/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
index df09b60305cbdaa6f62f0e250ae843fdfc24d68f..bc2a5a3938e58609bc5f4f23dc50b90eb3a47026 100644
--- a/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
+++ b/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
@@ -56,6 +56,15 @@ public class MessagesFile {
             if(!config.contains("buildworldNotSet")) {
                 config.set("buildworldNotSet", "&cThe buildworld is not set");
             }
+            if (!config.contains("maintenanceKick")) {
+                config.set("maintenanceKick", "&cThe server is currently in maintenance mode");
+            }
+            if (!config.contains("maintenanceOn")) {
+                config.set("maintenanceOn", "&aYou turned on the maintenance mode");
+            }
+            if (!config.contains("maintenanceOff")) {
+                config.set("maintenanceOff", "&aYou turned off the maintenance mode");
+            }
             save();
         } catch (IOException | InvalidConfigurationException e) {
             e.printStackTrace();
@@ -110,5 +119,17 @@ public class MessagesFile {
         String prefix = config.getString("buildworldNotSet");
         return prefix.replace("&", "§");
     }
+    public static String getMaintenanceKick() {
+        String prefix = config.getString("maintenanceKick");
+        return prefix.replace("&", "§");
+    }
+    public static String getMaintenanceOn() {
+        String prefix = config.getString("maintenanceOn");
+        return prefix.replace("&", "§");
+    }
+    public static String getMaintenanceOff() {
+        String prefix = config.getString("maintenanceOff");
+        return prefix.replace("&", "§");
+    }
 
 }
diff --git a/src/main/java/net/pascalpex/deepslatemc/commands/MaintenanceMode.java b/src/main/java/net/pascalpex/deepslatemc/commands/MaintenanceMode.java
new file mode 100644
index 0000000000000000000000000000000000000000..213d709d0c0ff430420f6ee885f5ea6e4a94e837
--- /dev/null
+++ b/src/main/java/net/pascalpex/deepslatemc/commands/MaintenanceMode.java
@@ -0,0 +1,43 @@
+package net.pascalpex.deepslatemc.commands;
+
+import net.pascalpex.deepslatemc.Config;
+import net.pascalpex.deepslatemc.MessagesFile;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+
+public class MaintenanceMode extends Command {
+    public MaintenanceMode(String name) {
+        super(name);
+        this.description = "Toggles the maintenance mode";
+        this.usageMessage = "/maintenance";
+        setPermission("deepslate.maintenance");
+    }
+
+    public boolean execute(CommandSender sender, String label, String[] args) {
+        String prefix = MessagesFile.getPrefix() + " ";
+        if (sender instanceof Player) {
+            Player player = (Player)sender;
+            if (label.equalsIgnoreCase("maintenance") && player.hasPermission("deepslate.maintenance")) {
+                if (Config.getMaintenanceMode()) {
+                    player.sendMessage(prefix + MessagesFile.getMaintenanceOff());
+                } else {
+                    player.sendMessage(prefix + MessagesFile.getMaintenanceOn());
+                }
+                Config.toggleMaintenanceMode();
+            } else {
+                player.sendMessage(prefix + MessagesFile.getNoPermissions());
+            }
+        }
+        else if (label.equalsIgnoreCase("maintenance")) {
+            if (Config.getMaintenanceMode()) {
+                sender.sendMessage(prefix + MessagesFile.getMaintenanceOff());
+            } else {
+                sender.sendMessage(prefix + MessagesFile.getMaintenanceOn());
+            }
+            Config.toggleMaintenanceMode();
+        }
+
+        return true;
+    }
+}
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 8fde452b6b0785b12811bb2a8858a1fa5c98c4b5..eaf6042264ae51f77621589693270da147b550bb 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -1099,6 +1099,7 @@ public final class CraftServer implements Server {
         MinecraftServer.getServer().server.getCommandMap().register("clearchat", "Deepslate", new ClearchatCommand("clearchat"));
         MinecraftServer.getServer().server.getCommandMap().register("setbuildworld", "Deepslate", new BuildworldCommand("setbuildworld"));
         MinecraftServer.getServer().server.getCommandMap().register("buildworld", "Deepslate", new BuildworldCommand("buildworld"));
+        MinecraftServer.getServer().server.getCommandMap().register("maintenance", "Deepslate", new MaintenanceMode("maintenance"));
         // Deepslate end
 
         this.overrideAllCommandBlockCommands = this.commandsConfiguration.getStringList("command-block-overrides").contains("*");
