From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Mon, 25 Dec 2023 15:20:22 +0100
Subject: [PATCH] Maintenance mode


diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
index 33ae466bf8598ffb013a53f9ba0bc6f041c7f988..115b128d60de5c9f8cf41774f5ad13f80184cc50 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
@@ -251,6 +251,7 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
         MinecraftServer.getServer().server.getCommandMap().register("clearchat", "Deepslate", new ClearchatCommand("clearchat"));
         MinecraftServer.getServer().server.getCommandMap().register("setbuildworld", "Deepslate", new BuildworldCommand("setbuildworld"));
         MinecraftServer.getServer().server.getCommandMap().register("buildworld", "Deepslate", new BuildworldCommand("buildworld"));
+        MinecraftServer.getServer().server.getCommandMap().register("maintenance", "Deepslate", new MaintenanceMode("maintenance"));
         // Deepslate end
 
         this.setPvpAllowed(dedicatedserverproperties.pvp);
diff --git a/src/main/java/net/minecraft/server/players/PlayerList.java b/src/main/java/net/minecraft/server/players/PlayerList.java
index 25c7f47382f8a9949e2785c36f67dcf455d33f5a..eb8a18c4ebc5522ac867415528490bd37b808b33 100644
--- a/src/main/java/net/minecraft/server/players/PlayerList.java
+++ b/src/main/java/net/minecraft/server/players/PlayerList.java
@@ -103,6 +103,8 @@ import net.minecraft.world.phys.Vec3;
 import net.minecraft.world.scores.DisplaySlot;
 import net.minecraft.world.scores.Objective;
 import net.minecraft.world.scores.PlayerTeam;
+import net.pascalpex.deepslatemc.Config;
+import net.pascalpex.deepslatemc.MessagesFile;
 import org.galemc.gale.configuration.GaleGlobalConfiguration;
 import org.slf4j.Logger;
 
@@ -355,6 +357,15 @@ public abstract class PlayerList {
         player.containerMenu.transferTo(player.containerMenu, bukkitPlayer);
 
         PlayerJoinEvent playerJoinEvent = new PlayerJoinEvent(bukkitPlayer, io.papermc.paper.adventure.PaperAdventure.asAdventure(ichatmutablecomponent)); // Paper - Adventure
+        // Deepslate start - Maintenance mode
+        if(Config.getMaintenanceMode()) {
+            if(!bukkitPlayer.hasPermission("deepslate.maintenancebypass")) {
+                String prefix = MessagesFile.getPrefix();
+                bukkitPlayer.kickPlayer(prefix + MessagesFile.getMaintenanceKick());
+                return;
+            }
+        }
+        // Deepslate end
         this.cserver.getPluginManager().callEvent(playerJoinEvent);
 
         if (!player.connection.isAcceptingMessages()) {
diff --git a/src/main/java/net/pascalpex/deepslatemc/Config.java b/src/main/java/net/pascalpex/deepslatemc/Config.java
index 14630aad16fe8ac38030674c52bc91d619e9abde..97119bdc69d79216241bf1cf0c0b2ae5b230134b 100644
--- a/src/main/java/net/pascalpex/deepslatemc/Config.java
+++ b/src/main/java/net/pascalpex/deepslatemc/Config.java
@@ -30,6 +30,9 @@ public class Config {
             if(!config.contains("discordLink")) {
                 config.set("discordLink", "https://discord.gg/BGrhNnVczp");
             }
+            if(!config.contains("maintenance")) {
+                config.set("maintenance", false);
+            }
             save();
         } catch (IOException | InvalidConfigurationException e) {
             e.printStackTrace();
@@ -54,5 +57,12 @@ public class Config {
         config.set("buildworld", worldName);
         save();
     }
+    public static boolean getMaintenanceMode() {
+        return config.getBoolean("maintenance");
+    }
+    public static void toggleMaintenanceMode() {
+        config.set("maintenance", !getMaintenanceMode());
+        save();
+    }
 
 }
diff --git a/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java b/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
index b9a05e9efa2db4d7a9b93e2df3394cd8351c844c..b17adb308859c5c9fe0958d56d0cb31286c72774 100644
--- a/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
+++ b/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
@@ -56,6 +56,15 @@ public class MessagesFile {
             if(!config.contains("buildworldNotSet")) {
                 config.set("buildworldNotSet", "&cThe buildworld is not set");
             }
+            if (!config.contains("maintenanceKick")) {
+                config.set("maintenanceKick", "&cThe server is currently in maintenance mode");
+            }
+            if (!config.contains("maintenanceOn")) {
+                config.set("maintenanceOn", "&aYou turned on the maintenance mode");
+            }
+            if (!config.contains("maintenanceOff")) {
+                config.set("maintenanceOff", "&aYou turned off the maintenance mode");
+            }
             save();
         } catch (IOException | InvalidConfigurationException e) {
             e.printStackTrace();
@@ -110,5 +119,17 @@ public class MessagesFile {
         String prefix = config.getString("buildworldNotSet");
         return prefix.replace("&", "§");
     }
+    public static String getMaintenanceKick() {
+        String prefix = config.getString("maintenanceKick");
+        return prefix.replace("&", "§");
+    }
+    public static String getMaintenanceOn() {
+        String prefix = config.getString("maintenanceOn");
+        return prefix.replace("&", "§");
+    }
+    public static String getMaintenanceOff() {
+        String prefix = config.getString("maintenanceOff");
+        return prefix.replace("&", "§");
+    }
 
 }
diff --git a/src/main/java/net/pascalpex/deepslatemc/commands/MaintenanceMode.java b/src/main/java/net/pascalpex/deepslatemc/commands/MaintenanceMode.java
new file mode 100644
index 0000000000000000000000000000000000000000..213d709d0c0ff430420f6ee885f5ea6e4a94e837
--- /dev/null
+++ b/src/main/java/net/pascalpex/deepslatemc/commands/MaintenanceMode.java
@@ -0,0 +1,43 @@
+package net.pascalpex.deepslatemc.commands;
+
+import net.pascalpex.deepslatemc.Config;
+import net.pascalpex.deepslatemc.MessagesFile;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+
+public class MaintenanceMode extends Command {
+    public MaintenanceMode(String name) {
+        super(name);
+        this.description = "Toggles the maintenance mode";
+        this.usageMessage = "/maintenance";
+        setPermission("deepslate.maintenance");
+    }
+
+    public boolean execute(CommandSender sender, String label, String[] args) {
+        String prefix = MessagesFile.getPrefix() + " ";
+        if (sender instanceof Player) {
+            Player player = (Player)sender;
+            if (label.equalsIgnoreCase("maintenance") && player.hasPermission("deepslate.maintenance")) {
+                if (Config.getMaintenanceMode()) {
+                    player.sendMessage(prefix + MessagesFile.getMaintenanceOff());
+                } else {
+                    player.sendMessage(prefix + MessagesFile.getMaintenanceOn());
+                }
+                Config.toggleMaintenanceMode();
+            } else {
+                player.sendMessage(prefix + MessagesFile.getNoPermissions());
+            }
+        }
+        else if (label.equalsIgnoreCase("maintenance")) {
+            if (Config.getMaintenanceMode()) {
+                sender.sendMessage(prefix + MessagesFile.getMaintenanceOff());
+            } else {
+                sender.sendMessage(prefix + MessagesFile.getMaintenanceOn());
+            }
+            Config.toggleMaintenanceMode();
+        }
+
+        return true;
+    }
+}
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index d162516cf51171446ba949ef509702dd38b48127..5cea77b63b797081d71d2aadc7d3748ad6e81a52 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -1116,6 +1116,7 @@ public final class CraftServer implements Server {
         MinecraftServer.getServer().server.getCommandMap().register("clearchat", "Deepslate", new ClearchatCommand("clearchat"));
         MinecraftServer.getServer().server.getCommandMap().register("setbuildworld", "Deepslate", new BuildworldCommand("setbuildworld"));
         MinecraftServer.getServer().server.getCommandMap().register("buildworld", "Deepslate", new BuildworldCommand("buildworld"));
+        MinecraftServer.getServer().server.getCommandMap().register("maintenance", "Deepslate", new MaintenanceMode("maintenance"));
         // Deepslate end
 
         this.overrideAllCommandBlockCommands = this.commandsConfiguration.getStringList("command-block-overrides").contains("*");
