From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Wed, 22 Dec 2021 09:57:58 +0100
Subject: [PATCH] more rebranding


diff --git a/src/main/java/com/destroystokyo/paper/PaperVersionFetcher.java b/src/main/java/com/destroystokyo/paper/PaperVersionFetcher.java
index 5c4a1afb10d869c4719640119a35152e3cb872b9..a0121496060a5fd9aaef4c848e6e9e53314e2d66 100644
--- a/src/main/java/com/destroystokyo/paper/PaperVersionFetcher.java
+++ b/src/main/java/com/destroystokyo/paper/PaperVersionFetcher.java
@@ -21,7 +21,7 @@ import org.sugarcanemc.sugarcane.util.versioning.BranchReader;
 
 public class PaperVersionFetcher implements VersionFetcher {
     private static final java.util.regex.Pattern VER_PATTERN = java.util.regex.Pattern.compile("^([0-9\\.]*)\\-.*R"); // R is an anchor, will always give '-R' at end
-
+    private static final String GITHUB_BRANCH_NAME = "ver/1.17.1";
     private static final String DOWNLOAD_PAGE = "https://ci.sugarcanemc.org/job/Sugarcane/";
 
     private static @Nullable String mcVer;
@@ -34,9 +34,8 @@ public class PaperVersionFetcher implements VersionFetcher {
     @Nonnull
     @Override
     public Component getVersionMessage(@Nonnull String serverVersion) {
-        String[] parts = serverVersion.substring("git-Sugarcane-".length()).split("[-\\s]"); // Airplane // Purpur // Sugarcane
-        final @Nonnull String branch = BranchReader.getBranch();
-        final Component updateMessage = getUpdateStatusMessage("SugarcaneMC/Sugarcane", branch, parts[0]); // Airplane // Purpur // Sugarcane
+        String[] parts = serverVersion.substring("git-DeepslateMC-".length()).split("[-\\s]"); // Airplane // Purpur // Sugarcane  //DeepslateMC
+        final Component updateMessage = getUpdateStatusMessage("Pascalpex/DeepslateMC", GITHUB_BRANCH_NAME, parts[0]); // Airplane // Purpur // Sugarcane
         final Component history = getHistory();
 
         return history != null ? TextComponent.ofChildren(updateMessage, Component.newline(), history) : updateMessage;
@@ -61,13 +60,13 @@ public class PaperVersionFetcher implements VersionFetcher {
     private static Component getUpdateStatusMessage(@Nonnull String repo, @Nonnull String branch, @Nonnull String versionInfo) {
         int distance;
         // Sugarcane start
-        try {
-            int jenkinsBuild = Integer.parseInt(versionInfo);
-            distance = fetchDistanceFromJenkins(jenkinsBuild, branch);
-        } catch (NumberFormatException ignored) {
+        //try {
+            //int jenkinsBuild = Integer.parseInt(versionInfo);
+            //distance = fetchDistanceFromJenkins(jenkinsBuild, branch);
+        //} catch (NumberFormatException ignored) {
             versionInfo = versionInfo.replace("\"", "");
             distance = fetchDistanceFromGitHub(repo, branch, versionInfo);
-        }
+        //}
         // Sugarcane end
 
         switch (distance) {
@@ -90,7 +89,7 @@ public class PaperVersionFetcher implements VersionFetcher {
     // Sugarcane start
     private static int fetchDistanceFromJenkins(int jenkinsBuild, @Nonnull String branch) { // Sugarcane
         try {
-            try (BufferedReader reader = Resources.asCharSource(new URL("https://ci.sugarcanemc.org/job/Sugarcane/job/" + URLEncoder.encode(branch, Charsets.UTF_8.name()) + "/lastStableBuild/buildNumber"), Charsets.UTF_8).openBufferedStream()) { // Sugarcane
+            try (BufferedReader reader = Resources.asCharSource(new URL("" + URLEncoder.encode(branch, Charsets.UTF_8.name()) + "/lastStableBuild/buildNumber"), Charsets.UTF_8).openBufferedStream()) { // Sugarcane
                 return Integer.decode(reader.readLine()) - jenkinsBuild;
             } catch (NumberFormatException ex) {
                 ex.printStackTrace();
diff --git a/src/main/java/org/bukkit/craftbukkit/util/Versioning.java b/src/main/java/org/bukkit/craftbukkit/util/Versioning.java
index ee60dca1025eec079d34c49b3c72f8d9abe32c54..698b95af2fa226e15080e2572e9f0f8c4499d6f9 100644
--- a/src/main/java/org/bukkit/craftbukkit/util/Versioning.java
+++ b/src/main/java/org/bukkit/craftbukkit/util/Versioning.java
@@ -11,7 +11,7 @@ public final class Versioning {
     public static String getBukkitVersion() {
         String result = "Unknown-Version";
 
-        InputStream stream = Bukkit.class.getClassLoader().getResourceAsStream("META-INF/maven/org.sugarcanemc.sugarcane/sugarcane-api/pom.properties"); // Airplane // Sugarcane
+        InputStream stream = Bukkit.class.getClassLoader().getResourceAsStream("META-INF/maven/net.pascalpex.deepslatemc/deepslatemc-api/pom.properties"); // Airplane // Sugarcane
         Properties properties = new Properties();
 
         if (stream != null) {
