From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Mon, 21 Mar 2022 15:24:16 +0100
Subject: [PATCH] Toggleable op command


diff --git a/src/main/java/net/minecraft/commands/Commands.java b/src/main/java/net/minecraft/commands/Commands.java
index 91a8f21481de9a92192b10eca202ebb8a28e0e3d..905d1dffa92d78448619dbf3ec3df70927ffcf8c 100644
--- a/src/main/java/net/minecraft/commands/Commands.java
+++ b/src/main/java/net/minecraft/commands/Commands.java
@@ -204,7 +204,9 @@ public class Commands {
             BanListCommands.register(this.dispatcher);
             BanPlayerCommands.register(this.dispatcher);
             DeOpCommands.register(this.dispatcher);
-            OpCommand.register(this.dispatcher);
+            if (Config.getOpActive()) { // Deepslate - Toggleable op command
+                OpCommand.register(this.dispatcher);
+            }
             PardonCommand.register(this.dispatcher);
             PardonIpCommand.register(this.dispatcher);
             PerfCommand.register(this.dispatcher);
diff --git a/src/main/java/net/pascalpex/deepslatemc/Config.java b/src/main/java/net/pascalpex/deepslatemc/Config.java
index 32c369a09346b40d0a683291badfe0addcde9897..cf33202b996839c11bb52e5550b3e588e5b2dc2e 100644
--- a/src/main/java/net/pascalpex/deepslatemc/Config.java
+++ b/src/main/java/net/pascalpex/deepslatemc/Config.java
@@ -29,6 +29,9 @@ public class Config {
                 if(!config.contains("discordLink")) {
                     config.set("discordLink", "https://google.de/");
                 }
+                if(!config.contains("opCommandActive")) {
+                    config.set("opCommandActive", true);
+                }
                 save();
         } catch (IOException | InvalidConfigurationException e) {
             e.printStackTrace();
@@ -43,6 +46,10 @@ public class Config {
         }
     }
 
+        public static boolean getOpActive() {
+            return config.getBoolean("opCommandActive");
+        }
+
         public static String getDiscordLink() {
             return config.getString("discordLink");
         }
