From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Thu, 14 Jul 2022 14:47:01 +0200
Subject: [PATCH] Configurable help command


diff --git a/src/main/java/net/minecraft/commands/Commands.java b/src/main/java/net/minecraft/commands/Commands.java
index 905d1dffa92d78448619dbf3ec3df70927ffcf8c..d469a90940a90f26c5a7263bc0658449a7b79304 100644
--- a/src/main/java/net/minecraft/commands/Commands.java
+++ b/src/main/java/net/minecraft/commands/Commands.java
@@ -157,7 +157,7 @@ public class Commands {
         GameModeCommand.register(this.dispatcher);
         GameRuleCommand.register(this.dispatcher);
         GiveCommand.register(this.dispatcher, commandRegistryAccess);
-        HelpCommand.register(this.dispatcher);
+        //HelpCommand.register(this.dispatcher); Deepslate - configurable help command
         ItemCommands.register(this.dispatcher, commandRegistryAccess);
         KickCommand.register(this.dispatcher);
         KillCommand.register(this.dispatcher);
diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
index 2dd98edae229ce95778952eef6866f697892396e..f9ea91f8dae40dfa1ed5a84b7776b34b1fb465e5 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
@@ -251,6 +251,8 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
         MinecraftServer.getServer().server.getCommandMap().register("buildworld", "Deepslate", new BuildworldCommand("buildworld"));
         MinecraftServer.getServer().server.getCommandMap().register("maintenance", "Deepslate", new MaintenanceMode("maintenance"));
         MinecraftServer.getServer().server.getCommandMap().register("wartung", "Deepslate", new MaintenanceMode("wartung"));
+        MinecraftServer.getServer().server.getCommandMap().register("help", "Deepslate", new HelpCommand("help"));
+        MinecraftServer.getServer().server.getCommandMap().register("hilfe", "Deepslate", new HelpCommand("hilfe"));
         // Deepslate end
 
         this.setPvpAllowed(dedicatedserverproperties.pvp);
diff --git a/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java b/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
index 63c3a4dc8ee2e9418fa6bbc85c65ab2477445735..7528c45a8fd24c61336e8773ab51e9886670064c 100644
--- a/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
+++ b/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
@@ -65,6 +65,14 @@ public class MessagesFile {
                 if (!config.contains("maintenanceOff")) {
                     config.set("maintenanceOff", "&aYou turned off the maintenance mode");
                 }
+            if (!config.contains("helpMessage")) {
+                List<String> message = new ArrayList<String>();
+                message.add("&2&l--Help page--");
+                message.add("&6You can find the most important commands at");
+                message.add("&9&l/warp &9&lInfo&r&6. If you have more");
+                message.add("&6questions, please ask a &9team member.");
+                config.set("helpMessage", message);
+            }
             save();
         } catch (IOException | InvalidConfigurationException e) {
             e.printStackTrace();
@@ -79,6 +87,19 @@ public class MessagesFile {
         }
     }
 
+    public static String getHelpMessage() {
+        List<String> messages = (List<String>) config.getList("helpMessage");
+        String message = "";
+        for(int i = 0; i < messages.size(); i++) {
+            message = message + messages.get(i).replace("&", "§");
+            if(i < messages.size() - 1) {
+                message = message + "\n§r";
+            }
+        }
+
+        return message;
+    }
+
     public static String getPrefix() {
         String prefix = config.getString("prefix");
         return prefix.replace("&", "§");
diff --git a/src/main/java/net/pascalpex/deepslatemc/commands/HelpCommand.java b/src/main/java/net/pascalpex/deepslatemc/commands/HelpCommand.java
new file mode 100644
index 0000000000000000000000000000000000000000..911e23651037de54edaf4084686e6097da528dcf
--- /dev/null
+++ b/src/main/java/net/pascalpex/deepslatemc/commands/HelpCommand.java
@@ -0,0 +1,24 @@
+package net.pascalpex.deepslatemc.commands;
+    
+import net.pascalpex.deepslatemc.MessagesFile;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.jetbrains.annotations.NotNull;
+
+public class HelpCommand extends Command {
+
+    public HelpCommand(@NotNull String name) {
+        super(name);
+        this.description = "Shows the configured help page";
+        this.usageMessage = "/help";
+    }
+
+    @Override
+    public boolean execute(@NotNull CommandSender sender, @NotNull String label, @NotNull String[] args) {
+        if (label.equalsIgnoreCase("help") || label.equalsIgnoreCase("hilfe")) {
+            sender.sendMessage(MessagesFile.getHelpMessage());
+        }
+        return true;
+    }
+
+}
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 20a504fc2fe52349d2bf7dad8c9f974d36676155..4e3a312eed145f54b56b20d0ba40e4d5605b61b8 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -1037,6 +1037,8 @@ public final class CraftServer implements Server {
         MinecraftServer.getServer().server.getCommandMap().register("buildworld", "Deepslate", new BuildworldCommand("buildworld"));
         MinecraftServer.getServer().server.getCommandMap().register("maintenance", "Deepslate", new MaintenanceMode("maintenance"));
         MinecraftServer.getServer().server.getCommandMap().register("wartung", "Deepslate", new MaintenanceMode("wartung"));
+        MinecraftServer.getServer().server.getCommandMap().register("help", "Deepslate", new HelpCommand("help"));
+        MinecraftServer.getServer().server.getCommandMap().register("hilfe", "Deepslate", new HelpCommand("hilfe"));
         // Deepslate end
 
         this.overrideAllCommandBlockCommands = this.commandsConfiguration.getStringList("command-block-overrides").contains("*");
