From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Wed, 22 Dec 2021 10:17:14 +0100
Subject: [PATCH] Better configurable prefix


diff --git a/src/main/java/net/pascalpex/deepslatemc/Config.java b/src/main/java/net/pascalpex/deepslatemc/Config.java
index b6be04bf7a594652da371d24eb6a5f762b59b5f6..532a0f027e6e56a3805f9987e20c88c665ab81f3 100644
--- a/src/main/java/net/pascalpex/deepslatemc/Config.java
+++ b/src/main/java/net/pascalpex/deepslatemc/Config.java
@@ -20,7 +20,7 @@ public class Config {
             configFile.getParentFile().mkdirs();
             if (!configFile.exists()) {
                 configFile.createNewFile();
-                config.set("prefix", ChatColor.BLACK + "[" + ChatColor.DARK_GRAY + "DeepslateMC" + ChatColor.BLACK + "]");
+                config.set("prefix", "&0[&8DeepslateMC&0]");
                 config.set("discordLink", "https://google.de/");
                 config.set("opCommandActive", true);
                 save();
@@ -46,7 +46,8 @@ public class Config {
     }
 
     public static String getPrefix() {
-        return config.getString("prefix");
+        String prefix = config.getString("prefix");
+        return prefix.replace("&", "§");
     }
 
     public static String getDiscordLink() {
diff --git a/src/main/java/net/pascalpex/deepslatemc/commands/BauweltCommand.java b/src/main/java/net/pascalpex/deepslatemc/commands/BauweltCommand.java
index 58144727ad8e134ba69ef6b4624aeeb013076d61..22a1f236248c9bb3033643dca9ec9494495e3438 100644
--- a/src/main/java/net/pascalpex/deepslatemc/commands/BauweltCommand.java
+++ b/src/main/java/net/pascalpex/deepslatemc/commands/BauweltCommand.java
@@ -20,27 +20,28 @@ public class BauweltCommand extends Command {
 
     @Override
     public boolean execute(@NotNull CommandSender sender, @NotNull String label, @NotNull String[] args) {
+        String prefix = Config.getPrefix() + " " + ChatColor.DARK_AQUA;
         if (sender instanceof Player) {
             Player player = (Player) sender;
             if(label.equalsIgnoreCase("setbauwelt")) {
                 if(player.hasPermission("deepslate.setbauwelt")) {
                     World world = player.getLocation().getWorld();
                     Config.setBauwelt(world.getName());
-                    player.sendMessage(ChatColor.GREEN + "Bauwelt gesetzt");
+                    player.sendMessage(prefix + ChatColor.GREEN + "Bauwelt gesetzt");
                 }
             }
             if(label.equalsIgnoreCase("bauwelt")) {
                 if(player.hasPermission("deepslate.bauwelt")) {
                     try {
                         player.teleport(Bukkit.getWorld(Config.getBauwelt()).getSpawnLocation());
-                        player.sendMessage(ChatColor.GREEN + "Wilkommen in der Bauwelt");
+                        player.sendMessage(prefix + ChatColor.GREEN + "Wilkommen in der Bauwelt");
                     } catch (Exception e) {
-                        player.sendMessage(ChatColor.RED + "Die Bauwelt wurde nicht gesetzt");
+                        player.sendMessage(prefix + ChatColor.RED + "Die Bauwelt wurde nicht gesetzt");
                     }
                 }
             }
         } else {
-            sender.sendMessage("Dieser Befehl ist nur für Spieler geeignet");
+            sender.sendMessage(prefix + "Dieser Befehl ist nur für Spieler geeignet");
         }
         return true;
     }
diff --git a/src/main/java/net/pascalpex/deepslatemc/commands/ClearchatCommand.java b/src/main/java/net/pascalpex/deepslatemc/commands/ClearchatCommand.java
index 49be5e430d713f4ec8cf1e1b25c1de5215f322b7..d6408e6e2ca3b1c7aaca21159f30b218f754637f 100644
--- a/src/main/java/net/pascalpex/deepslatemc/commands/ClearchatCommand.java
+++ b/src/main/java/net/pascalpex/deepslatemc/commands/ClearchatCommand.java
@@ -1,5 +1,6 @@
 package net.pascalpex.deepslatemc.commands;
 
+import net.pascalpex.deepslatemc.Config;
 import org.bukkit.Bukkit;
 import org.bukkit.ChatColor;
 import org.bukkit.command.Command;
@@ -17,6 +18,7 @@ public class ClearchatCommand extends Command {
 
     @Override
     public boolean execute(@NotNull CommandSender sender, @NotNull String commandLabel, @NotNull String[] args) {
+        String prefix = Config.getPrefix() + " " + ChatColor.DARK_AQUA;
         if (sender instanceof Player) {
             Player player = (Player) sender;
             if (commandLabel.equalsIgnoreCase("cc") || commandLabel.equalsIgnoreCase("clearchat")) {
@@ -27,10 +29,10 @@ public class ClearchatCommand extends Command {
                     Bukkit.broadcastMessage(ChatColor.YELLOW + "Der Chat wurde von " + ChatColor.RED + player.getName() + ChatColor.YELLOW + " geleert");
                 }
             } else {
-                player.sendMessage(ChatColor.RED + "Dafür hast du keine Rechte");
+                player.sendMessage(prefix + ChatColor.RED + "Dafür hast du keine Rechte");
             }
         } else {
-            sender.sendMessage("Dieser Befehl ist nur für Spieler geeignet");
+            sender.sendMessage(prefix + "Dieser Befehl ist nur für Spieler geeignet");
         }
         return true;
     }
diff --git a/src/main/java/net/pascalpex/deepslatemc/commands/DiscordCommand.java b/src/main/java/net/pascalpex/deepslatemc/commands/DiscordCommand.java
index 8ab1afab8817a1cbe089df79586e1c278bdf9e1b..3575ebffe92db859797076072bc02d572690962f 100644
--- a/src/main/java/net/pascalpex/deepslatemc/commands/DiscordCommand.java
+++ b/src/main/java/net/pascalpex/deepslatemc/commands/DiscordCommand.java
@@ -18,11 +18,12 @@ public class DiscordCommand extends Command {
 
     @Override
     public boolean execute(@NotNull CommandSender sender, @NotNull String commandLabel, @NotNull String[] args) {
+        String prefix = Config.getPrefix() + " " + ChatColor.DARK_AQUA;
         if (commandLabel.equalsIgnoreCase("dc") || commandLabel.equalsIgnoreCase("discord")) {
             if (sender instanceof Player) {
                 Player player = (Player) sender;
                 String link = Config.getDiscordLink();
-                player.sendMessage(ChatColor.DARK_GRAY + "[" + ChatColor.GREEN + "Pascalpex" + ChatColor.DARK_GRAY + "]" + ChatColor.GOLD + " Der Link zu unserem Discord:");
+                player.sendMessage(prefix + ChatColor.GOLD + "Der Link zu unserem Discord:");
                 player.sendMessage(ChatColor.GOLD + link);
             } else {
                 sender.sendMessage("Dieser Befehl ist nur für Spieler geeignet");
diff --git a/src/main/java/net/pascalpex/deepslatemc/commands/WartungCommand.java b/src/main/java/net/pascalpex/deepslatemc/commands/WartungCommand.java
index 2580dc98414b6b6113d00f44dadcaf7ef87e50f5..8111e877e7e224a76753d2659eda710a9379959d 100644
--- a/src/main/java/net/pascalpex/deepslatemc/commands/WartungCommand.java
+++ b/src/main/java/net/pascalpex/deepslatemc/commands/WartungCommand.java
@@ -20,8 +20,9 @@ public class WartungCommand extends Command {
 
     public static void playerJoin(Player player) {
         if(!player.hasPermission("deepslate.wartungbypass")) {
+            String prefix = Config.getPrefix() + " " + ChatColor.DARK_AQUA;
             if(wartungMode) {
-                player.kickPlayer(ChatColor.DARK_GRAY + "[" + ChatColor.GREEN + "Pascalpex" + ChatColor.DARK_GRAY + "]" + ChatColor.GOLD + " Der Server ist aktuell im Wartungsmodus");
+                player.kickPlayer(prefix + ChatColor.GOLD + " Der Server ist aktuell im Wartungsmodus");
             }
         }
     }
