From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Sat, 3 Dec 2022 14:02:35 +0100
Subject: [PATCH] Configurable help command


diff --git a/src/main/java/net/minecraft/commands/Commands.java b/src/main/java/net/minecraft/commands/Commands.java
index 53a3ecc7719c5c0876f176342d50373159435b3f..5465b865549f6abb98e9452a39425ec612facbbb 100644
--- a/src/main/java/net/minecraft/commands/Commands.java
+++ b/src/main/java/net/minecraft/commands/Commands.java
@@ -183,7 +183,7 @@ public class Commands {
         GameModeCommand.register(this.dispatcher);
         GameRuleCommand.register(this.dispatcher);
         GiveCommand.register(this.dispatcher, commandRegistryAccess);
-        HelpCommand.register(this.dispatcher);
+        //HelpCommand.register(this.dispatcher); Deepslate - configurable help command
         ItemCommands.register(this.dispatcher, commandRegistryAccess);
         KickCommand.register(this.dispatcher);
         KillCommand.register(this.dispatcher);
diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
index 7c7258d17c36c56cf7ebb42c5dafe0a3428cd2c4..bc4e503c9cf86bff80e8297398cfd26898944277 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
@@ -256,6 +256,7 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
         MinecraftServer.getServer().server.getCommandMap().register("setbuildworld", "Deepslate", new BuildworldCommand("setbuildworld"));
         MinecraftServer.getServer().server.getCommandMap().register("buildworld", "Deepslate", new BuildworldCommand("buildworld"));
         MinecraftServer.getServer().server.getCommandMap().register("maintenance", "Deepslate", new MaintenanceMode("maintenance"));
+        MinecraftServer.getServer().server.getCommandMap().register("help", "Deepslate", new HelpCommand("help"));
         // Deepslate end
 
         this.setPvpAllowed(dedicatedserverproperties.pvp);
diff --git a/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java b/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
index b17adb308859c5c9fe0958d56d0cb31286c72774..46f1943319afe84289c79c4a1a3da67e6bb6077b 100644
--- a/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
+++ b/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
@@ -65,6 +65,14 @@ public class MessagesFile {
             if (!config.contains("maintenanceOff")) {
                 config.set("maintenanceOff", "&aYou turned off the maintenance mode");
             }
+            if (!config.contains("helpMessage")) {
+                List<String> message = new ArrayList<String>();
+                message.add("&2&l--Help page--");
+                message.add("&6You can find the most important commands at");
+                message.add("&9&l/warp &9&lInfo&r&6. If you have more");
+                message.add("&6questions, please ask a &9team member.");
+                config.set("helpMessage", message);
+            }
             save();
         } catch (IOException | InvalidConfigurationException e) {
             e.printStackTrace();
@@ -131,5 +139,16 @@ public class MessagesFile {
         String prefix = config.getString("maintenanceOff");
         return prefix.replace("&", "§");
     }
+    public static String getHelpMessage() {
+        List<String> messages = (List<String>) config.getList("helpMessage");
+        String message = "";
+        for(int i = 0; i < messages.size(); i++) {
+            message = message + messages.get(i).replace("&", "§");
+            if(i < messages.size() - 1) {
+                message = message + "\n§r";
+            }
+        }
+        return message;
+    }
 
 }
diff --git a/src/main/java/net/pascalpex/deepslatemc/commands/HelpCommand.java b/src/main/java/net/pascalpex/deepslatemc/commands/HelpCommand.java
new file mode 100644
index 0000000000000000000000000000000000000000..18ff82e04111931e1d878abbf340d272e4ccbd61
--- /dev/null
+++ b/src/main/java/net/pascalpex/deepslatemc/commands/HelpCommand.java
@@ -0,0 +1,23 @@
+package net.pascalpex.deepslatemc.commands;
+    
+import net.pascalpex.deepslatemc.MessagesFile;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+
+public class HelpCommand extends Command {
+
+    public HelpCommand(String name) {
+        super(name);
+        this.description = "Shows the configured help page";
+        this.usageMessage = "/help";
+    }
+
+    @Override
+    public boolean execute(CommandSender sender, String label, String[] args) {
+        if (label.equalsIgnoreCase("help")) {
+            sender.sendMessage(MessagesFile.getHelpMessage());
+        }
+        return true;
+    }
+
+}
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 1d1d04ed6846b7d527eac4c247a6a565ccfd52d1..bc1beada5115ab5e729297520672dc150dc0bfd5 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -1108,6 +1108,7 @@ public final class CraftServer implements Server {
         MinecraftServer.getServer().server.getCommandMap().register("setbuildworld", "Deepslate", new BuildworldCommand("setbuildworld"));
         MinecraftServer.getServer().server.getCommandMap().register("buildworld", "Deepslate", new BuildworldCommand("buildworld"));
         MinecraftServer.getServer().server.getCommandMap().register("maintenance", "Deepslate", new MaintenanceMode("maintenance"));
+        MinecraftServer.getServer().server.getCommandMap().register("help", "Deepslate", new HelpCommand("help"));
         // Deepslate end
 
         this.overrideAllCommandBlockCommands = this.commandsConfiguration.getStringList("command-block-overrides").contains("*");
