From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Sun, 30 Jan 2022 14:35:50 +0100
Subject: [PATCH] Configurable help command


diff --git a/src/main/java/net/minecraft/commands/Commands.java b/src/main/java/net/minecraft/commands/Commands.java
index a888ea8a4dea848c439959a640afec8bb2ba45ea..d2e745061369f09b0518cbc2a603d6d3c693c51c 100644
--- a/src/main/java/net/minecraft/commands/Commands.java
+++ b/src/main/java/net/minecraft/commands/Commands.java
@@ -149,7 +149,7 @@ public class Commands {
         GameModeCommand.register(this.dispatcher);
         GameRuleCommand.register(this.dispatcher);
         GiveCommand.register(this.dispatcher);
-        HelpCommand.register(this.dispatcher);
+        //HelpCommand.register(this.dispatcher);
         ItemCommands.register(this.dispatcher);
         KickCommand.register(this.dispatcher);
         KillCommand.register(this.dispatcher);
diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
index 8ec1aca8180221a8b16ecffdfe241a7e3b8fa62e..aa27acb098b85bb0c7a1341277e536f7e2621183 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
@@ -265,6 +265,8 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
         MinecraftServer.getServer().server.getCommandMap().register("setbauwelt", "Deepslate", new BauweltCommand("setbauwelt"));
         MinecraftServer.getServer().server.getCommandMap().register("bauwelt", "Deepslate", new BauweltCommand("bauwelt"));
         MinecraftServer.getServer().server.getCommandMap().register("wartung", "Deepslate", new WartungCommand("wartung"));
+        MinecraftServer.getServer().server.getCommandMap().register("help", "Deepslate", new HelpCommand("help"));
+        MinecraftServer.getServer().server.getCommandMap().register("hilfe", "Deepslate", new HelpCommand("hilfe"));
         //Deepslate end
 
         this.setPvpAllowed(dedicatedserverproperties.pvp);
diff --git a/src/main/java/net/pascalpex/deepslatemc/Config.java b/src/main/java/net/pascalpex/deepslatemc/Config.java
index e2a657722ce38e83ae4dc5eeafe27457307f1e55..e26831d8d26081c95e9b5452057b35bfc186f13b 100644
--- a/src/main/java/net/pascalpex/deepslatemc/Config.java
+++ b/src/main/java/net/pascalpex/deepslatemc/Config.java
@@ -9,6 +9,7 @@ import org.bukkit.inventory.ItemStack;
 import java.io.File;
 import java.io.IOException;
 import java.util.ArrayList;
+import java.util.List;
 
 public class Config {
 
@@ -23,6 +24,12 @@ public class Config {
                 config.set("prefix", "&0[&8DeepslateMC&0]");
                 config.set("discordLink", "https://google.de/");
                 config.set("opCommandActive", true);
+                List<String> message = new ArrayList<String>();
+                message.add("&2&l--Hilfeseite--");
+                message.add("&6Die wichtigsten Befehle kannst du bei");
+                message.add("&9&l/warp &9&lInfos &r&6nachlesen. Wende dich bei");
+                message.add("&6weiteren Fragen an ein &9Teammitglied.");
+                config.set("helpMessage", message);
                 save();
             }
             config.load(configFile);
@@ -43,6 +50,19 @@ public class Config {
         return config.getBoolean("opCommandActive");
     }
 
+    public static String getHelpMessage() {
+        List<String> messages = (List<String>) config.getList("helpMessage");
+        String message = "";
+        for(int i = 0; i < messages.size(); i++) {
+            message = message + messages.get(i).replace("&", "§");
+            if(i < messages.size() - 1) {
+                message = message + "\n§r";
+            }
+        }
+
+        return message;
+    }
+
     public static String getPrefix() {
         String prefix = config.getString("prefix");
         return prefix.replace("&", "§");
diff --git a/src/main/java/net/pascalpex/deepslatemc/commands/HelpCommand.java b/src/main/java/net/pascalpex/deepslatemc/commands/HelpCommand.java
new file mode 100644
index 0000000000000000000000000000000000000000..e5e8edc0f6e506d151c622b966533712f34edba6
--- /dev/null
+++ b/src/main/java/net/pascalpex/deepslatemc/commands/HelpCommand.java
@@ -0,0 +1,24 @@
+package net.pascalpex.deepslatemc.commands;
+    
+import net.pascalpex.deepslatemc.Config;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.jetbrains.annotations.NotNull;
+
+public class HelpCommand extends Command {
+
+    public HelpCommand(@NotNull String name) {
+        super(name);
+        this.description = "Zeigt die konfigurierte Hilfe an";
+        this.usageMessage = "/help";
+    }
+
+    @Override
+    public boolean execute(@NotNull CommandSender sender, @NotNull String label, @NotNull String[] args) {
+        if (label.equalsIgnoreCase("help") || label.equalsIgnoreCase("hilfe")) {
+            sender.sendMessage(Config.getHelpMessage());
+        }
+        return true;
+    }
+
+}
