From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Thu, 14 Jul 2022 14:51:50 +0200
Subject: [PATCH] Spawn functions


diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
index 7f1bf370f7b377f704368462baf2be83e29c942d..e56899e369f082241d5f42a4e15b1e6714e26da1 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
@@ -253,6 +253,13 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
         MinecraftServer.getServer().server.getCommandMap().register("wartung", "Deepslate", new MaintenanceMode("wartung"));
         MinecraftServer.getServer().server.getCommandMap().register("help", "Deepslate", new HelpCommand("help"));
         MinecraftServer.getServer().server.getCommandMap().register("hilfe", "Deepslate", new HelpCommand("hilfe"));
+
+        MinecraftServer.getServer().server.getCommandMap().register("setspawn", "Deepslate", new SpawnCommand("setspawn"));
+        MinecraftServer.getServer().server.getCommandMap().register("spawn", "Deepslate", new SpawnCommand("spawn"));
+        MinecraftServer.getServer().server.getCommandMap().register("lobby", "Deepslate", new SpawnCommand("lobby"));
+        MinecraftServer.getServer().server.getCommandMap().register("l", "Deepslate", new SpawnCommand("l"));
+        MinecraftServer.getServer().server.getCommandMap().register("hub", "Deepslate", new SpawnCommand("hub"));
+        MinecraftServer.getServer().server.getCommandMap().register("hubschrauber", "Deepslate", new SpawnCommand("hubschrauber"));
         // Deepslate end
 
         this.setPvpAllowed(dedicatedserverproperties.pvp);
diff --git a/src/main/java/net/minecraft/server/players/PlayerList.java b/src/main/java/net/minecraft/server/players/PlayerList.java
index 0a08ca4d89c9de14b3e967231f6d97a88cd1379f..df84ea01364288dbdc46fd5af7acc77f514747f3 100644
--- a/src/main/java/net/minecraft/server/players/PlayerList.java
+++ b/src/main/java/net/minecraft/server/players/PlayerList.java
@@ -102,6 +102,7 @@ import net.minecraft.world.scores.Scoreboard; // Paper
 import net.minecraft.world.scores.Team;
 import net.pascalpex.deepslatemc.MessagesFile;
 import net.pascalpex.deepslatemc.commands.MaintenanceMode;
+import net.pascalpex.deepslatemc.commands.SpawnCommand;
 import org.slf4j.Logger;
 
 // CraftBukkit start
@@ -364,6 +365,7 @@ public abstract class PlayerList {
         player.containerMenu.transferTo(player.containerMenu, bukkitPlayer);
 
         PlayerJoinEvent playerJoinEvent = new PlayerJoinEvent(bukkitPlayer, PaperAdventure.asAdventure(ichatmutablecomponent)); // Paper - Adventure
+        SpawnCommand.playerJoin(bukkitPlayer); // Deepslate - Spawn functions
         this.cserver.getPluginManager().callEvent(playerJoinEvent);
 
         if (!player.connection.connection.isConnected()) {
diff --git a/src/main/java/net/pascalpex/deepslatemc/Config.java b/src/main/java/net/pascalpex/deepslatemc/Config.java
index cf33202b996839c11bb52e5550b3e588e5b2dc2e..7645e1e393108c7dd27f7b9a03deeba151ae93cc 100644
--- a/src/main/java/net/pascalpex/deepslatemc/Config.java
+++ b/src/main/java/net/pascalpex/deepslatemc/Config.java
@@ -1,5 +1,8 @@
 package net.pascalpex.deepslatemc;
     
+import org.bukkit.Bukkit;
+import org.bukkit.Location;
+import org.bukkit.World;
 import org.bukkit.configuration.InvalidConfigurationException;
 import org.bukkit.configuration.file.FileConfiguration;
 import org.bukkit.configuration.file.YamlConfiguration;
@@ -32,6 +35,9 @@ public class Config {
                 if(!config.contains("opCommandActive")) {
                     config.set("opCommandActive", true);
                 }
+            if(!config.contains("spawnOnJoin")) {
+                config.set("spawnOnJoin", false);
+            }
                 save();
         } catch (IOException | InvalidConfigurationException e) {
             e.printStackTrace();
@@ -50,6 +56,30 @@ public class Config {
             return config.getBoolean("opCommandActive");
         }
 
+    public static boolean getSpawnOnJoin() {
+        return config.getBoolean("spawnOnJoin");
+    }
+
+    public static void setSpawn(Location loc) {
+        config.set("spawn" + ".world", loc.getWorld().getName());
+        config.set("spawn" + ".X", loc.getX());
+        config.set("spawn" + ".Y", loc.getY());
+        config.set("spawn" + ".Z", loc.getZ());
+        config.set("spawn" + ".pitch",(double) loc.getPitch());
+        config.set("spawn" + ".yaw",(double) loc.getYaw());
+        save();
+    }
+
+    public static Location getSpawn() {
+        World world = Bukkit.getWorld(config.getString("spawn" + ".world"));
+        double x = config.getDouble("spawn" + ".X");
+        double y = config.getDouble("spawn" + ".Y");
+        double z = config.getDouble("spawn" + ".Z");
+        float pitch = (float) config.getDouble("spawn" + ".pitch");
+        float yaw = (float) config.getDouble("spawn" + ".yaw");
+        return new Location(world, x, y, z, yaw, pitch);
+    }
+
         public static String getDiscordLink() {
             return config.getString("discordLink");
         }
diff --git a/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java b/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
index 7528c45a8fd24c61336e8773ab51e9886670064c..f28b4e1f3f228210aca64e66c0a2e2e933377dae 100644
--- a/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
+++ b/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
@@ -73,6 +73,15 @@ public class MessagesFile {
                 message.add("&6questions, please ask a &9team member.");
                 config.set("helpMessage", message);
             }
+            if(!config.contains("spawnSet")) {
+                config.set("spawnSet", "&aThe spawn got set successfully");
+            }
+            if(!config.contains("spawnNotSet")) {
+                config.set("spawnNotSet", "&cThe spawn is not set");
+            }
+            if(!config.contains("spawnTeleport")) {
+                config.set("spawnTeleport", "&aYou got teleported to the spawn");
+            }
             save();
         } catch (IOException | InvalidConfigurationException e) {
             e.printStackTrace();
@@ -159,5 +168,17 @@ public class MessagesFile {
             String prefix = config.getString("maintenanceOff");
             return prefix.replace("&", "§");
         }
+    public static String getSpawnSet() {
+        String prefix = config.getString("spawnSet");
+        return prefix.replace("&", "§");
+    }
+    public static String getSpawnNotSet() {
+        String prefix = config.getString("spawnNotSet");
+        return prefix.replace("&", "§");
+    }
+    public static String getSpawnTeleport() {
+        String prefix = config.getString("spawnTeleport");
+        return prefix.replace("&", "§");
+    }
 
 }
diff --git a/src/main/java/net/pascalpex/deepslatemc/commands/SpawnCommand.java b/src/main/java/net/pascalpex/deepslatemc/commands/SpawnCommand.java
new file mode 100644
index 0000000000000000000000000000000000000000..f77b1929677128a9aadcb07a89ed42c091d5ae45
--- /dev/null
+++ b/src/main/java/net/pascalpex/deepslatemc/commands/SpawnCommand.java
@@ -0,0 +1,62 @@
+package net.pascalpex.deepslatemc.commands;
+    
+import net.pascalpex.deepslatemc.Config;
+import net.pascalpex.deepslatemc.MessagesFile;
+import org.bukkit.Bukkit;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+import org.jetbrains.annotations.NotNull;
+
+public class SpawnCommand extends Command {
+
+    public SpawnCommand(@NotNull String name) {
+        super(name);
+        this.description = "Teleports you to the spawn";
+        this.usageMessage = "/spawn";
+        this.setPermission("deepslate.spawn");
+    }
+
+    public static void playerJoin(Player player) {
+        if(Config.getSpawnOnJoin()) {
+            try {
+                player.teleport(Config.getSpawn());
+            } catch (Exception e) {
+                Bukkit.getLogger().warning("The option spawnOnJoin in the deepslate.yml file is turned on but there is no spawn set!");
+            }
+        }
+    }
+
+    @Override
+    public boolean execute(@NotNull CommandSender sender, @NotNull String label, @NotNull String[] args) {
+        String prefix = MessagesFile.getPrefix() + " ";
+        if(sender instanceof Player) {
+            Player player = (Player) sender;
+            if(label.equalsIgnoreCase("setspawn")) {
+                if(player.hasPermission("deepslate.setspawn")) {
+                    Config.setSpawn(player.getLocation());
+                    player.sendMessage(prefix + MessagesFile.getSpawnSet());
+                } else {
+                    player.sendMessage(prefix + MessagesFile.getNoPermissions());
+                }
+            }
+            if(label.equalsIgnoreCase("l") || label.equalsIgnoreCase("hub") || label.equalsIgnoreCase("lobby") || label.equalsIgnoreCase("spawn") || label.equalsIgnoreCase("hubschrauber")) {
+                if(player.hasPermission("deepslate.spawn")) {
+                    try {
+                        player.teleport(Config.getSpawn());
+                        player.sendMessage(prefix + MessagesFile.getSpawnTeleport());
+                    } catch (Exception e) {
+                        player.sendMessage(prefix + MessagesFile.getSpawnNotSet());
+                    }
+                } else {
+                    player.sendMessage(prefix + MessagesFile.getNoPermissions());
+                }
+            }
+
+        } else {
+            sender.sendMessage(prefix + MessagesFile.getOnlyForPlayers());
+        }
+        return true;
+    }
+
+}
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index fc6f037c37a3e567244943aa3cfb5d32a51becb3..4cb7c0a0e1c25d483a7ec8f21da1934fb0dd045d 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -1039,6 +1039,13 @@ public final class CraftServer implements Server {
         MinecraftServer.getServer().server.getCommandMap().register("wartung", "Deepslate", new MaintenanceMode("wartung"));
         MinecraftServer.getServer().server.getCommandMap().register("help", "Deepslate", new HelpCommand("help"));
         MinecraftServer.getServer().server.getCommandMap().register("hilfe", "Deepslate", new HelpCommand("hilfe"));
+
+        MinecraftServer.getServer().server.getCommandMap().register("setspawn", "Deepslate", new SpawnCommand("setspawn"));
+        MinecraftServer.getServer().server.getCommandMap().register("spawn", "Deepslate", new SpawnCommand("spawn"));
+        MinecraftServer.getServer().server.getCommandMap().register("lobby", "Deepslate", new SpawnCommand("lobby"));
+        MinecraftServer.getServer().server.getCommandMap().register("l", "Deepslate", new SpawnCommand("l"));
+        MinecraftServer.getServer().server.getCommandMap().register("hub", "Deepslate", new SpawnCommand("hub"));
+        MinecraftServer.getServer().server.getCommandMap().register("hubschrauber", "Deepslate", new SpawnCommand("hubschrauber"));
         // Deepslate end
 
         this.overrideAllCommandBlockCommands = this.commandsConfiguration.getStringList("command-block-overrides").contains("*");
