From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Sat, 3 Dec 2022 14:10:18 +0100
Subject: [PATCH] Spawn functions


diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
index aa81fec34387d0ec0ed919cccc15762b02b05f40..e843f4ab9edc46521e059fc8347e310f492feea0 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
@@ -254,6 +254,9 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
         MinecraftServer.getServer().server.getCommandMap().register("buildworld", "Deepslate", new BuildworldCommand("buildworld"));
         MinecraftServer.getServer().server.getCommandMap().register("maintenance", "Deepslate", new MaintenanceMode("maintenance"));
         MinecraftServer.getServer().server.getCommandMap().register("help", "Deepslate", new HelpCommand("help"));
+        MinecraftServer.getServer().server.getCommandMap().register("setspawn", "Deepslate", new SpawnCommand("setspawn"));
+        MinecraftServer.getServer().server.getCommandMap().register("spawn", "Deepslate", new SpawnCommand("spawn"));
+        MinecraftServer.getServer().server.getCommandMap().register("lobby", "Deepslate", new SpawnCommand("lobby"));
         // Deepslate end
 
         this.setPvpAllowed(dedicatedserverproperties.pvp);
diff --git a/src/main/java/net/minecraft/server/players/PlayerList.java b/src/main/java/net/minecraft/server/players/PlayerList.java
index de0619e618a29859c0768dafabf5726289d9576d..16f9b1af3b379f7109217b201e68ac376e8ca984 100644
--- a/src/main/java/net/minecraft/server/players/PlayerList.java
+++ b/src/main/java/net/minecraft/server/players/PlayerList.java
@@ -102,6 +102,7 @@ import net.minecraft.world.scores.Scoreboard; // Paper
 import net.minecraft.world.scores.Team;
 import net.pascalpex.deepslatemc.MessagesFile;
 import net.pascalpex.deepslatemc.commands.MaintenanceMode;
+import net.pascalpex.deepslatemc.commands.SpawnCommand;
 import org.slf4j.Logger;
 
 // CraftBukkit start
@@ -318,6 +319,7 @@ public abstract class PlayerList {
         player.containerMenu.transferTo(player.containerMenu, bukkitPlayer);
 
         PlayerJoinEvent playerJoinEvent = new PlayerJoinEvent(bukkitPlayer, PaperAdventure.asAdventure(ichatmutablecomponent)); // Paper - Adventure
+        SpawnCommand.playerJoin(bukkitPlayer); // Deepslate - Spawn functions
         this.cserver.getPluginManager().callEvent(playerJoinEvent);
 
         if (!player.connection.connection.isConnected()) {
diff --git a/src/main/java/net/pascalpex/deepslatemc/Config.java b/src/main/java/net/pascalpex/deepslatemc/Config.java
index 442fcec773f6232e9a84cfadd28a36f79cec43d9..6962584f335a75a4e23b68cc731e3d5856881e04 100644
--- a/src/main/java/net/pascalpex/deepslatemc/Config.java
+++ b/src/main/java/net/pascalpex/deepslatemc/Config.java
@@ -1,5 +1,8 @@
 package net.pascalpex.deepslatemc;
     
+import org.bukkit.Bukkit;
+import org.bukkit.Location;
+import org.bukkit.World;
 import org.bukkit.configuration.InvalidConfigurationException;
 import org.bukkit.configuration.file.FileConfiguration;
 import org.bukkit.configuration.file.YamlConfiguration;
@@ -33,6 +36,9 @@ public class Config {
             if(!config.contains("opCommandActive")) {
                 config.set("opCommandActive", true);
             }
+            if(!config.contains("spawnOnJoin")) {
+                config.set("spawnOnJoin", false);
+            }
             save();
         } catch (IOException | InvalidConfigurationException e) {
             e.printStackTrace();
@@ -60,5 +66,28 @@ public class Config {
         config.set("buildworld", worldName);
         save();
     }
+    public static boolean getSpawnOnJoin() {
+        return config.getBoolean("spawnOnJoin");
+    }
+
+    public static void setSpawn(Location loc) {
+        config.set("spawn" + ".world", loc.getWorld().getName());
+        config.set("spawn" + ".X", loc.getX());
+        config.set("spawn" + ".Y", loc.getY());
+        config.set("spawn" + ".Z", loc.getZ());
+        config.set("spawn" + ".pitch",(double) loc.getPitch());
+        config.set("spawn" + ".yaw",(double) loc.getYaw());
+        save();
+    }
+
+    public static Location getSpawn() {
+        World world = Bukkit.getWorld(config.getString("spawn" + ".world"));
+        double x = config.getDouble("spawn" + ".X");
+        double y = config.getDouble("spawn" + ".Y");
+        double z = config.getDouble("spawn" + ".Z");
+        float pitch = (float) config.getDouble("spawn" + ".pitch");
+        float yaw = (float) config.getDouble("spawn" + ".yaw");
+        return new Location(world, x, y, z, yaw, pitch);
+    }
 
 }
diff --git a/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java b/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
index a6fa4d46db7cd065fd8d49c99a621a5fe9fdab0f..155380686dcccb368820b45eab67a21bed9d85f1 100644
--- a/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
+++ b/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
@@ -73,6 +73,15 @@ public class MessagesFile {
                 message.add("&6questions, please ask a &9team member.");
                 config.set("helpMessage", message);
             }
+            if(!config.contains("spawnSet")) {
+                config.set("spawnSet", "&aThe spawn got set successfully");
+            }
+            if(!config.contains("spawnNotSet")) {
+                config.set("spawnNotSet", "&cThe spawn is not set");
+            }
+            if(!config.contains("spawnTeleport")) {
+                config.set("spawnTeleport", "&aYou got teleported to the spawn");
+            }
             save();
         } catch (IOException | InvalidConfigurationException e) {
             e.printStackTrace();
@@ -150,5 +159,17 @@ public class MessagesFile {
         }
         return message;
     }
+    public static String getSpawnSet() {
+        String prefix = config.getString("spawnSet");
+        return prefix.replace("&", "§");
+    }
+    public static String getSpawnNotSet() {
+        String prefix = config.getString("spawnNotSet");
+        return prefix.replace("&", "§");
+    }
+    public static String getSpawnTeleport() {
+        String prefix = config.getString("spawnTeleport");
+        return prefix.replace("&", "§");
+    }
 
 }
diff --git a/src/main/java/net/pascalpex/deepslatemc/commands/SpawnCommand.java b/src/main/java/net/pascalpex/deepslatemc/commands/SpawnCommand.java
new file mode 100644
index 0000000000000000000000000000000000000000..ec53a7ba37bf6360dea2c9b98049986317122518
--- /dev/null
+++ b/src/main/java/net/pascalpex/deepslatemc/commands/SpawnCommand.java
@@ -0,0 +1,60 @@
+package net.pascalpex.deepslatemc.commands;
+    
+import net.pascalpex.deepslatemc.Config;
+import net.pascalpex.deepslatemc.MessagesFile;
+import org.bukkit.Bukkit;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+
+public class SpawnCommand extends Command {
+
+    public SpawnCommand(String name) {
+        super(name);
+        this.description = "Teleports you to the spawn";
+        this.usageMessage = "/spawn";
+        this.setPermission("deepslate.spawn");
+    }
+
+    public static void playerJoin(Player player) {
+        if(Config.getSpawnOnJoin()) {
+            try {
+                player.teleport(Config.getSpawn());
+            } catch (Exception e) {
+                Bukkit.getLogger().warning("The option spawnOnJoin in the deepslate.yml file is turned on but there is no spawn set!");
+            }
+        }
+    }
+
+    @Override
+    public boolean execute(CommandSender sender, String label, String[] args) {
+        String prefix = MessagesFile.getPrefix() + " ";
+        if(sender instanceof Player) {
+            Player player = (Player) sender;
+            if(label.equalsIgnoreCase("setspawn")) {
+                if(player.hasPermission("deepslate.setspawn")) {
+                    Config.setSpawn(player.getLocation());
+                    player.sendMessage(prefix + MessagesFile.getSpawnSet());
+                } else {
+                    player.sendMessage(prefix + MessagesFile.getNoPermissions());
+                }
+            }
+            if(label.equalsIgnoreCase("lobby") || label.equalsIgnoreCase("spawn")) {
+                if(player.hasPermission("deepslate.spawn")) {
+                    try {
+                        player.teleport(Config.getSpawn());
+                        player.sendMessage(prefix + MessagesFile.getSpawnTeleport());
+                    } catch (Exception e) {
+                        player.sendMessage(prefix + MessagesFile.getSpawnNotSet());
+                    }
+                } else {
+                    player.sendMessage(prefix + MessagesFile.getNoPermissions());
+                }
+            }
+        } else {
+            sender.sendMessage(prefix + MessagesFile.getOnlyForPlayers());
+        }
+        return true;
+    }
+
+}
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 517349974282c0a5384c69b114d58d032db85674..01c8ac3bd76e611e604ed1e458623fca5277d6fb 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -1078,6 +1078,9 @@ public final class CraftServer implements Server {
         MinecraftServer.getServer().server.getCommandMap().register("buildworld", "Deepslate", new BuildworldCommand("buildworld"));
         MinecraftServer.getServer().server.getCommandMap().register("maintenance", "Deepslate", new MaintenanceMode("maintenance"));
         MinecraftServer.getServer().server.getCommandMap().register("help", "Deepslate", new HelpCommand("help"));
+        MinecraftServer.getServer().server.getCommandMap().register("setspawn", "Deepslate", new SpawnCommand("setspawn"));
+        MinecraftServer.getServer().server.getCommandMap().register("spawn", "Deepslate", new SpawnCommand("spawn"));
+        MinecraftServer.getServer().server.getCommandMap().register("lobby", "Deepslate", new SpawnCommand("lobby"));
         // Deepslate end
 
         this.overrideAllCommandBlockCommands = this.commandsConfiguration.getStringList("command-block-overrides").contains("*");
