From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Wed, 22 Dec 2021 10:23:04 +0100
Subject: [PATCH] Configurable help command


diff --git a/src/main/java/net/minecraft/commands/Commands.java b/src/main/java/net/minecraft/commands/Commands.java
index cbba8d3080117c795db5a718b7ea49d708a00d9c..09c82fb94b89633a58aac8bd8755493245141d2f 100644
--- a/src/main/java/net/minecraft/commands/Commands.java
+++ b/src/main/java/net/minecraft/commands/Commands.java
@@ -147,7 +147,7 @@ public class Commands {
         GameModeCommand.register(this.dispatcher);
         GameRuleCommand.register(this.dispatcher);
         GiveCommand.register(this.dispatcher);
-        HelpCommand.register(this.dispatcher);
+        //HelpCommand.register(this.dispatcher);
         ItemCommands.register(this.dispatcher);
         KickCommand.register(this.dispatcher);
         KillCommand.register(this.dispatcher);
diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
index ca9e4a35d48f736a2381a023fcaf9ef4b102249e..b3f3dd8a93c69a0103a012766ac5fe15d8493496 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
@@ -258,6 +258,8 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
         MinecraftServer.getServer().server.getCommandMap().register("setbauwelt", "Deepslate", new BauweltCommand("setbauwelt"));
         MinecraftServer.getServer().server.getCommandMap().register("bauwelt", "Deepslate", new BauweltCommand("bauwelt"));
         MinecraftServer.getServer().server.getCommandMap().register("wartung", "Deepslate", new WartungCommand("wartung"));
+        MinecraftServer.getServer().server.getCommandMap().register("help", "Deepslate", new HelpCommand("help"));
+        MinecraftServer.getServer().server.getCommandMap().register("hilfe", "Deepslate", new HelpCommand("hilfe"));
         // Deepslate end
 
         // Sugarcane start - config
diff --git a/src/main/java/net/pascalpex/deepslatemc/Config.java b/src/main/java/net/pascalpex/deepslatemc/Config.java
index 532a0f027e6e56a3805f9987e20c88c665ab81f3..7df4ad75f865d8c61526a21366841f1cc55b89b5 100644
--- a/src/main/java/net/pascalpex/deepslatemc/Config.java
+++ b/src/main/java/net/pascalpex/deepslatemc/Config.java
@@ -9,6 +9,7 @@ import org.bukkit.inventory.ItemStack;
 import java.io.File;
 import java.io.IOException;
 import java.util.ArrayList;
+import java.util.List;
 
 public class Config {
 
@@ -23,6 +24,12 @@ public class Config {
                 config.set("prefix", "&0[&8DeepslateMC&0]");
                 config.set("discordLink", "https://google.de/");
                 config.set("opCommandActive", true);
+                List<String> message = new ArrayList<String>();
+                message.add("&2&l--Hilfeseite--");
+                message.add("&6Die wichtigsten Befehle kannst du bei");
+                message.add("&9&l/warp &9&lInfos &r&6nachlesen. Wende dich bei");
+                message.add("&6weiteren Fragen an ein &9Teammitglied.");
+                config.set("helpMessage", message);
                 save();
             }
             config.load(configFile);
@@ -45,6 +52,19 @@ public class Config {
         return config.getBoolean("opCommandActive");
     }
 
+    public static String getHelpMessage() {
+        List<String> messages = (List<String>) config.getList("helpMessage");
+        String message = "";
+        for(int i = 0; i < messages.size(); i++) {
+            message = message + messages.get(i).replace("&", "§");
+            if(i < messages.size() - 1) {
+                message = message + "\n§r";
+                }
+            }
+
+            return message;
+        }
+
     public static String getPrefix() {
         String prefix = config.getString("prefix");
         return prefix.replace("&", "§");
diff --git a/src/main/java/net/pascalpex/deepslatemc/commands/HelpCommand.java b/src/main/java/net/pascalpex/deepslatemc/commands/HelpCommand.java
new file mode 100644
index 0000000000000000000000000000000000000000..3bc737c576508623dab3c24c955b131f25f7bd89
--- /dev/null
+++ b/src/main/java/net/pascalpex/deepslatemc/commands/HelpCommand.java
@@ -0,0 +1,24 @@
+package net.pascalpex.deepslatemc.commands;
+
+import net.pascalpex.deepslatemc.Config;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.jetbrains.annotations.NotNull;
+
+public class HelpCommand extends Command {
+
+    public HelpCommand(@NotNull String name) {
+        super(name);
+        this.description = "Zeigt die konfigurierte Hilfe an";
+        this.usageMessage = "/help";
+    }
+
+    @Override
+    public boolean execute(@NotNull CommandSender sender, @NotNull String label, @NotNull String[] args) {
+        if (label.equalsIgnoreCase("help") || label.equalsIgnoreCase("hilfe")) {
+            sender.sendMessage(Config.getHelpMessage());
+        }
+        return true;
+    }
+
+}
