From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Sat, 24 Jun 2023 16:42:31 +0200
Subject: [PATCH] Configurable knowledgebook click


diff --git a/src/main/java/net/pascalpex/deepslatemc/Config.java b/src/main/java/net/pascalpex/deepslatemc/Config.java
index 5971700571a7347dc017a541c2901ab35b1e8d5a..467c6fe3c2c01fd66e858b20f8e4c6bab501cd7c 100644
--- a/src/main/java/net/pascalpex/deepslatemc/Config.java
+++ b/src/main/java/net/pascalpex/deepslatemc/Config.java
@@ -42,6 +42,9 @@ public class Config {
             if(!config.contains("spawnOnJoin")) {
                 config.set("spawnOnJoin", false);
             }
+            if(!config.contains("preventKnowledgebookClick")) {
+                config.set("preventKnowledgebookClick", false);
+            }
             save();
         } catch (IOException | InvalidConfigurationException e) {
             e.printStackTrace();
@@ -97,5 +100,8 @@ public class Config {
         float yaw = (float) config.getDouble("spawn" + ".yaw");
         return new Location(world, x, y, z, yaw, pitch);
     }
+    public static boolean getPreventKnowledgebookClick() {
+        return config.getBoolean("preventKnowledgebookClick");
+    }
 
 }
diff --git a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
index 98891112f992978c97b21ab0d7d54228819b7536..c6b72197f50177f2fad8d8f4a4e8fe10546bceaa 100644
--- a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
+++ b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
@@ -62,6 +62,7 @@ import net.minecraft.world.phys.BlockHitResult;
 import net.minecraft.world.phys.EntityHitResult;
 import net.minecraft.world.phys.HitResult;
 import net.minecraft.world.phys.Vec3;
+import net.pascalpex.deepslatemc.Config;
 import org.bukkit.Bukkit;
 import org.bukkit.Location;
 import org.bukkit.Material;
@@ -595,6 +596,17 @@ public class CraftEventFactory {
             event.setUseItemInHand(Result.DENY);
         }
         // Paper end
+        // Deepslate start - Configurable knowledgebook click
+        if (Config.getPreventKnowledgebookClick()) {
+            if (event.hasItem()) {
+                if (event.getAction() == Action.RIGHT_CLICK_AIR || event.getAction() == Action.RIGHT_CLICK_BLOCK) {
+                    if (event.getItem().getType() == Material.KNOWLEDGE_BOOK) {
+                        event.setCancelled(true);
+                    }
+                }
+            }
+        }
+        // Deepslate end
         craftServer.getPluginManager().callEvent(event);
 
         // Purpur start
