From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Sun, 23 Jan 2022 20:29:03 +0100
Subject: [PATCH] Spawn functions


diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
index 7e2f07dfd928ce22b6f32985b689ed01536dca2d..29f7d11eb90bb4dd640673ca63e3e220d0c54a1f 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
@@ -257,6 +257,13 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
         MinecraftServer.getServer().server.getCommandMap().register("wartung", "Deepslate", new WartungCommand("wartung"));
         MinecraftServer.getServer().server.getCommandMap().register("help", "Deepslate", new HelpCommand("help"));
         MinecraftServer.getServer().server.getCommandMap().register("hilfe", "Deepslate", new HelpCommand("hilfe"));
+
+        MinecraftServer.getServer().server.getCommandMap().register("setspawn", "Deepslate", new SpawnCommand("setspawn"));
+        MinecraftServer.getServer().server.getCommandMap().register("spawn", "Deepslate", new SpawnCommand("spawn"));
+        MinecraftServer.getServer().server.getCommandMap().register("lobby", "Deepslate", new SpawnCommand("lobby"));
+        MinecraftServer.getServer().server.getCommandMap().register("l", "Deepslate", new SpawnCommand("l"));
+        MinecraftServer.getServer().server.getCommandMap().register("hub", "Deepslate", new SpawnCommand("hub"));
+        MinecraftServer.getServer().server.getCommandMap().register("hubschrauber", "Deepslate", new SpawnCommand("hubschrauber"));
         //Deepslate end
 
         // Yatopia start
diff --git a/src/main/java/net/minecraft/server/players/PlayerList.java b/src/main/java/net/minecraft/server/players/PlayerList.java
index 571fd37ed09a2bc8b5838b7a3d159fa0d99753c8..9f4e743da2d589ba55b93c169430b7ac7a81697f 100644
--- a/src/main/java/net/minecraft/server/players/PlayerList.java
+++ b/src/main/java/net/minecraft/server/players/PlayerList.java
@@ -92,6 +92,7 @@ import net.minecraft.world.scores.PlayerTeam;
 import net.minecraft.world.scores.Scoreboard; // Paper
 import net.minecraft.world.scores.Team;
 import net.pascalpex.deepslatemc.commands.WartungCommand;
+import net.pascalpex.deepslatemc.commands.SpawnCommand;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
@@ -358,6 +359,7 @@ public abstract class PlayerList {
 
         PlayerJoinEvent playerJoinEvent = new PlayerJoinEvent(bukkitPlayer, PaperAdventure.asAdventure(chatmessage)); // Paper - Adventure
         WartungCommand.playerJoin(bukkitPlayer);
+        SpawnCommand.playerJoin(bukkitPlayer);
         this.cserver.getPluginManager().callEvent(playerJoinEvent);
 
         if (!player.connection.connection.isConnected()) {
diff --git a/src/main/java/net/pascalpex/deepslatemc/Config.java b/src/main/java/net/pascalpex/deepslatemc/Config.java
index e26831d8d26081c95e9b5452057b35bfc186f13b..db6e3068d4395bc0e70aba38e5ee19c4a5574301 100644
--- a/src/main/java/net/pascalpex/deepslatemc/Config.java
+++ b/src/main/java/net/pascalpex/deepslatemc/Config.java
@@ -1,6 +1,9 @@
 package net.pascalpex.deepslatemc;
-    
+
+import org.bukkit.Bukkit;
 import org.bukkit.ChatColor;
+import org.bukkit.Location;
+import org.bukkit.World;
 import org.bukkit.configuration.InvalidConfigurationException;
 import org.bukkit.configuration.file.FileConfiguration;
 import org.bukkit.configuration.file.YamlConfiguration;
@@ -30,6 +33,7 @@ public class Config {
                 message.add("&9&l/warp &9&lInfos &r&6nachlesen. Wende dich bei");
                 message.add("&6weiteren Fragen an ein &9Teammitglied.");
                 config.set("helpMessage", message);
+                config.set("spawnOnJoin", false);
                 save();
             }
             config.load(configFile);
@@ -46,10 +50,34 @@ public class Config {
         }
     }
 
+    public static boolean getSpawnOnJoin() {
+        return config.getBoolean("spawnOnJoin");
+    }
+
     public static boolean getOpActive() {
         return config.getBoolean("opCommandActive");
     }
 
+    public static void setSpawn(Location loc) {
+        config.set("spawn" + ".world", loc.getWorld().getName());
+        config.set("spawn" + ".X", loc.getX());
+        config.set("spawn" + ".Y", loc.getY());
+        config.set("spawn" + ".Z", loc.getZ());
+        config.set("spawn" + ".pitch",(double) loc.getPitch());
+        config.set("spawn" + ".yaw",(double) loc.getYaw());
+        save();
+    }
+
+    public static Location getSpawn() {
+        World world = Bukkit.getWorld(config.getString("spawn" + ".world"));
+        double x = config.getDouble("spawn" + ".X");
+        double y = config.getDouble("spawn" + ".Y");
+        double z = config.getDouble("spawn" + ".Z");
+        float pitch = (float) config.getDouble("spawn" + ".pitch");
+        float yaw = (float) config.getDouble("spawn" + ".yaw");
+        return new Location(world, x, y, z, yaw, pitch);
+    }
+
     public static String getHelpMessage() {
         List<String> messages = (List<String>) config.getList("helpMessage");
         String message = "";
diff --git a/src/main/java/net/pascalpex/deepslatemc/commands/SpawnCommand.java b/src/main/java/net/pascalpex/deepslatemc/commands/SpawnCommand.java
new file mode 100644
index 0000000000000000000000000000000000000000..2add389551771752f482f25c11fa115a43deea51
--- /dev/null
+++ b/src/main/java/net/pascalpex/deepslatemc/commands/SpawnCommand.java
@@ -0,0 +1,54 @@
+package net.pascalpex.deepslatemc.commands;
+    
+import net.pascalpex.deepslatemc.Config;
+import org.bukkit.ChatColor;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+import org.jetbrains.annotations.NotNull;
+
+public class SpawnCommand extends Command {
+
+    public SpawnCommand(@NotNull String name) {
+        super(name);
+        this.description = "Teleportiert dich zum Spawn";
+        this.usageMessage = "/spawn";
+        this.setPermission("deepslate.spawn");
+    }
+
+    public static void playerJoin(Player player) {
+        if(Config.getSpawnOnJoin()) {
+            player.teleport(Config.getSpawn());
+        }
+    }
+
+    @Override
+    public boolean execute(@NotNull CommandSender sender, @NotNull String label, @NotNull String[] args) {
+        String prefix = Config.getPrefix() + " " + ChatColor.DARK_AQUA;
+        if(sender instanceof Player) {
+            
+            Player player = (Player) sender;
+            if(label.equalsIgnoreCase("setspawn")) {
+                if(player.hasPermission("deepslate.setspawn")) {
+                    Config.setSpawn(player.getLocation());
+                    player.sendMessage(prefix + ChatColor.GREEN + "Spawn erfolgreich gesetzt");
+                } else {
+                    player.sendMessage(prefix + ChatColor.RED + "Dafür hast du keine Rechte");
+                }
+            }
+            if(label.equalsIgnoreCase("l") || label.equalsIgnoreCase("hub") || label.equalsIgnoreCase("lobby") || label.equalsIgnoreCase("spawn") || label.equalsIgnoreCase("hubschrauber")) {
+                if(player.hasPermission("deepslate.spawn")) {
+                    player.teleport(Config.getSpawn());
+                    sender.sendMessage(prefix + ChatColor.GOLD + "Du wurdest zum Spawn teleportiert");
+                } else {
+                    player.sendMessage(prefix + ChatColor.RED + "Dafür hast du keine Rechte");
+                }
+            }
+            
+        } else {
+            sender.sendMessage(prefix + "Dieser Befehl ist nur für Spieler geeignet");
+        }
+        return true;
+    }
+
+}
