From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Sat, 24 Jun 2023 16:42:31 +0200
Subject: [PATCH] Configurable knowledgebook click


diff --git a/src/main/java/net/pascalpex/deepslatemc/Config.java b/src/main/java/net/pascalpex/deepslatemc/Config.java
index 6962584f335a75a4e23b68cc731e3d5856881e04..70cf7172bf95259294454ce94924ac24cf950279 100644
--- a/src/main/java/net/pascalpex/deepslatemc/Config.java
+++ b/src/main/java/net/pascalpex/deepslatemc/Config.java
@@ -39,6 +39,9 @@ public class Config {
             if(!config.contains("spawnOnJoin")) {
                 config.set("spawnOnJoin", false);
             }
+            if(!config.contains("preventKnowledgebookClick")) {
+                config.set("preventKnowledgebookClick", false);
+            }
             save();
         } catch (IOException | InvalidConfigurationException e) {
             e.printStackTrace();
@@ -89,5 +92,8 @@ public class Config {
         float yaw = (float) config.getDouble("spawn" + ".yaw");
         return new Location(world, x, y, z, yaw, pitch);
     }
+    public static boolean getPreventKnowledgebookClick() {
+        return config.getBoolean("preventKnowledgebookClick");
+    }
 
 }
diff --git a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
index 6db204dead67aac300410a6d2627cde8446630b5..a767e3af7503a7ff16f1616cf43f65772bd333bd 100644
--- a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
+++ b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
@@ -60,6 +60,7 @@ import net.minecraft.world.phys.BlockHitResult;
 import net.minecraft.world.phys.EntityHitResult;
 import net.minecraft.world.phys.HitResult;
 import net.minecraft.world.phys.Vec3;
+import net.pascalpex.deepslatemc.Config;
 import org.bukkit.Bukkit;
 import org.bukkit.Location;
 import org.bukkit.Material;
@@ -588,6 +589,17 @@ public class CraftEventFactory {
             event.setUseItemInHand(Result.DENY);
         }
         // Paper end
+        // Deepslate start - Configurable knowledgebook click
+        if (Config.getPreventKnowledgebookClick()) {
+            if (event.hasItem()) {
+                if (event.getAction() == Action.RIGHT_CLICK_AIR || event.getAction() == Action.RIGHT_CLICK_BLOCK) {
+                    if (event.getItem().getType() == Material.KNOWLEDGE_BOOK) {
+                        event.setCancelled(true);
+                    }
+                }
+            }
+        }
+        // Deepslate end
         craftServer.getPluginManager().callEvent(event);
 
         // Purpur start
