From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Mon, 21 Mar 2022 15:37:52 +0100
Subject: [PATCH] Configurable knowledgebook click


diff --git a/src/main/java/net/pascalpex/deepslatemc/Config.java b/src/main/java/net/pascalpex/deepslatemc/Config.java
index c591634e173da39764b4064963e4d500aaf07eee..145f6d6739f997a334cd30e7ce093947cba46ab8 100644
--- a/src/main/java/net/pascalpex/deepslatemc/Config.java
+++ b/src/main/java/net/pascalpex/deepslatemc/Config.java
@@ -38,6 +38,9 @@ public class Config {
                 if(!config.contains("spawnOnJoin")) {
                     config.set("spawnOnJoin", false);
                 }
+                if(!config.contains("preventKnowledgebookClick")) {
+                    config.set("preventKnowledgebookClick", false);
+                }
                 save();
             } catch (IOException | InvalidConfigurationException e) {
                 e.printStackTrace();
@@ -52,6 +55,10 @@ public class Config {
             }
         }
 
+    public static boolean getPreventKnowledgebookClick() {
+            return config.getBoolean("preventKnowledgebookClick");
+        }
+
         public static boolean getOpActive() {
             return config.getBoolean("opCommandActive");
         }
diff --git a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
index 361bd8edcc6e6d0fbac958e3ec91d74a622d630a..6b08fbf36ae6088bf6e5f7e690d630d00ef07306 100644
--- a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
+++ b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
@@ -57,6 +57,7 @@ import net.minecraft.world.phys.BlockHitResult;
 import net.minecraft.world.phys.EntityHitResult;
 import net.minecraft.world.phys.HitResult;
 import net.minecraft.world.phys.Vec3;
+import net.pascalpex.deepslatemc.Config;
 import org.bukkit.Bukkit;
 import org.bukkit.Location; // Paper
 import org.bukkit.Material;
@@ -538,6 +539,17 @@ public class CraftEventFactory {
         Location interactionPoint =  hitVec == null ? null : new Location(craftWorld, hitVec.x, hitVec.y, hitVec.z);
         PlayerInteractEvent event = new PlayerInteractEvent(player, action, itemInHand, blockClicked, blockFace, (hand == null) ? null : ((hand == InteractionHand.OFF_HAND) ? EquipmentSlot.OFF_HAND : EquipmentSlot.HAND), interactionPoint);
         // Paper end
+        // Deepslate start - Configurable knowledgebook click
+        if (Config.getPreventKnowledgebookClick()) {
+            if (event.hasItem()) {
+                if (event.getAction() == Action.RIGHT_CLICK_AIR || event.getAction() == Action.RIGHT_CLICK_BLOCK) {
+                    if (event.getItem().getType() == Material.KNOWLEDGE_BOOK) {
+                        event.setCancelled(true);
+                    }
+                }
+            }
+        }
+        // Deepslate end
         if (cancelledBlock) {
             event.setUseInteractedBlock(Event.Result.DENY);
         }
