From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Fri, 18 Feb 2022 16:45:10 +0100
Subject: [PATCH] Configurable knowledgebook click


diff --git a/src/main/java/net/pascalpex/deepslatemc/Config.java b/src/main/java/net/pascalpex/deepslatemc/Config.java
index 0c373fc76600e9889ace4666efba5560a32eba45..6008a38b1f0b51a9b85911d605d1bd455367de77 100644
--- a/src/main/java/net/pascalpex/deepslatemc/Config.java
+++ b/src/main/java/net/pascalpex/deepslatemc/Config.java
@@ -32,6 +32,9 @@ public class Config {
             if(!config.contains("spawnOnJoin")) {
                 config.set("spawnOnJoin", false);
             }
+            if(!config.contains("preventKnowledgebookClick")) {
+                config.set("preventKnowledgebookClick", false);
+            }
             save();
         } catch (IOException | InvalidConfigurationException e) {
             e.printStackTrace();
@@ -46,6 +49,10 @@ public class Config {
         }
     }
 
+    public static boolean getPreventKnowledgebookClick() {
+        return config.getBoolean("preventKnowledgebookClick");
+    }
+
     public static boolean getOpActive() {
         return config.getBoolean("opCommandActive");
     }
diff --git a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
index 361bd8edcc6e6d0fbac958e3ec91d74a622d630a..7503f4b2a32c70c9e81be75eaa6acc546e158016 100644
--- a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
+++ b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
@@ -57,6 +57,7 @@ import net.minecraft.world.phys.BlockHitResult;
 import net.minecraft.world.phys.EntityHitResult;
 import net.minecraft.world.phys.HitResult;
 import net.minecraft.world.phys.Vec3;
+import net.pascalpex.deepslatemc.Config;
 import org.bukkit.Bukkit;
 import org.bukkit.Location; // Paper
 import org.bukkit.Material;
@@ -537,6 +538,17 @@ public class CraftEventFactory {
         // Paper start
         Location interactionPoint =  hitVec == null ? null : new Location(craftWorld, hitVec.x, hitVec.y, hitVec.z);
         PlayerInteractEvent event = new PlayerInteractEvent(player, action, itemInHand, blockClicked, blockFace, (hand == null) ? null : ((hand == InteractionHand.OFF_HAND) ? EquipmentSlot.OFF_HAND : EquipmentSlot.HAND), interactionPoint);
+        // Deepslate start
+        if (Config.getPreventKnowledgebookClick()) {
+            if (event.hasItem()) {
+                if (event.getAction() == Action.RIGHT_CLICK_AIR || event.getAction() == Action.RIGHT_CLICK_BLOCK) {
+                    if (event.getItem().getType() == Material.KNOWLEDGE_BOOK) {
+                        event.setCancelled(true);
+                    }
+                }
+            }
+        }
+        // Deepslate end
         // Paper end
         if (cancelledBlock) {
             event.setUseInteractedBlock(Event.Result.DENY);
