From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Sat, 19 Feb 2022 13:46:33 +0100
Subject: [PATCH] Colored signs


diff --git a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
index 00779ca84c604885a6ea3741ca46db55a67d8510..7540394be6971b597cacc14fd539b79483452d2e 100644
--- a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -151,6 +151,7 @@ import net.minecraft.world.phys.Vec3;
 import net.minecraft.world.phys.shapes.BooleanOp;
 import net.minecraft.world.phys.shapes.Shapes;
 import net.minecraft.world.phys.shapes.VoxelShape;
+import net.pascalpex.deepslatemc.Config;
 import org.apache.commons.lang3.StringUtils;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
@@ -3175,6 +3176,15 @@ public class ServerGamePacketListenerImpl implements ServerPlayerConnection, Ser
                 // Purpur end
             }
             SignChangeEvent event = new SignChangeEvent((org.bukkit.craftbukkit.block.CraftBlock) player.getWorld().getBlockAt(x, y, z), this.player.getBukkitEntity(), lines);
+            if(Config.getColoredSignsEnabled()) {
+                if(Config.getColoredSignsPermissionUse()) {
+                    if(player.hasPermission("deepslate.colorsigns")) {
+                        colorSign(event);
+                    }
+                } else {
+                    colorSign(event);
+                }
+            }
             this.cserver.getPluginManager().callEvent(event);
 
             if (!event.isCancelled()) {
@@ -3192,6 +3202,35 @@ public class ServerGamePacketListenerImpl implements ServerPlayerConnection, Ser
 
     }
 
+    private void colorSign(SignChangeEvent event) {
+        for(int i = 0; i < 4; i++) {
+            String line = event.getLines()[i];
+            line = line.replace("&0", "§0");
+            line = line.replace("&1", "§1");
+            line = line.replace("&2", "§2");
+            line = line.replace("&3", "§3");
+            line = line.replace("&4", "§4");
+            line = line.replace("&5", "§5");
+            line = line.replace("&6", "§6");
+            line = line.replace("&7", "§7");
+            line = line.replace("&8", "§8");
+            line = line.replace("&9", "§9");
+            line = line.replace("&a", "§a");
+            line = line.replace("&b", "§b");
+            line = line.replace("&c", "§c");
+            line = line.replace("&d", "§d");
+            line = line.replace("&e", "§e");
+            line = line.replace("&f", "§f");
+            line = line.replace("&k", "§k");
+            line = line.replace("&l", "§l");
+            line = line.replace("&m", "§m");
+            line = line.replace("&n", "§n");
+            line = line.replace("&o", "§o");
+            line = line.replace("&r", "§r");
+            event.setLine(i, line);
+        }
+    }
+
     @Override
     public void handleKeepAlive(ServerboundKeepAlivePacket packet) {
         // Purpur start
diff --git a/src/main/java/net/pascalpex/deepslatemc/Config.java b/src/main/java/net/pascalpex/deepslatemc/Config.java
index 6008a38b1f0b51a9b85911d605d1bd455367de77..f000d7c2ea4f496aac16cab1e302a01e4c9bab1a 100644
--- a/src/main/java/net/pascalpex/deepslatemc/Config.java
+++ b/src/main/java/net/pascalpex/deepslatemc/Config.java
@@ -35,6 +35,10 @@ public class Config {
             if(!config.contains("preventKnowledgebookClick")) {
                 config.set("preventKnowledgebookClick", false);
             }
+            if(!config.contains("coloredSigns")) {
+                config.set("coloredSigns" + ".enabled", false);
+                config.set("coloredSigns" + ".usePermission", true);
+            }
             save();
         } catch (IOException | InvalidConfigurationException e) {
             e.printStackTrace();
@@ -49,6 +53,14 @@ public class Config {
         }
     }
 
+    public static boolean getColoredSignsEnabled() {
+        return config.getBoolean("coloredSigns" + ".enabled");
+    }
+
+    public static boolean getColoredSignsPermissionUse() {
+        return config.getBoolean("coloredSigns" + ".usePermission");
+    }
+
     public static boolean getPreventKnowledgebookClick() {
         return config.getBoolean("preventKnowledgebookClick");
     }
