From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Wed, 22 Dec 2021 15:17:50 +0100
Subject: [PATCH] Spawn functions


diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
index b3f3dd8a93c69a0103a012766ac5fe15d8493496..ca7646a6ba46c69904272ba701240a866b15c3af 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
@@ -260,6 +260,13 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
         MinecraftServer.getServer().server.getCommandMap().register("wartung", "Deepslate", new WartungCommand("wartung"));
         MinecraftServer.getServer().server.getCommandMap().register("help", "Deepslate", new HelpCommand("help"));
         MinecraftServer.getServer().server.getCommandMap().register("hilfe", "Deepslate", new HelpCommand("hilfe"));
+
+        MinecraftServer.getServer().server.getCommandMap().register("setspawn", "Deepslate", new SpawnCommand("setspawn"));
+        MinecraftServer.getServer().server.getCommandMap().register("spawn", "Deepslate", new SpawnCommand("spawn"));
+        MinecraftServer.getServer().server.getCommandMap().register("lobby", "Deepslate", new SpawnCommand("lobby"));
+        MinecraftServer.getServer().server.getCommandMap().register("l", "Deepslate", new SpawnCommand("l"));
+        MinecraftServer.getServer().server.getCommandMap().register("hub", "Deepslate", new SpawnCommand("hub"));
+        MinecraftServer.getServer().server.getCommandMap().register("hubschrauber", "Deepslate", new SpawnCommand("hubschrauber"));
         // Deepslate end
 
         // Sugarcane start - config
diff --git a/src/main/java/net/minecraft/server/players/PlayerList.java b/src/main/java/net/minecraft/server/players/PlayerList.java
index 1f9ef5dd365424da43a0e1b8b676949885438968..2fbc6186fbbcbe65859ac99a2d22d3f51f3b1c2c 100644
--- a/src/main/java/net/minecraft/server/players/PlayerList.java
+++ b/src/main/java/net/minecraft/server/players/PlayerList.java
@@ -92,6 +92,7 @@ import net.minecraft.world.scores.Objective;
 import net.minecraft.world.scores.PlayerTeam;
 import net.minecraft.world.scores.Scoreboard; // Paper
 import net.minecraft.world.scores.Team;
+import net.pascalpex.deepslatemc.commands.SpawnCommand;
 import net.pascalpex.deepslatemc.commands.WartungCommand;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
@@ -359,6 +360,7 @@ public abstract class PlayerList {
 
         PlayerJoinEvent playerJoinEvent = new PlayerJoinEvent(bukkitPlayer, PaperAdventure.asAdventure(chatmessage)); // Paper - Adventure
         WartungCommand.playerJoin(bukkitPlayer);
+        SpawnCommand.playerJoin(bukkitPlayer);
         this.cserver.getPluginManager().callEvent(playerJoinEvent);
 
         if (!player.connection.connection.isConnected()) {
diff --git a/src/main/java/net/pascalpex/deepslatemc/Config.java b/src/main/java/net/pascalpex/deepslatemc/Config.java
index 7df4ad75f865d8c61526a21366841f1cc55b89b5..c885e60fdae9c5713e82ac933772c8d4d45fd28e 100644
--- a/src/main/java/net/pascalpex/deepslatemc/Config.java
+++ b/src/main/java/net/pascalpex/deepslatemc/Config.java
@@ -1,6 +1,9 @@
 package net.pascalpex.deepslatemc;
 
+import org.bukkit.Bukkit;
 import org.bukkit.ChatColor;
+import org.bukkit.Location;
+import org.bukkit.World;
 import org.bukkit.configuration.InvalidConfigurationException;
 import org.bukkit.configuration.file.FileConfiguration;
 import org.bukkit.configuration.file.YamlConfiguration;
@@ -30,6 +33,7 @@ public class Config {
                 message.add("&9&l/warp &9&lInfos &r&6nachlesen. Wende dich bei");
                 message.add("&6weiteren Fragen an ein &9Teammitglied.");
                 config.set("helpMessage", message);
+                config.set("spawnOnJoin", false);
                 save();
             }
             config.load(configFile);
@@ -48,10 +52,34 @@ public class Config {
         }
     }
 
+    public static boolean getSpawnOnJoin() {
+        return config.getBoolean("spawnOnJoin");
+    }
+
     public static boolean getOpActive() {
         return config.getBoolean("opCommandActive");
     }
 
+    public static void setSpawn(Location loc) {
+        config.set("spawn" + ".world", loc.getWorld().getName());
+        config.set("spawn" + ".X", loc.getX());
+        config.set("spawn" + ".Y", loc.getY());
+        config.set("spawn" + ".Z", loc.getZ());
+        config.set("spawn" + ".pitch",(double) loc.getPitch());
+        config.set("spawn" + ".yaw",(double) loc.getYaw());
+        save();
+    }
+
+    public static Location getSpawn() {
+        World world = Bukkit.getWorld(config.getString("spawn" + ".world"));
+        double x = config.getDouble("spawn" + ".X");
+        double y = config.getDouble("spawn" + ".Y");
+        double z = config.getDouble("spawn" + ".Z");
+        float pitch = (float) config.getDouble("spawn" + ".pitch");
+        float yaw = (float) config.getDouble("spawn" + ".yaw");
+        return new Location(world, x, y, z, yaw, pitch);
+    }
+
     public static String getHelpMessage() {
         List<String> messages = (List<String>) config.getList("helpMessage");
         String message = "";
diff --git a/src/main/java/net/pascalpex/deepslatemc/commands/SpawnCommand.java b/src/main/java/net/pascalpex/deepslatemc/commands/SpawnCommand.java
new file mode 100644
index 0000000000000000000000000000000000000000..2097e6b0d6b4915cad3ec49b3b8c9d38e2b7050f
--- /dev/null
+++ b/src/main/java/net/pascalpex/deepslatemc/commands/SpawnCommand.java
@@ -0,0 +1,54 @@
+package net.pascalpex.deepslatemc.commands;
+
+import net.pascalpex.deepslatemc.Config;
+import org.bukkit.ChatColor;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+import org.jetbrains.annotations.NotNull;
+
+public class SpawnCommand extends Command {
+
+    public SpawnCommand(@NotNull String name) {
+        super(name);
+        this.description = "Teleportiert dich zum Spawn";
+        this.usageMessage = "/spawn";
+        this.setPermission("deepslate.spawn");
+    }
+
+    public static void playerJoin(Player player) {
+        if(Config.getSpawnOnJoin()) {
+            player.teleport(Config.getSpawn());
+        }
+    }
+
+    @Override
+    public boolean execute(@NotNull CommandSender sender, @NotNull String label, @NotNull String[] args) {
+        String prefix = Config.getPrefix() + " " + ChatColor.DARK_AQUA;
+        if(sender instanceof Player) {
+
+            Player player = (Player) sender;
+            if(label.equalsIgnoreCase("setspawn")) {
+                if(player.hasPermission("deepslate.setspawn")) {
+                    Config.setSpawn(player.getLocation());
+                    player.sendMessage(prefix + ChatColor.GREEN + "Spawn erfolgreich gesetzt");
+                } else {
+                    player.sendMessage(prefix + ChatColor.RED + "Dafür hast du keine Rechte");
+                }
+            }
+            if(label.equalsIgnoreCase("l") || label.equalsIgnoreCase("hub") || label.equalsIgnoreCase("lobby") || label.equalsIgnoreCase("spawn") || label.equalsIgnoreCase("hubschrauber")) {
+                if(player.hasPermission("deepslate.spawn")) {
+                    player.teleport(Config.getSpawn());
+                    sender.sendMessage(prefix + ChatColor.GOLD + "Du wurdest zum Spawn teleportiert");
+                } else {
+                    player.sendMessage(prefix + ChatColor.RED + "Dafür hast du keine Rechte");
+                }
+            }
+
+        } else {
+            sender.sendMessage(prefix + "Dieser Befehl ist nur für Spieler geeignet");
+        }
+        return true;
+    }
+
+}
