From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Sat, 1 Jan 2022 17:05:56 +0100
Subject: [PATCH] Configurable Knowledgebook Click


diff --git a/src/main/java/net/pascalpex/deepslatemc/Config.java b/src/main/java/net/pascalpex/deepslatemc/Config.java
index 70a82964ac31cab89a0d27e1944f4b5aba6b98c8..191e8bedf91bc6d38951c0731b156cf0ed1e1755 100644
--- a/src/main/java/net/pascalpex/deepslatemc/Config.java
+++ b/src/main/java/net/pascalpex/deepslatemc/Config.java
@@ -32,6 +32,7 @@ public class Config {
                 message.add("&6weiteren Fragen an ein &9Teammitglied.");
                 config.set("helpMessage", message);
                 config.set("spawnOnJoin", false);
+                config.set("preventKnowledgebookClick", false);
                 save();
             }
             config.load(configFile);
@@ -53,6 +54,10 @@ public class Config {
         return config.getBoolean("spawnOnJoin");
     }
 
+    public static boolean getPreventKnowledgebookClick() {
+        return config.getBoolean("preventKnowledgebookClick");
+    }
+
     public static boolean getOpActive() {
         return config.getBoolean("opCommandActive");
     }
diff --git a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
index 4bf67cdbe9516e4d167ddcfca3d0c033a7ca5172..ee7c0009cb490141a7fb061d730e8a4a83cdf8fc 100644
--- a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
+++ b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
@@ -57,6 +57,7 @@ import net.minecraft.world.phys.BlockHitResult;
 import net.minecraft.world.phys.EntityHitResult;
 import net.minecraft.world.phys.HitResult;
 import net.minecraft.world.phys.Vec3;
+import net.pascalpex.deepslatemc.Config;
 import org.bukkit.Bukkit;
 import org.bukkit.Location; // Paper
 import org.bukkit.Material;
@@ -536,6 +537,15 @@ public class CraftEventFactory {
         // Paper start
         Location interactionPoint =  hitVec == null ? null : new Location(craftWorld, hitVec.x, hitVec.y, hitVec.z);
         PlayerInteractEvent event = new PlayerInteractEvent(player, action, itemInHand, blockClicked, blockFace, (hand == null) ? null : ((hand == InteractionHand.OFF_HAND) ? EquipmentSlot.OFF_HAND : EquipmentSlot.HAND), interactionPoint);
+        if (Config.getPreventKnowledgebookClick()) {
+            if (event.hasItem()) {
+                if (event.getAction() == Action.RIGHT_CLICK_AIR || event.getAction() == Action.RIGHT_CLICK_BLOCK) {
+                    if (event.getItem().getType() == Material.KNOWLEDGE_BOOK) {
+                        event.setCancelled(true);
+                    }
+                }
+            }
+        }
         // Paper end
         if (cancelledBlock) {
             event.setUseInteractedBlock(Event.Result.DENY);
