From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Sat, 3 Dec 2022 14:28:20 +0100
Subject: [PATCH] Autosmelt pickaxes


diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
index c6e72eab5c29837f656a74a309535c239774e47a..6bde3a10de141161f9d81a2871572cbd521b0b53 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
@@ -243,6 +243,7 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
         MinecraftServer.getServer().server.getCommandMap().register("setspawn", "Deepslate", new SpawnCommand("setspawn"));
         MinecraftServer.getServer().server.getCommandMap().register("spawn", "Deepslate", new SpawnCommand("spawn"));
         MinecraftServer.getServer().server.getCommandMap().register("lobby", "Deepslate", new SpawnCommand("lobby"));
+        MinecraftServer.getServer().server.getCommandMap().register("autosmelt", "Deepslate", new AutosmeltCommand("autosmelt"));
         // Deepslate end
 
         this.setPvpAllowed(dedicatedserverproperties.pvp);
diff --git a/src/main/java/net/minecraft/server/level/ServerPlayerGameMode.java b/src/main/java/net/minecraft/server/level/ServerPlayerGameMode.java
index ebf1ac089202c06fd2cc593dc12c21fe2d0a8de8..92291db1e8c16f91321a6b01399baf1dfa444f68 100644
--- a/src/main/java/net/minecraft/server/level/ServerPlayerGameMode.java
+++ b/src/main/java/net/minecraft/server/level/ServerPlayerGameMode.java
@@ -27,6 +27,9 @@ import net.minecraft.world.level.block.state.BlockState;
 import net.minecraft.world.level.block.state.properties.DoubleBlockHalf;
 import net.minecraft.world.phys.BlockHitResult;
 import net.minecraft.world.phys.Vec3;
+import net.pascalpex.deepslatemc.Config;
+import org.bukkit.Material;
+import org.bukkit.entity.Player;
 import org.slf4j.Logger;
 
 // CraftBukkit start
@@ -353,6 +356,59 @@ public class ServerPlayerGameMode {
 
             event = new BlockBreakEvent(bblock, this.player.getBukkitEntity());
 
+            // Deepslate start - Autosmelt pickaxes
+            Player Bplayer = this.player.getBukkitEntity();
+            if (Bplayer.getItemInHand() != null && Bplayer.getItemInHand().getType() != Material.AIR) {
+                if (Bplayer.getItemInHand().getItemMeta().hasLore()) {
+                    if (Bplayer.getItemInHand().getItemMeta().getLore().contains("Auto Smelt")) {
+                        if (Config.getAutoSmeltWorlds().contains(Bplayer.getWorld().getName())) {
+                            if (Bplayer.getGameMode() != GameMode.CREATIVE) {
+                                switch (event.getBlock().getType()) {
+                                    case IRON_ORE:
+                                    case DEEPSLATE_IRON_ORE:
+                                        event.setDropItems(false);
+                                        Bplayer.getInventory().addItem(new org.bukkit.inventory.ItemStack(Material.IRON_INGOT, 1));
+                                        break;
+                                    case GOLD_ORE:
+                                    case DEEPSLATE_GOLD_ORE:
+                                        event.setDropItems(false);
+                                        Bplayer.getInventory().addItem(new org.bukkit.inventory.ItemStack(Material.GOLD_INGOT, 1));
+                                        break;
+                                    case COBBLESTONE:
+                                        event.setDropItems(false);
+                                        Bplayer.getInventory().addItem(new org.bukkit.inventory.ItemStack(Material.STONE, 1));
+                                        break;
+                                    case COBBLED_DEEPSLATE:
+                                        event.setDropItems(false);
+                                        Bplayer.getInventory().addItem(new org.bukkit.inventory.ItemStack(Material.DEEPSLATE, 1));
+                                        break;
+                                    case STONE:
+                                        event.setDropItems(false);
+                                        Bplayer.getInventory().addItem(new org.bukkit.inventory.ItemStack(Material.SMOOTH_STONE, 1));
+                                        break;
+                                    case SAND:
+                                        event.setDropItems(false);
+                                        Bplayer.getInventory().addItem(new org.bukkit.inventory.ItemStack(Material.GLASS, 1));
+                                        break;
+                                    case ANCIENT_DEBRIS:
+                                        event.setDropItems(false);
+                                        Bplayer.getInventory().addItem(new org.bukkit.inventory.ItemStack(Material.NETHERITE_SCRAP, 1));
+                                        break;
+                                    case COPPER_ORE:
+                                    case DEEPSLATE_COPPER_ORE:
+                                        event.setDropItems(false);
+                                        Bplayer.getInventory().addItem(new org.bukkit.inventory.ItemStack(Material.COPPER_INGOT, 1));
+                                        break;
+                                    default:
+                                        break;
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            // Deepslate end
+
             // Sword + Creative mode pre-cancel
             event.setCancelled(isSwordNoBreak);
 
diff --git a/src/main/java/net/pascalpex/deepslatemc/Config.java b/src/main/java/net/pascalpex/deepslatemc/Config.java
index 9a2ecd57135827d22448ae19650122af069faad0..51e722413faa6bf4c74b3162dc59c9a44f6bc44a 100644
--- a/src/main/java/net/pascalpex/deepslatemc/Config.java
+++ b/src/main/java/net/pascalpex/deepslatemc/Config.java
@@ -46,6 +46,12 @@ public class Config {
                 config.set("coloredSigns" + ".enabled", false);
                 config.set("coloredSigns" + ".usePermission", true);
             }
+            if(!config.contains("autoSmelt")) {
+                config.set("autoSmelt" + ".price", 30);
+                List<String> worldsList = new ArrayList<>();
+                worldsList.add("FantasyWorld");
+                config.set("autoSmelt" + ".enabledWorlds", worldsList);
+            }
             save();
         } catch (IOException | InvalidConfigurationException e) {
             e.printStackTrace();
@@ -105,5 +111,11 @@ public class Config {
     public static boolean getColoredSignsPermissionUse() {
         return config.getBoolean("coloredSigns" + ".usePermission");
     }
+    public static int getAutoSmeltPrice() {
+        return config.getInt("autoSmelt" + ".price");
+    }
+    public static List<String> getAutoSmeltWorlds() {
+        return (List<String>) config.getList("autoSmelt" + ".enabledWorlds");
+    }
 
 }
diff --git a/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java b/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
index 155380686dcccb368820b45eab67a21bed9d85f1..60182222713d3c412f0332fd57911108e3fbea83 100644
--- a/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
+++ b/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
@@ -82,6 +82,21 @@ public class MessagesFile {
             if(!config.contains("spawnTeleport")) {
                 config.set("spawnTeleport", "&aYou got teleported to the spawn");
             }
+            if(!config.contains("autoSmeltEnchanted")) {
+                config.set("autoSmeltEnchanted", "&aYour pickaxe now has autosmelt");
+            }
+            if(!config.contains("autoSmeltUnable")) {
+                config.set("autoSmeltUnable", "&cThis pickaxe cannot be enchanted");
+            }
+            if(!config.contains("autoSmeltNoPickaxe")) {
+                config.set("autoSmeltNoPickaxe", "&cYou have to hold a pickaxe in your hand");
+            }
+            if(!config.contains("autoSmeltLevels")) {
+                config.set("autoSmeltLevels", "&cYou need %price% experience levels");
+            }
+            if(!config.contains("autoSmeltNotHere")) {
+                config.set("autoSmeltNotHere", "&cAutosmelt does not work here");
+            }
             save();
         } catch (IOException | InvalidConfigurationException e) {
             e.printStackTrace();
@@ -171,5 +186,25 @@ public class MessagesFile {
         String prefix = config.getString("spawnTeleport");
         return prefix.replace("&", "§");
     }
+    public static String getAutoSmeltEnchanted() {
+        String prefix = config.getString("autoSmeltEnchanted");
+        return prefix.replace("&", "§");
+    }
+    public static String getAutoSmeltUnable() {
+        String prefix = config.getString("autoSmeltUnable");
+        return prefix.replace("&", "§");
+    }
+    public static String getAutoSmeltNoPickaxe() {
+        String prefix = config.getString("autoSmeltNoPickaxe");
+        return prefix.replace("&", "§");
+    }
+    public static String getAutoSmeltLevels() {
+        String prefix = config.getString("autoSmeltLevels");
+        return prefix.replace("&", "§");
+    }
+    public static String getAutoSmeltNotHere() {
+        String prefix = config.getString("autoSmeltNotHere");
+        return prefix.replace("&", "§");
+    }
 
 }
diff --git a/src/main/java/net/pascalpex/deepslatemc/commands/AutosmeltCommand.java b/src/main/java/net/pascalpex/deepslatemc/commands/AutosmeltCommand.java
new file mode 100644
index 0000000000000000000000000000000000000000..baf1f1d28c6610aeec912e858a3e3fb2f17b74fb
--- /dev/null
+++ b/src/main/java/net/pascalpex/deepslatemc/commands/AutosmeltCommand.java
@@ -0,0 +1,64 @@
+package net.pascalpex.deepslatemc.commands;
+    
+import net.pascalpex.deepslatemc.Config;
+import net.pascalpex.deepslatemc.MessagesFile;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ItemMeta;
+
+import java.util.ArrayList;
+
+public class AutosmeltCommand extends Command {
+
+    public AutosmeltCommand(String name) {
+        super(name);
+        this.description = "Enchants a pickaxe with autosmelt";
+        this.usageMessage = "/autosmelt";
+        this.setPermission("deepslate.autosmelt");
+    }
+
+    @Override
+    public boolean execute(CommandSender sender, String label, String[] args) {
+        String prefix = MessagesFile.getPrefix() + " ";
+        if (sender instanceof Player) {
+            Player player = (Player) sender;
+            if(label.equalsIgnoreCase("autosmelt")) {
+                if(player.hasPermission("deepslate.autosmelt")) {
+                    if (Config.getAutoSmeltWorlds().contains(player.getWorld().getName())) {
+                        if (player.getLevel() >= Config.getAutoSmeltPrice()) {
+                            if (player.getItemInHand().getType().toString().contains("PICKAXE")) {
+                                if (!player.getItemInHand().getItemMeta().hasLore()) {
+                                    ItemStack item = player.getItemInHand();
+                                    ItemMeta meta = item.getItemMeta();
+                                    ArrayList<String> lore = new ArrayList<>();
+                                    lore.add("Auto Smelt");
+                                    meta.setLore(lore);
+                                    item.setItemMeta(meta);
+                                    player.getInventory().setItemInHand(item);
+                                    player.sendMessage(prefix + MessagesFile.getAutoSmeltEnchanted());
+                                    player.giveExpLevels(-Config.getAutoSmeltPrice());
+                                } else {
+                                    player.sendMessage(prefix + MessagesFile.getAutoSmeltUnable());
+                                }
+                            } else {
+                                player.sendMessage(prefix + MessagesFile.getAutoSmeltNoPickaxe());
+                            }
+                        } else {
+                            player.sendMessage(prefix + MessagesFile.getAutoSmeltLevels().replace("%price%", String.valueOf(Config.getAutoSmeltPrice())));
+                        }
+                    } else {
+                        player.sendMessage(prefix + MessagesFile.getAutoSmeltNotHere());
+                    }
+                } else {
+                    player.sendMessage(prefix + MessagesFile.getNoPermissions());
+                }
+            }
+        } else {
+            sender.sendMessage(prefix + MessagesFile.getOnlyForPlayers());
+        }
+        return true;
+    }
+
+}
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index f7f04eabbb0d056ec359ee4616cd7e7fb8e5c65e..d1a075a9de626e63edda5457d1fa9cc6700418e1 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -1019,6 +1019,7 @@ public final class CraftServer implements Server {
         MinecraftServer.getServer().server.getCommandMap().register("setspawn", "Deepslate", new SpawnCommand("setspawn"));
         MinecraftServer.getServer().server.getCommandMap().register("spawn", "Deepslate", new SpawnCommand("spawn"));
         MinecraftServer.getServer().server.getCommandMap().register("lobby", "Deepslate", new SpawnCommand("lobby"));
+        MinecraftServer.getServer().server.getCommandMap().register("autosmelt", "Deepslate", new AutosmeltCommand("autosmelt"));
         // Deepslate end
 
         this.overrideAllCommandBlockCommands = this.commandsConfiguration.getStringList("command-block-overrides").contains("*");
