From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Sat, 19 Feb 2022 14:20:01 +0100
Subject: [PATCH] Autosmelt pickaxes


diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
index fa09df0f45972ed5ecc5242ea56b1fea8e216c84..728a2c9dc98121dc60c95d5d76320f1406d64802 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
@@ -271,6 +271,7 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
         MinecraftServer.getServer().server.getCommandMap().register("wartung", "Deepslate", new MaintenanceMode("wartung"));
         MinecraftServer.getServer().server.getCommandMap().register("help", "Deepslate", new HelpCommand("help"));
         MinecraftServer.getServer().server.getCommandMap().register("hilfe", "Deepslate", new HelpCommand("hilfe"));
+        MinecraftServer.getServer().server.getCommandMap().register("autosmelt", "Deepslate", new AutosmeltCommand("autosmelt"));
 
         MinecraftServer.getServer().server.getCommandMap().register("setspawn", "Deepslate", new SpawnCommand("setspawn"));
         MinecraftServer.getServer().server.getCommandMap().register("spawn", "Deepslate", new SpawnCommand("spawn"));
diff --git a/src/main/java/net/minecraft/server/level/ServerPlayerGameMode.java b/src/main/java/net/minecraft/server/level/ServerPlayerGameMode.java
index 29329381ee67a409ed6b324c8c1f0d7ae05970da..b4a1df84265f3600de2e2925d9352a1829c57578 100644
--- a/src/main/java/net/minecraft/server/level/ServerPlayerGameMode.java
+++ b/src/main/java/net/minecraft/server/level/ServerPlayerGameMode.java
@@ -19,6 +19,7 @@ import net.minecraft.world.level.block.entity.BlockEntity;
 import net.minecraft.world.level.block.state.BlockState;
 import net.minecraft.world.level.block.state.properties.DoubleBlockHalf;
 import net.minecraft.world.phys.BlockHitResult;
+import net.pascalpex.deepslatemc.Config;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
@@ -36,7 +37,9 @@ import net.minecraft.world.level.block.Blocks;
 import net.minecraft.world.level.block.CakeBlock;
 import net.minecraft.world.level.block.DoorBlock;
 import org.bukkit.GameMode;
+import org.bukkit.Material;
 import org.bukkit.craftbukkit.block.CraftBlock;
+import org.bukkit.entity.Player;
 import org.bukkit.event.block.BlockBreakEvent;
 import org.bukkit.craftbukkit.event.CraftEventFactory;
 import org.bukkit.event.Event;
@@ -385,6 +388,56 @@ public class ServerPlayerGameMode {
             }
 
             event = new BlockBreakEvent(bblock, this.player.getBukkitEntity());
+            Player Bplayer = this.player.getBukkitEntity();
+            if (Bplayer.getItemInHand() != null && Bplayer.getItemInHand().getType() != Material.AIR) {
+                if (Bplayer.getItemInHand().getItemMeta().hasLore()) {
+                    if (Bplayer.getItemInHand().getItemMeta().getLore().contains("Auto Smelt")) {
+                        if (Config.getAutoSmeltWorlds().contains(Bplayer.getWorld().getName())) {
+                            if (Bplayer.getGameMode() != GameMode.CREATIVE) {
+                                switch (event.getBlock().getType()) {
+                                    case IRON_ORE:
+                                    case DEEPSLATE_IRON_ORE:
+                                        event.setDropItems(false);
+                                        Bplayer.getInventory().addItem(new org.bukkit.inventory.ItemStack(Material.IRON_INGOT, 1));
+                                        break;
+                                    case GOLD_ORE:
+                                    case DEEPSLATE_GOLD_ORE:
+                                        event.setDropItems(false);
+                                        Bplayer.getInventory().addItem(new org.bukkit.inventory.ItemStack(Material.GOLD_INGOT, 1));
+                                        break;
+                                    case COBBLESTONE:
+                                        event.setDropItems(false);
+                                        Bplayer.getInventory().addItem(new org.bukkit.inventory.ItemStack(Material.STONE, 1));
+                                        break;
+                                    case COBBLED_DEEPSLATE:
+                                        event.setDropItems(false);
+                                        Bplayer.getInventory().addItem(new org.bukkit.inventory.ItemStack(Material.DEEPSLATE, 1));
+                                        break;
+                                    case STONE:
+                                        event.setDropItems(false);
+                                        Bplayer.getInventory().addItem(new org.bukkit.inventory.ItemStack(Material.SMOOTH_STONE, 1));
+                                        break;
+                                    case SAND:
+                                        event.setDropItems(false);
+                                        Bplayer.getInventory().addItem(new org.bukkit.inventory.ItemStack(Material.GLASS, 1));
+                                        break;
+                                    case ANCIENT_DEBRIS:
+                                        event.setDropItems(false);
+                                        Bplayer.getInventory().addItem(new org.bukkit.inventory.ItemStack(Material.NETHERITE_SCRAP, 1));
+                                        break;
+                                    case COPPER_ORE:
+                                    case DEEPSLATE_COPPER_ORE:
+                                        event.setDropItems(false);
+                                        Bplayer.getInventory().addItem(new org.bukkit.inventory.ItemStack(Material.COPPER_INGOT, 1));
+                                        break;
+                                    default:
+                                        break;
+                                }
+                            }
+                        }
+                    }
+                }
+            }
 
             // Sword + Creative mode pre-cancel
             event.setCancelled(isSwordNoBreak);
diff --git a/src/main/java/net/pascalpex/deepslatemc/Config.java b/src/main/java/net/pascalpex/deepslatemc/Config.java
index f000d7c2ea4f496aac16cab1e302a01e4c9bab1a..a67218c323556d5bb85f47b4dff729f4c1d7b34d 100644
--- a/src/main/java/net/pascalpex/deepslatemc/Config.java
+++ b/src/main/java/net/pascalpex/deepslatemc/Config.java
@@ -9,6 +9,8 @@ import org.bukkit.configuration.file.YamlConfiguration;
 
 import java.io.File;
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
 
 public class Config {
 
@@ -39,6 +41,12 @@ public class Config {
                 config.set("coloredSigns" + ".enabled", false);
                 config.set("coloredSigns" + ".usePermission", true);
             }
+            if(!config.contains("autoSmelt")) {
+                config.set("autoSmelt" + ".price", 30);
+                List<String> worldsList = new ArrayList<>();
+                worldsList.add("FantasyWorld");
+                config.set("autoSmelt" + ".enabledWorlds", worldsList);
+            }
             save();
         } catch (IOException | InvalidConfigurationException e) {
             e.printStackTrace();
@@ -53,6 +61,14 @@ public class Config {
         }
     }
 
+    public static int getAutoSmeltPrice() {
+        return config.getInt("autoSmelt" + ".price");
+    }
+
+    public static List<String> getAutoSmeltWorlds() {
+        return (List<String>) config.getList("autoSmelt" + ".enabledWorlds");
+    }
+
     public static boolean getColoredSignsEnabled() {
         return config.getBoolean("coloredSigns" + ".enabled");
     }
diff --git a/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java b/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
index a443d3d70b3a1a85a28b77ea3f58c3731e92248d..0934cf79afcace5d5dc069255f51c753c8683056 100644
--- a/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
+++ b/src/main/java/net/pascalpex/deepslatemc/MessagesFile.java
@@ -75,6 +75,21 @@ public class MessagesFile {
             if(!config.contains("spawnTeleport")) {
                 config.set("spawnTeleport", "&aYou got teleported to the spawn");
             }
+            if(!config.contains("autoSmeltEnchanted")) {
+                config.set("autoSmeltEnchanted", "&aYour pickaxe now has autosmelt");
+            }
+            if(!config.contains("autoSmeltUnable")) {
+                config.set("autoSmeltUnable", "&cThis pickaxe cannot be enchanted");
+            }
+            if(!config.contains("autoSmeltNoPickaxe")) {
+                config.set("autoSmeltNoPickaxe", "&cYou have to hold a pickaxe in your hand");
+            }
+            if(!config.contains("autoSmeltLevels")) {
+                config.set("autoSmeltLevels", "&cYou need %price% experience levels");
+            }
+            if(!config.contains("autoSmeltNotHere")) {
+                config.set("autoSmeltNotHere", "&cAutosmelt does not work here");
+            }
             save();
         } catch (IOException | InvalidConfigurationException e) {
             e.printStackTrace();
@@ -168,5 +183,25 @@ public class MessagesFile {
         String prefix = config.getString("spawnTeleport");
         return prefix.replace("&", "§");
     }
+    public static String getAutoSmeltEnchanted() {
+        String prefix = config.getString("autoSmeltEnchanted");
+        return prefix.replace("&", "§");
+    }
+    public static String getAutoSmeltUnable() {
+        String prefix = config.getString("autoSmeltUnable");
+        return prefix.replace("&", "§");
+    }
+    public static String getAutoSmeltNoPickaxe() {
+        String prefix = config.getString("autoSmeltNoPickaxe");
+        return prefix.replace("&", "§");
+    }
+    public static String getAutoSmeltLevels() {
+        String prefix = config.getString("autoSmeltLevels");
+        return prefix.replace("&", "§");
+    }
+    public static String getAutoSmeltNotHere() {
+        String prefix = config.getString("autoSmeltNotHere");
+        return prefix.replace("&", "§");
+    }
 
 }
diff --git a/src/main/java/net/pascalpex/deepslatemc/commands/AutosmeltCommand.java b/src/main/java/net/pascalpex/deepslatemc/commands/AutosmeltCommand.java
new file mode 100644
index 0000000000000000000000000000000000000000..2e7c7466fafd7c7ffb24e3568fb220a5c7d0f364
--- /dev/null
+++ b/src/main/java/net/pascalpex/deepslatemc/commands/AutosmeltCommand.java
@@ -0,0 +1,66 @@
+package net.pascalpex.deepslatemc.commands;
+    
+import net.pascalpex.deepslatemc.Config;
+import net.pascalpex.deepslatemc.MessagesFile;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ItemMeta;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+
+public class AutosmeltCommand extends Command {
+
+    public AutosmeltCommand(@NotNull String name) {
+        super(name);
+        this.description = "Enchants a pickaxe with autosmelt";
+        this.usageMessage = "/autosmelt";
+        this.setPermission("deepslate.autosmelt");
+    }
+
+    @Override
+    public boolean execute(@NotNull CommandSender sender, @NotNull String label, @NotNull String[] args) {
+        String prefix = MessagesFile.getPrefix() + " ";
+        if (sender instanceof Player) {
+            Player player = (Player) sender;
+            if(label.equalsIgnoreCase("autosmelt")) {
+                if(player.hasPermission("deepslate.autosmelt")) {
+                    if (Config.getAutoSmeltWorlds().contains(player.getWorld().getName())) {
+                        if (player.getLevel() >= Config.getAutoSmeltPrice()) {
+                            if (player.getItemInHand().getType().toString().contains("PICKAXE")) {
+                                if (!player.getItemInHand().getItemMeta().hasLore()) {
+                                    ItemStack item = player.getItemInHand();
+                                    ItemMeta meta = item.getItemMeta();
+                                    ArrayList<String> lore = new ArrayList<>();
+                                    lore.add("Auto Smelt");
+                                    meta.setLore(lore);
+                                    item.setItemMeta(meta);
+                                    player.getInventory().setItemInHand(item);
+                                    player.sendMessage(prefix + MessagesFile.getAutoSmeltEnchanted());
+                                    player.giveExpLevels(-Config.getAutoSmeltPrice());
+                                } else {
+                                    player.sendMessage(prefix + MessagesFile.getAutoSmeltUnable());
+                                }
+                                
+                            } else {
+                                player.sendMessage(prefix + MessagesFile.getAutoSmeltNoPickaxe());
+                            }
+                        } else {
+                            player.sendMessage(prefix + MessagesFile.getAutoSmeltLevels().replace("%price%", String.valueOf(Config.getAutoSmeltPrice())));
+                        }
+                    } else {
+                        player.sendMessage(prefix + MessagesFile.getAutoSmeltNotHere());
+                    }
+                } else {
+                    player.sendMessage(prefix + MessagesFile.getNoPermissions());
+                }
+            }
+        } else {
+            sender.sendMessage(prefix + MessagesFile.getOnlyForPlayers());
+        }
+        return true;
+    }
+
+}
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index a5838b3fb69ddf7b24f20ff31ce487990d7ee178..7cc9f76ba97ca6b1d1a84ad31f30a3c906dcaf49 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -1011,6 +1011,7 @@ public final class CraftServer implements Server {
         MinecraftServer.getServer().server.getCommandMap().register("wartung", "Deepslate", new MaintenanceMode("wartung"));
         MinecraftServer.getServer().server.getCommandMap().register("help", "Deepslate", new HelpCommand("help"));
         MinecraftServer.getServer().server.getCommandMap().register("hilfe", "Deepslate", new HelpCommand("hilfe"));
+        MinecraftServer.getServer().server.getCommandMap().register("autosmelt", "Deepslate", new AutosmeltCommand("autosmelt"));
 
         MinecraftServer.getServer().server.getCommandMap().register("setspawn", "Deepslate", new SpawnCommand("setspawn"));
         MinecraftServer.getServer().server.getCommandMap().register("spawn", "Deepslate", new SpawnCommand("spawn"));
