From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Wed, 1 May 2024 20:19:00 +0200
Subject: [PATCH] Metrics stats


diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
index 42d31ab978af8586f86e1c819bb45a3018c03124..e7f1b7dbe5bcb659a899428e32c98100067adc0b 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
@@ -14,6 +14,7 @@ import java.nio.file.Path;
 import java.util.List;
 import java.util.Locale;
 import java.util.Optional;
+import java.util.UUID;
 import java.util.function.BooleanSupplier;
 import javax.annotation.Nullable;
 
@@ -55,6 +56,7 @@ import net.minecraft.world.level.GameRules;
 import net.minecraft.world.level.GameType;
 import net.minecraft.world.level.block.entity.SkullBlockEntity;
 import net.minecraft.world.level.storage.LevelStorageSource;
+import net.pascalpex.deepslatemc.Config;
 import net.pascalpex.deepslatemc.commands.*;
 import org.galemc.gale.command.GaleCommands;
 import org.slf4j.Logger;
@@ -238,6 +240,12 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
         io.papermc.paper.command.PaperCommands.registerCommands(this); // Paper - setup /paper command
         GaleCommands.registerCommands(this); // Gale - Gale commands - register commands
         com.destroystokyo.paper.Metrics.PaperMetrics.startMetrics(); // Paper - start metrics
+        // Deepslate start - Metrics stats
+        if(Config.getMetricsEnabled()) {
+            net.pascalpex.deepslatemc.util.Metrics metrics = new net.pascalpex.deepslatemc.util.Metrics(UUID.randomUUID());
+            metrics.startSending();
+        }
+        // Deepslate end
         com.destroystokyo.paper.VersionHistoryManager.INSTANCE.getClass(); // Paper - load version history now
 
         // Gale start - Pufferfish - SIMD support
diff --git a/src/main/java/net/pascalpex/deepslatemc/Config.java b/src/main/java/net/pascalpex/deepslatemc/Config.java
index b42ac2fb0bfca93aff0549c1409a27524af42501..907357b7cb92aabc5c1835f593246fef289c0483 100644
--- a/src/main/java/net/pascalpex/deepslatemc/Config.java
+++ b/src/main/java/net/pascalpex/deepslatemc/Config.java
@@ -51,6 +51,9 @@ public class Config {
                 worldsList.add("FantasyWorld");
                 config.set("autoSmelt" + ".enabledWorlds", worldsList);
             }
+            if (!config.contains("metrics")) {
+                config.set("metrics", true);
+            }
             save();
         } catch (IOException | InvalidConfigurationException e) {
             e.printStackTrace();
@@ -115,5 +118,8 @@ public class Config {
     public static List<String> getAutoSmeltWorlds() {
         return (List<String>) config.getList("autoSmelt" + ".enabledWorlds");
     }
+    public static boolean getMetricsEnabled() {
+        return config.getBoolean("metrics");
+    }
 
 }
diff --git a/src/main/java/net/pascalpex/deepslatemc/util/Metrics.java b/src/main/java/net/pascalpex/deepslatemc/util/Metrics.java
new file mode 100644
index 0000000000000000000000000000000000000000..72fff84c97d23e9bb11d048e33f25a689d583b24
--- /dev/null
+++ b/src/main/java/net/pascalpex/deepslatemc/util/Metrics.java
@@ -0,0 +1,44 @@
+package net.pascalpex.deepslatemc.util;
+
+import org.bukkit.Bukkit;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.UUID;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+
+public class Metrics {
+
+    UUID uuid;
+    Timer timer = new Timer();
+    private ScheduledExecutorService scheduler;
+
+    public Metrics(UUID uuid) {
+        this.uuid = uuid;
+        scheduler = Executors.newScheduledThreadPool(1);
+    }
+
+    public void startSending() {
+        timer.scheduleAtFixedRate(task, 10000, 300000);
+    }
+
+    TimerTask task = new TimerTask() {
+        @Override
+        public void run() {
+            int players = Bukkit.getOnlinePlayers().size();
+            String minecraftVersion = Bukkit.getVersion();
+            minecraftVersion = minecraftVersion.substring(minecraftVersion.indexOf("MC: ") + 4, minecraftVersion.length() - 1);
+            String deepslateVersion = this.getClass().getPackage().getImplementationVersion().replace('"', ' ').replace(" ", "");
+            String urlString = "https://pascalpex.de/metrics/deepslate/submit.php?uuid=" + uuid.toString() + "&players=" + players + "&mc=" + minecraftVersion + "&deepslate=" + deepslateVersion;
+            try {
+                URL url = new URL(urlString);
+                url.openConnection().getInputStream();
+            } catch (IOException ignored) {
+            }
+        }
+    };
+
+}
