From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Sun, 30 Jan 2022 14:44:31 +0100
Subject: [PATCH] Colored Signs


diff --git a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
index 04ef0bf3e0efd4ce416e1f111ead9aea6531ba6a..db6188e1673504b5efacfbaf699722e7edc52a10 100644
--- a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -151,6 +151,7 @@ import net.minecraft.world.phys.Vec3;
 import net.minecraft.world.phys.shapes.BooleanOp;
 import net.minecraft.world.phys.shapes.Shapes;
 import net.minecraft.world.phys.shapes.VoxelShape;
+import net.pascalpex.deepslatemc.Config;
 import org.apache.commons.lang3.StringUtils;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
@@ -3178,6 +3179,15 @@ public class ServerGamePacketListenerImpl implements ServerPlayerConnection, Ser
                 // Purpur end
             }
             SignChangeEvent event = new SignChangeEvent((org.bukkit.craftbukkit.block.CraftBlock) player.getWorld().getBlockAt(x, y, z), this.player.getBukkitEntity(), lines);
+            if(Config.getColoredSignsEnabled()) {
+                if(Config.getColoredSignsPermissionUse()) {
+                    if(player.hasPermission("deepslate.colorsigns")) {
+                        colorSign(event);
+                    }
+                } else {
+                    colorSign(event);
+                }
+            }
             this.cserver.getPluginManager().callEvent(event);
 
             if (!event.isCancelled()) {
@@ -3195,6 +3205,35 @@ public class ServerGamePacketListenerImpl implements ServerPlayerConnection, Ser
 
     }
 
+    public void colorSign(SignChangeEvent event) {
+        for(int i = 0; i < 4; i++) {
+            String line = event.getLines()[i];
+            line = line.replace("&0", "§0");
+            line = line.replace("&1", "§1");
+            line = line.replace("&2", "§2");
+            line = line.replace("&3", "§3");
+            line = line.replace("&4", "§4");
+            line = line.replace("&5", "§5");
+            line = line.replace("&6", "§6");
+            line = line.replace("&7", "§7");
+            line = line.replace("&8", "§8");
+            line = line.replace("&9", "§9");
+            line = line.replace("&a", "§a");
+            line = line.replace("&b", "§b");
+            line = line.replace("&c", "§c");
+            line = line.replace("&d", "§d");
+            line = line.replace("&e", "§e");
+            line = line.replace("&f", "§f");
+            line = line.replace("&k", "§k");
+            line = line.replace("&l", "§l");
+            line = line.replace("&m", "§m");
+            line = line.replace("&n", "§n");
+            line = line.replace("&o", "§o");
+            line = line.replace("&r", "§r");
+            event.setLine(i, line);
+        }
+    }
+
     @Override
     public void handleKeepAlive(ServerboundKeepAlivePacket packet) {
         // Purpur start
diff --git a/src/main/java/net/pascalpex/deepslatemc/Config.java b/src/main/java/net/pascalpex/deepslatemc/Config.java
index 191e8bedf91bc6d38951c0731b156cf0ed1e1755..e6dcc90c84ad72dd9efacd0d24ee98549bc700cb 100644
--- a/src/main/java/net/pascalpex/deepslatemc/Config.java
+++ b/src/main/java/net/pascalpex/deepslatemc/Config.java
@@ -33,6 +33,8 @@ public class Config {
                 config.set("helpMessage", message);
                 config.set("spawnOnJoin", false);
                 config.set("preventKnowledgebookClick", false);
+                config.set("coloredSigns" + ".enabled", false);
+                config.set("coloredSigns" + ".usePermission", true);
                 save();
             }
             config.load(configFile);
@@ -49,6 +51,13 @@ public class Config {
         }
     }
 
+    public static boolean getColoredSignsEnabled() {
+        return config.getBoolean("coloredSigns" + ".enabled");
+    }
+
+    public static boolean getColoredSignsPermissionUse() {
+        return config.getBoolean("coloredSigns" + ".usePermission");
+    }
 
     public static boolean getSpawnOnJoin() {
         return config.getBoolean("spawnOnJoin");
