From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Wed, 1 May 2024 20:23:20 +0200
Subject: [PATCH] Configurable move messages


diff --git a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
index c41987b5995e27c9b1f8bb2e8af341315c6f338c..62bec65a609997c027411498c650d8868a3198ff 100644
--- a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -195,6 +195,7 @@ import net.minecraft.world.phys.Vec3;
 import net.minecraft.world.phys.shapes.BooleanOp;
 import net.minecraft.world.phys.shapes.Shapes;
 import net.minecraft.world.phys.shapes.VoxelShape;
+import net.pascalpex.deepslatemc.Config;
 import org.bukkit.craftbukkit.util.permissions.CraftDefaultPermissions;
 import org.apache.commons.lang3.StringUtils;
 import org.galemc.gale.configuration.GaleGlobalConfiguration;
@@ -1446,7 +1447,7 @@ public class ServerGamePacketListenerImpl extends ServerCommonPacketListenerImpl
                                         io.papermc.paper.event.player.PlayerFailMoveEvent event = fireFailMove(io.papermc.paper.event.player.PlayerFailMoveEvent.FailReason.MOVED_TOO_QUICKLY,
                                             toX, toY, toZ, toYaw, toPitch, true);
                                         if (!event.isAllowed()) {
-                                            if (!org.dreeam.leaf.config.modules.gameplay.DisableMovedWronglyThreshold.enabled && event.getLogWarning()) // Leaves - disable can
+                                            if (Config.getMovedQuickly() && !org.dreeam.leaf.config.modules.gameplay.DisableMovedWronglyThreshold.enabled && event.getLogWarning()) // Leaves - disable can // Deepslate
                                                 ServerGamePacketListenerImpl.LOGGER.warn("{} moved too quickly! {},{},{}", new Object[]{this.player.getName().getString(), d6, d7, d8});
                                             this.teleport(this.player.getX(), this.player.getY(), this.player.getZ(), this.player.getYRot(), this.player.getXRot());
                                             return;
@@ -1521,7 +1522,7 @@ public class ServerGamePacketListenerImpl extends ServerCommonPacketListenerImpl
                                     toX, toY, toZ, toYaw, toPitch, true);
                                 if (!event.isAllowed()) {
                                     movedWrongly = true;
-                                    if (event.getLogWarning())
+                                    if (event.getLogWarning() && Config.getMovedWrongly()) // Deepslate
                                 // Paper end
                                 ServerGamePacketListenerImpl.LOGGER.warn("{} moved wrongly!, ({})", this.player.getName().getString(), d11); // Purpur
                                 } // Paper
diff --git a/src/main/java/net/pascalpex/deepslatemc/Config.java b/src/main/java/net/pascalpex/deepslatemc/Config.java
index b42ac2fb0bfca93aff0549c1409a27524af42501..d4b1b3099ea2bade572ad4f88405f8b72f4cf8ce 100644
--- a/src/main/java/net/pascalpex/deepslatemc/Config.java
+++ b/src/main/java/net/pascalpex/deepslatemc/Config.java
@@ -51,6 +51,10 @@ public class Config {
                 worldsList.add("FantasyWorld");
                 config.set("autoSmelt" + ".enabledWorlds", worldsList);
             }
+            if (!config.contains("messages")) {
+                config.set("messages" + ".movedWrongly", true);
+                config.set("messages" + ".movedQuickly", true);
+            }
             save();
         } catch (IOException | InvalidConfigurationException e) {
             e.printStackTrace();
@@ -115,5 +119,11 @@ public class Config {
     public static List<String> getAutoSmeltWorlds() {
         return (List<String>) config.getList("autoSmelt" + ".enabledWorlds");
     }
+    public static boolean getMovedQuickly() {
+        return config.getBoolean("messages" + ".movedQuickly");
+    }
+    public static boolean getMovedWrongly() {
+        return config.getBoolean("messages" + ".movedWrongly");
+    }
 
 }
