From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Mon, 21 Mar 2022 15:55:41 +0100
Subject: [PATCH] Metrics stats


diff --git a/src/main/java/com/destroystokyo/paper/PaperConfig.java b/src/main/java/com/destroystokyo/paper/PaperConfig.java
index 19d7ea19f91bb2ef94187d6db91f26a93a6d0be2..a64019984928df84aaf5d1d897ceb9c5796dc1dc 100644
--- a/src/main/java/com/destroystokyo/paper/PaperConfig.java
+++ b/src/main/java/com/destroystokyo/paper/PaperConfig.java
@@ -13,12 +13,14 @@ import java.nio.charset.StandardCharsets;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.UUID;
 import java.util.concurrent.TimeUnit;
 import java.util.logging.Level;
 import java.util.regex.Pattern;
 
 import com.google.common.collect.Lists;
 import net.minecraft.server.MinecraftServer;
+import net.pascalpex.deepslatemc.Config;
 import org.bukkit.Bukkit;
 import org.bukkit.ChatColor;
 import org.bukkit.command.Command;
@@ -127,6 +129,14 @@ public class PaperConfig {
             Metrics.PaperMetrics.startMetrics();
             metricsStarted = true;
         }
+
+        // Deepslate start - Metrics stats
+        if(Config.getMetricsEnabled()) {
+            net.pascalpex.deepslatemc.util.Metrics metrics = new net.pascalpex.deepslatemc.util.Metrics(UUID.randomUUID());
+            metrics.startSending();
+        }
+        // Deepslate end
+
     }
 
     static void readConfig(Class<?> clazz, Object instance) {
diff --git a/src/main/java/net/pascalpex/deepslatemc/Config.java b/src/main/java/net/pascalpex/deepslatemc/Config.java
index 462ac1f4b48ac54d58c92f79a3c15b50e915071a..ba80d45928ff1c25f213b4ebf1fd72c527c7d70e 100644
--- a/src/main/java/net/pascalpex/deepslatemc/Config.java
+++ b/src/main/java/net/pascalpex/deepslatemc/Config.java
@@ -51,6 +51,9 @@ public class Config {
                     worldsList.add("FantasyWorld");
                     config.set("autoSmelt" + ".enabledWorlds", worldsList);
                 }
+                if (!config.contains("metrics")) {
+                    config.set("metrics", true);
+                }
                 save();
             } catch (IOException | InvalidConfigurationException e) {
                 e.printStackTrace();
@@ -65,6 +68,10 @@ public class Config {
             }
         }
 
+        public static boolean getMetricsEnabled() {
+            return config.getBoolean("metrics");
+        }
+
         public static int getAutoSmeltPrice() {
             return config.getInt("autoSmelt" + ".price");
         }
diff --git a/src/main/java/net/pascalpex/deepslatemc/util/Metrics.java b/src/main/java/net/pascalpex/deepslatemc/util/Metrics.java
new file mode 100644
index 0000000000000000000000000000000000000000..136d30d7e25895adf5be857aeac94c46cb330ec0
--- /dev/null
+++ b/src/main/java/net/pascalpex/deepslatemc/util/Metrics.java
@@ -0,0 +1,43 @@
+package net.pascalpex.deepslatemc.util;
+    
+import org.bukkit.Bukkit;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.UUID;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+
+public class Metrics {
+
+    UUID uuid;
+    Timer timer = new Timer();
+    private ScheduledExecutorService scheduler;
+
+    public Metrics(UUID uuid) {
+        this.uuid = uuid;
+        scheduler = Executors.newScheduledThreadPool(1);
+    }
+
+    public void startSending() {
+        timer.scheduleAtFixedRate(task, 10000, 300000);
+    }
+
+    TimerTask task = new TimerTask() {
+        @Override
+        public void run() {
+            int players = Bukkit.getOnlinePlayers().size();
+            String minecraftVersion = Bukkit.getVersion();
+            minecraftVersion = minecraftVersion.substring(minecraftVersion.indexOf("MC: ") + 4, minecraftVersion.length() - 1);
+            String deepslateVersion = this.getClass().getPackage().getImplementationVersion().replace('"', ' ').replace(" ", "");
+            String urlString = "https://pascalpex.ddns.net/metrics/deepslate/submit.php?uuid=" + uuid.toString() + "&players=" + players + "&mc=" + minecraftVersion + "&deepslate=" + deepslateVersion;
+            try {
+                URL url = new URL(urlString);
+                url.openConnection().getInputStream();
+            } catch (IOException ignored) {}
+        }
+    };
+
+}
