From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Wed, 1 May 2024 20:23:20 +0200
Subject: [PATCH] Configurable move messages


diff --git a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
index 054c9ae84ca504db7b1b660d7e8f589c1f9a4dae..0bc5c3cf85961f6a17d85f745e9c35e239852d30 100644
--- a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -195,6 +195,7 @@ import net.minecraft.world.phys.Vec3;
 import net.minecraft.world.phys.shapes.BooleanOp;
 import net.minecraft.world.phys.shapes.Shapes;
 import net.minecraft.world.phys.shapes.VoxelShape;
+import net.pascalpex.deepslatemc.Config;
 import org.bukkit.craftbukkit.util.permissions.CraftDefaultPermissions;
 import org.apache.commons.lang3.StringUtils;
 import org.galemc.gale.configuration.GaleGlobalConfiguration;
@@ -1387,7 +1388,7 @@ public class ServerGamePacketListenerImpl extends ServerCommonPacketListenerImpl
                                         io.papermc.paper.event.player.PlayerFailMoveEvent event = fireFailMove(io.papermc.paper.event.player.PlayerFailMoveEvent.FailReason.MOVED_TOO_QUICKLY,
                                             toX, toY, toZ, toYaw, toPitch, true);
                                         if (!event.isAllowed()) {
-                                            if (event.getLogWarning())
+                                            if (Config.getMovedQuickly() && event.getLogWarning()) // Deepslate
                                                 ServerGamePacketListenerImpl.LOGGER.warn("{} moved too quickly! {},{},{}", new Object[]{this.player.getName().getString(), d6, d7, d8});
                                             this.teleport(this.player.getX(), this.player.getY(), this.player.getZ(), this.player.getYRot(), this.player.getXRot());
                                             return;
@@ -1462,7 +1463,7 @@ public class ServerGamePacketListenerImpl extends ServerCommonPacketListenerImpl
                                     toX, toY, toZ, toYaw, toPitch, true);
                                 if (!event.isAllowed()) {
                                     movedWrongly = true;
-                                    if (event.getLogWarning())
+                                    if (event.getLogWarning() && Config.getMovedWrongly()) // Deepslate
                                 // Paper end
                                 ServerGamePacketListenerImpl.LOGGER.warn("{} moved wrongly!", this.player.getName().getString());
                                 } // Paper
diff --git a/src/main/java/net/pascalpex/deepslatemc/Config.java b/src/main/java/net/pascalpex/deepslatemc/Config.java
index 907357b7cb92aabc5c1835f593246fef289c0483..2ac86918a792cafc877efd3ab4d8f8d5c10e91b7 100644
--- a/src/main/java/net/pascalpex/deepslatemc/Config.java
+++ b/src/main/java/net/pascalpex/deepslatemc/Config.java
@@ -54,6 +54,10 @@ public class Config {
             if (!config.contains("metrics")) {
                 config.set("metrics", true);
             }
+            if (!config.contains("messages")) {
+                config.set("messages" + ".movedWrongly", true);
+                config.set("messages" + ".movedQuickly", true);
+            }
             save();
         } catch (IOException | InvalidConfigurationException e) {
             e.printStackTrace();
@@ -121,5 +125,11 @@ public class Config {
     public static boolean getMetricsEnabled() {
         return config.getBoolean("metrics");
     }
+    public static boolean getMovedQuickly() {
+        return config.getBoolean("messages" + ".movedQuickly");
+    }
+    public static boolean getMovedWrongly() {
+        return config.getBoolean("messages" + ".movedWrongly");
+    }
 
 }
