From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Mon, 25 Dec 2023 15:35:26 +0100
Subject: [PATCH] Configurable move messages


diff --git a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
index 1acd2580294af4e7f07a5c5e767d8e3093efca17..e6e4ff696eaa8faac9a4f4f9342fa0cdce20f467 100644
--- a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -189,6 +189,7 @@ import net.minecraft.world.phys.Vec3;
 import net.minecraft.world.phys.shapes.BooleanOp;
 import net.minecraft.world.phys.shapes.Shapes;
 import net.minecraft.world.phys.shapes.VoxelShape;
+import net.pascalpex.deepslatemc.Config;
 import org.bukkit.craftbukkit.util.permissions.CraftDefaultPermissions;
 import org.apache.commons.lang3.StringUtils;
 import org.galemc.gale.configuration.GaleGlobalConfiguration;
@@ -1489,7 +1490,7 @@ public class ServerGamePacketListenerImpl extends ServerCommonPacketListenerImpl
                                         io.papermc.paper.event.player.PlayerFailMoveEvent event = fireFailMove(io.papermc.paper.event.player.PlayerFailMoveEvent.FailReason.MOVED_TOO_QUICKLY,
                                             toX, toY, toZ, toYaw, toPitch, true);
                                         if (!event.isAllowed()) {
-                                            if (event.getLogWarning())
+                                            if (event.getLogWarning() && Config.getMovedQuickly()) // Deepslate
                                                 ServerGamePacketListenerImpl.LOGGER.warn("{} moved too quickly! {},{},{}", new Object[]{this.player.getName().getString(), d6, d7, d8});
                                             this.teleport(this.player.getX(), this.player.getY(), this.player.getZ(), this.player.getYRot(), this.player.getXRot());
                                             return;
@@ -1565,7 +1566,7 @@ public class ServerGamePacketListenerImpl extends ServerCommonPacketListenerImpl
                                     toX, toY, toZ, toYaw, toPitch, true);
                                 if (!event.isAllowed()) {
                                 flag2 = true; // Paper - diff on change, this should be moved wrongly
-                                    if (event.getLogWarning())
+                                    if (event.getLogWarning() && Config.getMovedWrongly()) // Deepslate
                                         ServerGamePacketListenerImpl.LOGGER.warn("{} moved wrongly!, ({})", this.player.getName().getString(), d11); // Purpur
                                 }
                                 // Paper end
diff --git a/src/main/java/net/pascalpex/deepslatemc/Config.java b/src/main/java/net/pascalpex/deepslatemc/Config.java
index f3346fcd2b6786218df690a31470a58c5d673ff0..46b696f83a80f7eef9d92ace646d55ef6165d726 100644
--- a/src/main/java/net/pascalpex/deepslatemc/Config.java
+++ b/src/main/java/net/pascalpex/deepslatemc/Config.java
@@ -51,6 +51,10 @@ public class Config {
             if (!config.contains("metrics")) {
                 config.set("metrics", true);
             }
+            if (!config.contains("messages")) {
+                config.set("messages" + ".movedWrongly", true);
+                config.set("messages" + ".movedQuickly", true);
+            }
             save();
         } catch (IOException | InvalidConfigurationException e) {
             e.printStackTrace();
@@ -111,5 +115,11 @@ public class Config {
     public static boolean getMetricsEnabled() {
         return config.getBoolean("metrics");
     }
+    public static boolean getMovedQuickly() {
+        return config.getBoolean("messages" + ".movedQuickly");
+    }
+    public static boolean getMovedWrongly() {
+        return config.getBoolean("messages" + ".movedWrongly");
+    }
 
 }
