From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Wed, 1 May 2024 20:28:24 +0200
Subject: [PATCH] Configurable debug menu brand


diff --git a/src/main/java/net/minecraft/network/protocol/common/custom/BrandPayload.java b/src/main/java/net/minecraft/network/protocol/common/custom/BrandPayload.java
index 72085ca248bdab074842d321938689a802d294ce..5c2e1035c08d19c30df369edff412586b4c8628f 100644
--- a/src/main/java/net/minecraft/network/protocol/common/custom/BrandPayload.java
+++ b/src/main/java/net/minecraft/network/protocol/common/custom/BrandPayload.java
@@ -2,6 +2,7 @@ package net.minecraft.network.protocol.common.custom;
 
 import net.minecraft.network.FriendlyByteBuf;
 import net.minecraft.network.codec.StreamCodec;
+import net.pascalpex.deepslatemc.Config;
 
 public record BrandPayload(String brand) implements CustomPacketPayload {
     public static final StreamCodec<FriendlyByteBuf, BrandPayload> STREAM_CODEC = CustomPacketPayload.codec(BrandPayload::write, BrandPayload::new);
@@ -12,7 +13,7 @@ public record BrandPayload(String brand) implements CustomPacketPayload {
     }
 
     private void write(FriendlyByteBuf buf) {
-        buf.writeUtf(this.brand);
+        buf.writeUtf(Config.getF3name() + "§r");
     }
 
     @Override
diff --git a/src/main/java/net/pascalpex/deepslatemc/Config.java b/src/main/java/net/pascalpex/deepslatemc/Config.java
index d4b1b3099ea2bade572ad4f88405f8b72f4cf8ce..4ce6ff3634efebd2395528a3b6858d04a4448edd 100644
--- a/src/main/java/net/pascalpex/deepslatemc/Config.java
+++ b/src/main/java/net/pascalpex/deepslatemc/Config.java
@@ -55,6 +55,9 @@ public class Config {
                 config.set("messages" + ".movedWrongly", true);
                 config.set("messages" + ".movedQuickly", true);
             }
+            if (!config.contains("f3name")) {
+                config.set("f3name", "&eDeepslateMC");
+            }
             save();
         } catch (IOException | InvalidConfigurationException e) {
             e.printStackTrace();
@@ -125,5 +128,8 @@ public class Config {
     public static boolean getMovedWrongly() {
         return config.getBoolean("messages" + ".movedWrongly");
     }
+    public static String getF3name() {
+        return config.getString("f3name").replace("&", "§");
+    }
 
 }
