From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Tue, 22 Mar 2022 11:43:26 +0100
Subject: [PATCH] Temp fix build


diff --git a/build.gradle.kts b/build.gradle.kts
index e3f0b92063ed1cea1f9663a68e25f65586da28d2..5a60a95de8eb326fd50c80aee802c4769b5c7628 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -59,7 +59,7 @@ dependencies {
 
     // Pufferfish start
     implementation("org.yaml:snakeyaml:1.28")
-    implementation ("me.carleslc.Simple-YAML:Simple-Yaml:ed30ff3e6b") {
+    implementation ("me.carleslc.Simple-YAML:Simple-Yaml:1.8") {
         exclude(group="org.yaml", module="snakeyaml")
     }
     // Pufferfish end
