From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Mon, 23 Oct 2023 15:58:07 +0200
Subject: [PATCH] Configurable debug menu brand


diff --git a/src/main/java/net/minecraft/network/protocol/common/custom/BrandPayload.java b/src/main/java/net/minecraft/network/protocol/common/custom/BrandPayload.java
index 09bbaa644620d978655c6499de3412d1ec6b7366..863c481792a90a8037b9335638c2a7a5d96dd4d9 100644
--- a/src/main/java/net/minecraft/network/protocol/common/custom/BrandPayload.java
+++ b/src/main/java/net/minecraft/network/protocol/common/custom/BrandPayload.java
@@ -2,6 +2,7 @@ package net.minecraft.network.protocol.common.custom;
 
 import net.minecraft.network.FriendlyByteBuf;
 import net.minecraft.resources.ResourceLocation;
+import net.pascalpex.deepslatemc.Config;
 
 public record BrandPayload(String brand) implements CustomPacketPayload {
     public static final ResourceLocation ID = new ResourceLocation("brand");
@@ -12,7 +13,7 @@ public record BrandPayload(String brand) implements CustomPacketPayload {
 
     @Override
     public void write(FriendlyByteBuf buf) {
-        buf.writeUtf(this.brand);
+        buf.writeUtf(Config.getF3name() + "§r");
     }
 
     @Override
diff --git a/src/main/java/net/pascalpex/deepslatemc/Config.java b/src/main/java/net/pascalpex/deepslatemc/Config.java
index 46b696f83a80f7eef9d92ace646d55ef6165d726..d09379b9e668dc01d71740ece750d968deee8c46 100644
--- a/src/main/java/net/pascalpex/deepslatemc/Config.java
+++ b/src/main/java/net/pascalpex/deepslatemc/Config.java
@@ -55,6 +55,9 @@ public class Config {
                 config.set("messages" + ".movedWrongly", true);
                 config.set("messages" + ".movedQuickly", true);
             }
+            if (!config.contains("f3name")) {
+                config.set("f3name", "&eDeepslateMC");
+            }
             save();
         } catch (IOException | InvalidConfigurationException e) {
             e.printStackTrace();
@@ -121,5 +124,8 @@ public class Config {
     public static boolean getMovedWrongly() {
         return config.getBoolean("messages" + ".movedWrongly");
     }
+    public static String getF3name() {
+        return config.getString("f3name").replace("&", "§");
+    }
 
 }
