From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Sun, 22 May 2022 13:50:56 +0200
Subject: [PATCH] Configurable move messages


diff --git a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
index be54adb6a435e9fb93d48e863de32443274b25d2..09f60a5be961e0fe196e285003df1d071d0b85f9 100644
--- a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -1478,7 +1478,9 @@ public class ServerGamePacketListenerImpl implements ServerPlayerConnection, Ser
 
                                 if (d11 - d10 > Math.max(f2, Math.pow((double) (org.spigotmc.SpigotConfig.movedTooQuicklyMultiplier * (float) i * speed), 2)) && !this.isSingleplayerOwner()) {
                                 // CraftBukkit end
-                                    ServerGamePacketListenerImpl.LOGGER.warn("{} moved too quickly! {},{},{}", new Object[]{this.player.getName().getString(), d7, d8, d9});
+                                    if (Config.getMovedQuickly()) { // Deepslate
+                                        ServerGamePacketListenerImpl.LOGGER.warn("{} moved too quickly! {},{},{}", new Object[]{this.player.getName().getString(), d7, d8, d9});
+                                    }
                                     this.teleport(this.player.getX(), this.player.getY(), this.player.getZ(), this.player.getYRot(), this.player.getXRot());
                                     return;
                                 }
@@ -1546,7 +1548,9 @@ public class ServerGamePacketListenerImpl implements ServerPlayerConnection, Ser
 
                             if (!this.player.isChangingDimension() && d11 > org.spigotmc.SpigotConfig.movedWronglyThreshold && !this.player.isSleeping() && !this.player.gameMode.isCreative() && this.player.gameMode.getGameModeForPlayer() != GameType.SPECTATOR) { // Spigot
                                 flag2 = true; // Paper - diff on change, this should be moved wrongly
-                                ServerGamePacketListenerImpl.LOGGER.warn("{} moved wrongly!, ({})", this.player.getName().getString(), d11); // Purpur
+                                if (Config.getMovedWrongly()) { // Deepslate
+                                    ServerGamePacketListenerImpl.LOGGER.warn("{} moved wrongly!, ({})", this.player.getName().getString(), d11); // Purpur
+                                }
                             }
 
                             this.player.absMoveTo(d0, d1, d2, f, f1);
diff --git a/src/main/java/net/pascalpex/deepslatemc/Config.java b/src/main/java/net/pascalpex/deepslatemc/Config.java
index 8d2ece0c1cb3474483a65b03a2a6349ecac7902f..e794bd569fc9ecf4fc34500d48a00ef9c2181727 100644
--- a/src/main/java/net/pascalpex/deepslatemc/Config.java
+++ b/src/main/java/net/pascalpex/deepslatemc/Config.java
@@ -54,6 +54,10 @@ public class Config {
             if (!config.contains("metrics")) {
                 config.set("metrics", true);
             }
+                if (!config.contains("messages")) {
+                    config.set("messages" + ".movedWrongly", true);
+                    config.set("messages" + ".movedQuickly", true);
+                }
                 save();
         } catch (IOException | InvalidConfigurationException e) {
             e.printStackTrace();
@@ -133,4 +137,12 @@ public class Config {
             save();
         }
 
+    public static boolean getMovedQuickly() {
+        return config.getBoolean("messages" + ".movedQuickly");
+    }
+
+    public static boolean getMovedWrongly() {
+        return config.getBoolean("messages" + ".movedWrongly");
+    }
+
 }
