From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Mon, 23 Oct 2023 16:03:28 +0200
Subject: [PATCH] Tablist header footer


diff --git a/src/main/java/net/minecraft/server/players/PlayerList.java b/src/main/java/net/minecraft/server/players/PlayerList.java
index cadf652ab1568a92594bdc361b6e062c03e1ac2f..86f41480ac23c86ccc20a7ecc8f0ffca7a9b38cd 100644
--- a/src/main/java/net/minecraft/server/players/PlayerList.java
+++ b/src/main/java/net/minecraft/server/players/PlayerList.java
@@ -102,9 +102,11 @@ import net.minecraft.world.scores.Objective;
 import net.minecraft.world.scores.PlayerTeam;
 import net.minecraft.world.scores.Scoreboard; // Paper
 import net.minecraft.world.scores.Team;
+import net.pascalpex.deepslatemc.Config;
 import net.pascalpex.deepslatemc.MessagesFile;
 import net.pascalpex.deepslatemc.commands.MaintenanceMode;
 import net.pascalpex.deepslatemc.commands.SpawnCommand;
+import net.pascalpex.deepslatemc.util.TablistUtil;
 import org.slf4j.Logger;
 
 // CraftBukkit start
@@ -326,6 +328,11 @@ public abstract class PlayerList {
 
         PlayerJoinEvent playerJoinEvent = new PlayerJoinEvent(bukkitPlayer, io.papermc.paper.adventure.PaperAdventure.asAdventure(ichatmutablecomponent)); // Paper - Adventure
         SpawnCommand.playerJoin(bukkitPlayer); // Deepslate - Spawn functions
+        // Deepslate start - Tablist
+        if (Config.getTablistEnabled()) {
+            TablistUtil.setTablist(bukkitPlayer);
+        }
+        // Deepslate end
         this.cserver.getPluginManager().callEvent(playerJoinEvent);
 
         if (!player.connection.isAcceptingMessages()) {
diff --git a/src/main/java/net/pascalpex/deepslatemc/Config.java b/src/main/java/net/pascalpex/deepslatemc/Config.java
index 1d4f094b0601c2167f5a5df51e14bc2aad9ec796..0cb2de6d8db888b85d563bd7031efcb54669d43d 100644
--- a/src/main/java/net/pascalpex/deepslatemc/Config.java
+++ b/src/main/java/net/pascalpex/deepslatemc/Config.java
@@ -58,6 +58,17 @@ public class Config {
             if (!config.contains("f3name")) {
                 config.set("f3name", "&eDeepslateMC");
             }
+            if (!config.contains("tablist")) {
+                config.set("tablist" + ".enabled", false);
+                List<String> headerList = new ArrayList<>();
+                List<String> footerList = new ArrayList<>();
+                headerList.add("&4This is the");
+                headerList.add("&4default header");
+                footerList.add("&6This is the");
+                footerList.add("&6default footer");
+                config.set("tablist" + ".header", headerList);
+                config.set("tablist" + ".footer", footerList);
+            }
             save();
         } catch (IOException | InvalidConfigurationException e) {
             e.printStackTrace();
@@ -129,5 +140,14 @@ public class Config {
     public static String getF3name() {
         return config.getString("f3name").replace("&", "§");
     }
+    public static boolean getTablistEnabled() {
+        return config.getBoolean("tablist" + ".enabled");
+    }
+    public static List<String> getHeader() {
+        return (List<String>) config.getList("tablist" + ".header");
+    }
+    public static List<String> getFooter() {
+        return (List<String>) config.getList("tablist" + ".footer");
+    }
 
 }
diff --git a/src/main/java/net/pascalpex/deepslatemc/commands/DeepslateCommand.java b/src/main/java/net/pascalpex/deepslatemc/commands/DeepslateCommand.java
index c01afb333b667c92275028ecf9e2c77a9e2f96b1..250ae07f0b22e2d9739623ea8691e4aa5b9ff914 100644
--- a/src/main/java/net/pascalpex/deepslatemc/commands/DeepslateCommand.java
+++ b/src/main/java/net/pascalpex/deepslatemc/commands/DeepslateCommand.java
@@ -2,6 +2,7 @@ package net.pascalpex.deepslatemc.commands;
     
 import net.pascalpex.deepslatemc.Config;
 import net.pascalpex.deepslatemc.MessagesFile;
+import net.pascalpex.deepslatemc.util.TablistUtil;
 import org.bukkit.ChatColor;
 import org.bukkit.Location;
 import org.bukkit.command.Command;
@@ -45,6 +46,9 @@ public class DeepslateCommand extends Command {
                         if(args[0].equalsIgnoreCase("reload")) {
                             MessagesFile.load();
                             Config.load();
+                            if (Config.getTablistEnabled()) {
+                                TablistUtil.reloadTablist();
+                            }
                             player.sendMessage(prefix + MessagesFile.getConfigReloaded());
                         } else {
                             if(args[0].equalsIgnoreCase("version")) {
diff --git a/src/main/java/net/pascalpex/deepslatemc/util/TablistUtil.java b/src/main/java/net/pascalpex/deepslatemc/util/TablistUtil.java
new file mode 100644
index 0000000000000000000000000000000000000000..c6df5cc5ce25c7edfbaaea3c24cc661dfca3cc84
--- /dev/null
+++ b/src/main/java/net/pascalpex/deepslatemc/util/TablistUtil.java
@@ -0,0 +1,46 @@
+package net.pascalpex.deepslatemc.util;
+    
+import net.pascalpex.deepslatemc.Config;
+import org.bukkit.Bukkit;
+import org.bukkit.entity.Player;
+
+public class TablistUtil {
+
+    private static boolean initiated = false;
+    private static String header = "";
+    private static String footer = "";
+
+    public static void setTablist(Player player) {
+        if(!initiated) {
+            reloadTablist();
+        }
+        player.setPlayerListHeader(header);
+        player.setPlayerListFooter(footer);
+    }
+
+    public static void reloadTablist() {
+        header = "";
+        footer = "";
+        for(String s : Config.getHeader()) {
+            if(!header.equals("")) {
+                header += "\n";
+            }
+            header += s;
+        }
+        for(String s : Config.getFooter()) {
+            if(!footer.equals("")) {
+                footer += "\n";
+            }
+            footer += s;
+        }
+        header = header.replace("&", "§");
+        footer = footer.replace("&", "§");
+
+        initiated = true;
+        
+        for(Player player : Bukkit.getOnlinePlayers()) {
+            setTablist(player);
+        }
+    }
+
+}
