From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Mon, 25 Dec 2023 15:56:49 +0100
Subject: [PATCH] Tablist header footer


diff --git a/src/main/java/net/minecraft/server/commands/ReloadCommand.java b/src/main/java/net/minecraft/server/commands/ReloadCommand.java
index c020c86194723a5c89816f91e0b7c5eeaf132b7e..68d318c71632b853d51649083b3445b3e7a1cc2e 100644
--- a/src/main/java/net/minecraft/server/commands/ReloadCommand.java
+++ b/src/main/java/net/minecraft/server/commands/ReloadCommand.java
@@ -11,6 +11,7 @@ import net.minecraft.network.chat.Component;
 import net.minecraft.server.MinecraftServer;
 import net.minecraft.server.packs.repository.PackRepository;
 import net.minecraft.world.level.storage.WorldData;
+import net.pascalpex.deepslatemc.util.TablistUtil;
 import org.slf4j.Logger;
 
 public class ReloadCommand {
@@ -51,6 +52,7 @@ public class ReloadCommand {
         Collection<String> collection = resourcepackrepository.getSelectedIds();
         Collection<String> collection1 = ReloadCommand.discoverNewPacks(resourcepackrepository, savedata, collection);
         minecraftserver.reloadResources(collection1, io.papermc.paper.event.server.ServerResourcesReloadedEvent.Cause.PLUGIN); // Paper - Add ServerResourcesReloadedEvent
+        TablistUtil.reloadTablist();
     }
     // CraftBukkit end
 
diff --git a/src/main/java/net/minecraft/server/players/PlayerList.java b/src/main/java/net/minecraft/server/players/PlayerList.java
index 11a1c3a641e6711eac73df1feb00ebf35378a37b..14da1a4effeaf9e317563f2e1e87892eb273e0b3 100644
--- a/src/main/java/net/minecraft/server/players/PlayerList.java
+++ b/src/main/java/net/minecraft/server/players/PlayerList.java
@@ -109,6 +109,7 @@ import net.minecraft.world.scores.PlayerTeam;
 import net.pascalpex.deepslatemc.Config;
 import net.pascalpex.deepslatemc.MessagesFile;
 import net.pascalpex.deepslatemc.commands.SpawnCommand;
+import net.pascalpex.deepslatemc.util.TablistUtil;
 import org.galemc.gale.configuration.GaleGlobalConfiguration;
 import org.slf4j.Logger;
 
@@ -363,7 +364,11 @@ public abstract class PlayerList {
         PlayerJoinEvent playerJoinEvent = new PlayerJoinEvent(bukkitPlayer, getJoinMsg(io.papermc.paper.adventure.PaperAdventure.asAdventure(ichatmutablecomponent), bukkitPlayer)); // Paper - Adventure // Leaf - Configurable connection message - join message
 
         SpawnCommand.playerJoin(bukkitPlayer); // Deepslate - Spawn functions
-
+        // Deepslate start - Tablist
+        if (Config.getTablistEnabled()) {
+            TablistUtil.setTablist(bukkitPlayer);
+        }
+        // Deepslate end
         this.cserver.getPluginManager().callEvent(playerJoinEvent);
 
         if (!player.connection.isAcceptingMessages()) {
diff --git a/src/main/java/net/pascalpex/deepslatemc/Config.java b/src/main/java/net/pascalpex/deepslatemc/Config.java
index 94e629c3d977e8f15cc5a3912feeed97ea02c839..cc38e0b87f4b78666372345140bc36d146f10c30 100644
--- a/src/main/java/net/pascalpex/deepslatemc/Config.java
+++ b/src/main/java/net/pascalpex/deepslatemc/Config.java
@@ -61,6 +61,17 @@ public class Config {
             if (!config.contains("f3name")) {
                 config.set("f3name", "&eDeepslateMC");
             }
+            if (!config.contains("tablist")) {
+                config.set("tablist" + ".enabled", false);
+                List<String> headerList = new ArrayList<>();
+                List<String> footerList = new ArrayList<>();
+                headerList.add("&4This is the");
+                headerList.add("&4default header");
+                footerList.add("&6This is the");
+                footerList.add("&6default footer");
+                config.set("tablist" + ".header", headerList);
+                config.set("tablist" + ".footer", footerList);
+            }
             save();
         } catch (IOException | InvalidConfigurationException e) {
             e.printStackTrace();
@@ -137,5 +148,14 @@ public class Config {
     public static String getF3name() {
         return config.getString("f3name").replace("&", "§");
     }
+    public static boolean getTablistEnabled() {
+        return config.getBoolean("tablist" + ".enabled");
+    }
+    public static List<String> getHeader() {
+        return (List<String>) config.getList("tablist" + ".header");
+    }
+    public static List<String> getFooter() {
+        return (List<String>) config.getList("tablist" + ".footer");
+    }
 
 }
diff --git a/src/main/java/net/pascalpex/deepslatemc/commands/DeepslateCommand.java b/src/main/java/net/pascalpex/deepslatemc/commands/DeepslateCommand.java
index c11d3597dbc49a1f6d75a55f427c55fc55d68a29..74f9bccd6bbf8fb8baedf1f91c70959365cf61b8 100644
--- a/src/main/java/net/pascalpex/deepslatemc/commands/DeepslateCommand.java
+++ b/src/main/java/net/pascalpex/deepslatemc/commands/DeepslateCommand.java
@@ -2,6 +2,7 @@ package net.pascalpex.deepslatemc.commands;
 
 import net.pascalpex.deepslatemc.Config;
 import net.pascalpex.deepslatemc.MessagesFile;
+import net.pascalpex.deepslatemc.util.TablistUtil;
 import org.bukkit.ChatColor;
 import org.bukkit.Location;
 import org.bukkit.command.Command;
@@ -45,6 +46,7 @@ public class DeepslateCommand extends Command {
                         if (args[0].equalsIgnoreCase("reload")) {
                             MessagesFile.load();
                             Config.load();
+                            TablistUtil.reloadTablist();
                             player.sendMessage(prefix + MessagesFile.getConfigReloaded());
                         } else {
                             if (args[0].equalsIgnoreCase("version")) {
@@ -69,6 +71,7 @@ public class DeepslateCommand extends Command {
                     if (args[0].equalsIgnoreCase("reload")) {
                         MessagesFile.load();
                         Config.load();
+                        TablistUtil.reloadTablist();
                         sender.sendMessage(prefix + MessagesFile.getConfigReloaded());
                     } else {
                         if (args[0].equalsIgnoreCase("version")) {
diff --git a/src/main/java/net/pascalpex/deepslatemc/util/TablistUtil.java b/src/main/java/net/pascalpex/deepslatemc/util/TablistUtil.java
new file mode 100644
index 0000000000000000000000000000000000000000..6263eb8b7b751deace9b8c58a9d4280a93eab0b3
--- /dev/null
+++ b/src/main/java/net/pascalpex/deepslatemc/util/TablistUtil.java
@@ -0,0 +1,56 @@
+package net.pascalpex.deepslatemc.util;
+
+import net.pascalpex.deepslatemc.Config;
+import org.bukkit.Bukkit;
+import org.bukkit.entity.Player;
+
+public class TablistUtil {
+
+    private static boolean initiated = false;
+    private static String header = "";
+    private static String footer = "";
+
+    public static void setTablist(Player player) {
+        if (!initiated) {
+            reloadTablist();
+        }
+        player.setPlayerListHeader(header);
+        player.setPlayerListFooter(footer);
+    }
+
+    public static void reloadTablist() {
+        if(!Config.getTablistEnabled()) {
+            header = null;
+            footer = null;
+            initiated = true;
+            for (Player player : Bukkit.getOnlinePlayers()) {
+                setTablist(player);
+            }
+            return;
+        }
+
+        header = "";
+        footer = "";
+        for (String s : Config.getHeader()) {
+            if (!header.equals("")) {
+                header += "\n";
+            }
+            header += s;
+        }
+        for (String s : Config.getFooter()) {
+            if (!footer.equals("")) {
+                footer += "\n";
+            }
+            footer += s;
+        }
+        header = header.replace("&", "§");
+        footer = footer.replace("&", "§");
+
+        initiated = true;
+
+        for (Player player : Bukkit.getOnlinePlayers()) {
+            setTablist(player);
+        }
+    }
+
+}
