From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pascalpex <pascalpex@gmail.com>
Date: Wed, 22 May 2024 22:24:30 +0200
Subject: [PATCH] Actionbar message


diff --git a/src/main/java/net/minecraft/server/commands/ReloadCommand.java b/src/main/java/net/minecraft/server/commands/ReloadCommand.java
index 59c6ba08817fea827988f4add74b4d0545f181a0..f74e7f2a40d3545b73193901445c11e8b721c0c5 100644
--- a/src/main/java/net/minecraft/server/commands/ReloadCommand.java
+++ b/src/main/java/net/minecraft/server/commands/ReloadCommand.java
@@ -11,6 +11,7 @@ import net.minecraft.network.chat.Component;
 import net.minecraft.server.MinecraftServer;
 import net.minecraft.server.packs.repository.PackRepository;
 import net.minecraft.world.level.storage.WorldData;
+import net.pascalpex.deepslatemc.util.ActionbarUtil;
 import net.pascalpex.deepslatemc.util.BossbarUtil;
 import net.pascalpex.deepslatemc.util.TablistUtil;
 import org.slf4j.Logger;
@@ -55,6 +56,7 @@ public class ReloadCommand {
         minecraftserver.reloadResources(collection1, io.papermc.paper.event.server.ServerResourcesReloadedEvent.Cause.PLUGIN); // Paper - Add ServerResourcesReloadedEvent
         TablistUtil.reloadTablist();
         BossbarUtil.reloadBossbar();
+        ActionbarUtil.reloadActionbar();
     }
     // CraftBukkit end
 
diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
index 72812eeab349164a1bc66bdce4008b269784bb1c..43f00c976c0f9cf0470a1867fe6f5f99ce778110 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
@@ -62,6 +62,7 @@ import net.minecraft.world.level.block.entity.SkullBlockEntity;
 import net.minecraft.world.level.storage.LevelStorageSource;
 import net.pascalpex.deepslatemc.Config;
 import net.pascalpex.deepslatemc.commands.*;
+import net.pascalpex.deepslatemc.util.ActionbarUtil;
 import org.galemc.gale.command.GaleCommands;
 import org.galemc.gale.configuration.GaleGlobalConfiguration;
 import org.slf4j.Logger;
@@ -286,6 +287,7 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
         MinecraftServer.getServer().server.getCommandMap().register("spawn", "Deepslate", new SpawnCommand("spawn"));
         MinecraftServer.getServer().server.getCommandMap().register("lobby", "Deepslate", new SpawnCommand("lobby"));
         MinecraftServer.getServer().server.getCommandMap().register("autosmelt", "Deepslate", new AutosmeltCommand("autosmelt"));
+        ActionbarUtil.reloadActionbar();
         // Deepslate end
 
         this.setPvpAllowed(dedicatedserverproperties.pvp);
diff --git a/src/main/java/net/minecraft/server/players/PlayerList.java b/src/main/java/net/minecraft/server/players/PlayerList.java
index 959d4c88a1caa8847c4a98a3f10972854d75a9c0..21c3fd96329ae1f1e8899f35c4a697bcec3e1a17 100644
--- a/src/main/java/net/minecraft/server/players/PlayerList.java
+++ b/src/main/java/net/minecraft/server/players/PlayerList.java
@@ -108,6 +108,7 @@ import net.minecraft.world.scores.PlayerTeam;
 import net.pascalpex.deepslatemc.Config;
 import net.pascalpex.deepslatemc.MessagesFile;
 import net.pascalpex.deepslatemc.commands.SpawnCommand;
+import net.pascalpex.deepslatemc.util.ActionbarUtil;
 import net.pascalpex.deepslatemc.util.BossbarUtil;
 import net.pascalpex.deepslatemc.util.TablistUtil;
 import org.galemc.gale.configuration.GaleGlobalConfiguration;
@@ -381,6 +382,11 @@ public abstract class PlayerList {
             BossbarUtil.setBossbar(bukkitPlayer);
         }
         // Deepslate end
+        // Deepslate start - Actionbar
+        if (Config.getActionbarEnabled()) {
+            ActionbarUtil.sendNow(bukkitPlayer);
+        }
+        // Deepslate end
         this.cserver.getPluginManager().callEvent(playerJoinEvent);
 
         if (!player.connection.isAcceptingMessages()) {
diff --git a/src/main/java/net/pascalpex/deepslatemc/Config.java b/src/main/java/net/pascalpex/deepslatemc/Config.java
index 21593cf6dc3880c2386c17b48192588dde3cf33b..ba1302586e99f28d3f6d41b40d56ce10e610066a 100644
--- a/src/main/java/net/pascalpex/deepslatemc/Config.java
+++ b/src/main/java/net/pascalpex/deepslatemc/Config.java
@@ -88,6 +88,10 @@ public class Config {
                 config.set("bossbar" + ".style", BossBar.Overlay.PROGRESS.name());
                 config.set("bossbar" + ".text", "&6Custom Bossbar");
             }
+            if (!config.contains("actionbar")) {
+                config.set("actionbar" + ".enabled", false);
+                config.set("actionbar" + ".text", "&6Custom Actionbar");
+            }
             save();
         } catch (IOException | InvalidConfigurationException e) {
             e.printStackTrace();
@@ -194,5 +198,11 @@ public class Config {
     public static float getBossbarProgress() {
         return (float) config.getDouble("bossbar" + ".progress");
     }
+    public static boolean getActionbarEnabled() {
+        return config.getBoolean("actionbar" + ".enabled");
+    }
+    public static String getActionbarText() {
+        return config.getString("actionbar" + ".text");
+    }
 
 }
diff --git a/src/main/java/net/pascalpex/deepslatemc/commands/DeepslateCommand.java b/src/main/java/net/pascalpex/deepslatemc/commands/DeepslateCommand.java
index b4f13f00a0a3524606cdd0441ea82de0eb415de0..cab1ed796ae30d1c95c955538d75d1f1e828870b 100644
--- a/src/main/java/net/pascalpex/deepslatemc/commands/DeepslateCommand.java
+++ b/src/main/java/net/pascalpex/deepslatemc/commands/DeepslateCommand.java
@@ -2,6 +2,7 @@ package net.pascalpex.deepslatemc.commands;
 
 import net.pascalpex.deepslatemc.Config;
 import net.pascalpex.deepslatemc.MessagesFile;
+import net.pascalpex.deepslatemc.util.ActionbarUtil;
 import net.pascalpex.deepslatemc.util.BossbarUtil;
 import net.pascalpex.deepslatemc.util.TablistUtil;
 import org.bukkit.ChatColor;
@@ -49,6 +50,7 @@ public class DeepslateCommand extends Command {
                             Config.load();
                             TablistUtil.reloadTablist();
                             BossbarUtil.reloadBossbar();
+                            ActionbarUtil.reloadActionbar();
                             player.sendMessage(prefix + MessagesFile.getConfigReloaded());
                         } else {
                             if (args[0].equalsIgnoreCase("version")) {
@@ -75,6 +77,7 @@ public class DeepslateCommand extends Command {
                         Config.load();
                         TablistUtil.reloadTablist();
                         BossbarUtil.reloadBossbar();
+                        ActionbarUtil.reloadActionbar();
                         sender.sendMessage(prefix + MessagesFile.getConfigReloaded());
                     } else {
                         if (args[0].equalsIgnoreCase("version")) {
diff --git a/src/main/java/net/pascalpex/deepslatemc/util/ActionbarUtil.java b/src/main/java/net/pascalpex/deepslatemc/util/ActionbarUtil.java
new file mode 100644
index 0000000000000000000000000000000000000000..1076bc2a27032668daf2966ef945e7290866e60e
--- /dev/null
+++ b/src/main/java/net/pascalpex/deepslatemc/util/ActionbarUtil.java
@@ -0,0 +1,37 @@
+package net.pascalpex.deepslatemc.util;
+
+import net.kyori.adventure.text.Component;
+import net.pascalpex.deepslatemc.Config;
+import org.bukkit.Bukkit;
+import org.bukkit.entity.Player;
+import java.util.Timer;
+import java.util.TimerTask;
+
+public class ActionbarUtil {
+
+    private static String title;
+    private static final Timer timer = new Timer();
+    private static TimerTask task;
+
+    public static void sendNow(Player player) {
+        player.sendActionBar(Component.text(title));
+    }
+
+    public static void reloadActionbar() {
+        if (task != null) {
+            task.cancel();
+        }
+        task = new TimerTask() {
+            @Override
+            public void run() {
+                for (Player player : Bukkit.getOnlinePlayers()) {
+                    player.sendActionBar(Component.text(title));
+                }
+            }
+        };
+        if (Config.getActionbarEnabled()) {
+            title = Config.getActionbarText().replace("&", "§");
+            timer.scheduleAtFixedRate(task, 500, 1500);
+        }
+    }
+}
