--- a/net/minecraft/CrashReport.java
+++ b/net/minecraft/CrashReport.java
@@ -23,7 +_,7 @@
 public class CrashReport {
     private static final Logger LOGGER = LogUtils.getLogger();
     private static final DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss", Locale.ROOT);
-    private static final List<String> EXTRA_INFO = List.of("", "DO NOT REPORT THIS TO PAPER! REPORT TO LEAF INSTEAD!", ""); // Leaf - Rebrand
+    private static final List<String> EXTRA_INFO = List.of("", "DO NOT REPORT THIS TO PAPER! REPORT TO DEEPSLATEMC INSTEAD!", ""); // Leaf - Rebrand // Deepslate
     private final String title;
     private final Throwable exception;
     private final List<CrashReportCategory> details = Lists.newArrayList();
