--- a/leaf-api/build.gradle.kts
+++ b/leaf-api/build.gradle.kts
@@ -112,18 +_,22 @@
             srcDir(generatedDir)
             // Leaf start - project setup
             srcDir(file("../paper-api/src/main/java"))
+            srcDir(file("../leaf-api/src/main/java")) // Deepslate - project setup
         }
         resources {
             srcDir(file("../paper-api/src/main/resources"))
+            srcDir(file("../leaf-api/src/main/resources")) // Deepslate - project setup
         }
     }
     test {
         java {
             srcDir(file("../paper-api/src/test/java"))
+            srcDir(file("../leaf-api/src/test/java")) // Deepslate - project setup
         }
         resources {
             srcDir(file("../paper-api/src/test/resources"))
             // Leaf end - project setup
+            srcDir(file("../leaf-api/src/test/resources")) // Deepslate - project setup
         }
     }
 }
